--- autofs-4.1.4/samples/rc.autofs.in	2005-04-11 13:30:54.000000000 +0200
+++ autofs-4.1.4/samples/rc.autofs.in-new	2007-07-15 11:49:40.000000000 +0200
@@ -433,7 +433,7 @@
 		done
 	done
 
-	if [ -n "`/sbin/pidof $DAEMON`" ] ; then
+	if [ -n "`/bin/pidof $DAEMON`" ] ; then
 		RETVAL=1
 	fi
 
@@ -530,15 +530,15 @@
 		RETVAL=$?
 	fi
 	count=0
-	while [ -n "`/sbin/pidof $DAEMON`" -a $count -lt 10 ] ; do
+	while [ -n "`/bin/pidof $DAEMON`" -a $count -lt 10 ] ; do
 	   killproc $DAEMON -USR2 >& /dev/null
 	   RETVAL=$?
-	   [ $RETVAL = 0 -a -z "`/sbin/pidof $DAEMON`" ] || sleep 3
+	   [ $RETVAL = 0 -a -z "`/bin/pidof $DAEMON`" ] || sleep 3
 	   count=`expr $count + 1`
 	done
 	umount -a -f -t autofs
 	rm -f /var/lock/subsys/autofs
-	if [ -n "`/sbin/pidof $DAEMON`" ] ; then
+	if [ -n "`/bin/pidof $DAEMON`" ] ; then
 	    failure "$prog shutdown"
 	else
 	    success "$prog shutdown"
