Just in case OPEN_MAX isn't defined....

--- autofs-4.0.0pre7/modules/lookup_program.c.open_max	Thu Jun 15 17:54:06 2000
+++ autofs-4.0.0pre7/modules/lookup_program.c	Thu Jun 15 17:54:50 2000
@@ -39,6 +39,12 @@
 
 #define MAPENT_MAX_LEN 4095
 
+#ifdef FOPEN_MAX
+#ifndef OPEN_MAX
+#define OPEN_MAX FOPEN_MAX
+#endif
+#endif
+
 struct lookup_context {
   const char *mapname;
   struct parse_mod *parse;
