--- autofs-3.1.7/samples/rc.autofs.in.orig	Sat Nov  4 07:05:18 2000
+++ autofs-3.1.7/samples/rc.autofs.in	Wed Jun 19 13:03:16 2002
@@ -34,7 +34,7 @@
 # Determine which kind of configuration we're using
 #
 system=unknown
-if [ -f /etc/debian_version ]; then
+if [ -f /etc/debian_version -o -f /etc/stm-release ]; then
     system=debian
 elif [ -f /etc/redhat-release ]; then
     system=redhat
@@ -157,7 +157,7 @@
 	fi
 	;;
   stop)
-	kill -TERM $(/sbin/pidof /usr/sbin/automount)
+	kill -TERM $(/bin/pidof /usr/sbin/automount)
 	rm -f /var/lock/subsys/autofs
 	;;
   reload|restart)
