--- autofs-4.1.4/Makefile.conf.in	2013-12-16 02:37:42.000000000 +0100
+++ autofs-4.1.4/Makefile.conf.in.patched	2013-12-16 02:23:22.000000000 +0100
@@ -37,6 +37,9 @@
 # Support for calling e3fsck when mounting ext3 filesystems
 EXT3FS = @HAVE_E3FSCK@
 
+# Support for calling e4fsck when mounting ext4 filesystems
+EXT4FS = @HAVE_E4FSCK@
+
 #
 # Note: the DESTDIR define is so you can build autofs into a temporary
 # directory and still have all the compiled-in paths point to the right
--- autofs-4.1.4/configure.in	2013-12-16 02:37:42.000000000 +0100
+++ autofs-4.1.4/configure.in.patched	2013-12-16 02:28:26.000000000 +0100
@@ -42,6 +42,7 @@
 AF_PATH_INCLUDE(UMOUNT, umount, /bin/umount, $searchpath)
 AF_PATH_INCLUDE(E2FSCK, fsck.ext2 e2fsck, , $searchpath)
 AF_PATH_INCLUDE(E3FSCK, fsck.ext3 e3fsck, , $searchpath)
+AF_PATH_INCLUDE(E4FSCK, fsck.ext4 e4fsck, , $searchpath)
 
 #
 # Newer mounts have the -s (sloppy) option to ignore unknown options,
--- autofs-4.1.4/include/config.h.in	2004-02-03 16:23:21.000000000 +0100
+++ autofs-4.1.4/include/config.h.in.patched	2013-12-16 02:29:39.000000000 +0100
@@ -21,6 +21,9 @@
 #undef HAVE_E3FSCK
 #undef PATH_E3FSCK
 
+#undef HAVE_E4FSCK
+#undef PATH_E4FSCK
+
 /* Define this option if mount(8) supports the -s (sloppy) option */
 #undef HAVE_SLOPPY_MOUNT
 
--- autofs-4.1.4/modules/Makefile	2013-12-16 03:02:26.000000000 +0100
+++ autofs-4.1.4/modules/Makefile.patched	2013-12-16 03:02:17.000000000 +0100
@@ -59,10 +59,16 @@
  ifeq ($(EXT3FS), 1)
 	ln -fs mount_ext2.so $(INSTALLROOT)$(autofslibdir)/mount_ext3.so
  endif
-else
- ifeq ($(EXT3FS), 1)
+ ifeq ($(EXT4FS), 1)
+	ln -fs mount_ext2.so $(INSTALLROOT)$(autofslibdir)/mount_ext4.so
+ endif
+else ifeq ($(EXT3FS), 1)
 	mv $(INSTALLROOT)$(autofslibdir)/mount_ext2.so $(INSTALLROOT)$(autofslibdir)/mount_ext3.so
+ ifeq ($(EXT4FS), 1)
+	ln -fs mount_ext3.so $(INSTALLROOT)$(autofslibdir)/mount_ext4.so
  endif
+else ifeq ($(EXT4FS), 1)
+	mv $(INSTALLROOT)$(autofslibdir)/mount_ext2.so $(INSTALLROOT)$(autofslibdir)/mount_ext4.so
 endif
 
 #
--- autofs-4.1.4/modules/mount_ext2.c	2005-01-10 14:28:29.000000000 +0100
+++ autofs-4.1.4/modules/mount_ext2.c.patched	2013-12-16 02:33:54.000000000 +0100
@@ -83,13 +83,14 @@
 			ro = 1;
 	}
 
+	fsck_prog = PATH_E2FSCK;
 #ifdef HAVE_E3FSCK
-	if (!strcmp(fstype,"ext3") || !strcmp(fstype,"auto"))
+	if (!strcmp(fstype,"ext3"))
 		fsck_prog = PATH_E3FSCK;
-	else
-		fsck_prog = PATH_E2FSCK;
-#else
-	fsck_prog = PATH_E2FSCK;
+#endif
+#ifdef HAVE_E4FSCK
+	if (!strcmp(fstype,"ext4"))
+		fsck_prog = PATH_E4FSCK;
 #endif
 	if (ro) {
 		debug(MODPREFIX "calling %s -n %s", fsck_prog, what);
