--- busybox-1.8.2/modutils/insmod.c	2007-11-10 02:40:49.000000000 +0100
+++ busybox-1.8.2/modutils/insmod.c-new	2007-12-12 10:58:11.000000000 +0100
@@ -3903,7 +3903,7 @@
 	char *opt_o, *arg1;
 	int len;
 	int k_crcs;
-	char *tmp, *tmp1;
+	char *tmp, *tmp1, *tmp2;
 	unsigned long m_size;
 	ElfW(Addr) m_addr;
 	struct obj_file *f;
@@ -3976,6 +3976,14 @@
 	}
 
 	/* Get a filedesc for the module.  Check that we have a complete path */
+	/* first look in /var/lib/modules */
+	tmp2 = alloca(strlen(m_fullName) + sizeof "/var/lib/modules//");
+	strcpy(tmp2, "/var/lib/modules/");
+	strcat(tmp2, m_fullName);
+	if (stat(tmp2, &st) >= 0 && S_ISREG(st.st_mode) && (fp = fopen(tmp2, "r")) != NULL) {
+		m_filename = xstrdup(tmp2);
+		printf("insmod: preferring module %s\n", m_filename);
+	} else
 	if (stat(arg1, &st) < 0 || !S_ISREG(st.st_mode)
 	 || (fp = fopen(arg1, "r")) == NULL
 	) {
