diff -u --recursive --new-file oldenigma2/lib/gdi/fb.h enigma2/lib/gdi/fb.h
--- oldenigma2/lib/gdi/fb.h	2008-03-25 16:59:38.000000000 +0100
+++ enigma2/lib/gdi/fb.h	2008-06-15 19:52:52.000000000 +0200
@@ -18,7 +18,7 @@
 	int m_manual_blit;
 	int m_number_of_pages;
 #ifdef SWIG
-	fbClass(const char *fb="/dev/fb/0");
+	fbClass(const char *fb="/dev/fb0");
 	~fbClass();
 public:
 #else
@@ -38,7 +38,7 @@
 	unsigned int Stride() { return stride; }
 	fb_cmap *CMAP() { return &cmap; }
 
-	fbClass(const char *fb="/dev/fb/0");
+	fbClass(const char *fb="/dev/fb0");
 	~fbClass();
 	
 			// low level gfx stuff 
