diff -Nhur e2fsprogs-1.40.2.org/lib/Makefile.elf-lib e2fsprogs-1.40.2/lib/Makefile.elf-lib
--- e2fsprogs-1.40.2.org/lib/Makefile.elf-lib	2010-02-13 00:06:11.429797150 +0100
+++ e2fsprogs-1.40.2/lib/Makefile.elf-lib	2010-02-13 00:15:08.411781301 +0100
@@ -45,7 +45,7 @@
 	@echo "	SYMLINK $(ELF_INSTALL_DIR)/$(ELF_SONAME)"
 	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
 	@echo "	SYMLINK $(libdir)/$(ELF_IMAGE).so"
-	@$(LN_S) -f $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
+	@$(LN_S) -f ../../lib/$(ELF_SONAME) \
 		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
 	@echo "	LDCONFIG"
 	@-$(LDCONFIG)
