diff -urN enigma2-nightly.org/keyids.py enigma2-nightly/keyids.py
--- enigma2-nightly.org/keyids.py	2009-12-13 18:46:44.000000000 +0100
+++ enigma2-nightly/keyids.py	2009-09-27 18:20:56.000000000 +0200
@@ -11,6 +11,60 @@
 "KEY_8": 9, 
 "KEY_9": 10, 
 "KEY_0": 11, 
+"KEY_RADIO": 59, 
+"KEY_RECALL": 60, 
+"KEY_AUDIO": 61, 
+"KEY_PHOTO": 62, 
+"KEY_SELECT": 63, 
+"KEY_AGAIN": 64, 
+"KEY_PIP": 65, 
+"KEY_FP_UP": 66, 
+"KEY_FP_DOWN": 67, 
+"KEY_FP_LEFT": 68, 
+"KEY_FP_RIGHT": 87, 
+"KEY_FP_OK": 88, 
+"KEY_UP": 103, 
+"KEY_PAGEUP": 104, 
+"KEY_LEFT": 105, 
+"KEY_RIGHT": 106, 
+"KEY_DOWN": 108, 
+"KEY_PAGEDOWN": 109, 
+"KEY_MUTE": 113, 
+"KEY_POWER": 116, 
+"KEY_PAUSE": 119, 
+"KEY_STOP": 128, 
+"KEY_MENU": 139, 
+"KEY_FILE": 144, 
+"KEY_WWW": 150, 
+"KEY_TV": 150, 
+"KEY_EXIT": 158, 
+"KEY_RECORD": 167, 
+"KEY_REWIND": 168, 
+"KEY_FP_BACK": 183, 
+"KEY_FP_MENU": 184, 
+"KEY_FP_FILE": 185, 
+"KEY_FP_POWER": 186, 
+"KEY_VOLUMEUP": 187, 
+"KEY_VOLUMEDOWN": 188, 
+"KEY_PAGES": 189, 
+"KEY_MOSAIC": 190, 
+"KEY_PIP_SWAP": 191, 
+"KEY_SUB_CHANNEL": 192, 
+"KEY_PLAYER": 193, 
+"KEY_MUSIC": 193, 
+"KEY_PLAY": 207, 
+"KEY_FASTFORWARD": 208, 
+"KEY_OK": 352, 
+"KEY_INFO": 358, 
+"KEY_FAVORITES": 364, 
+"KEY_EPG": 365, 
+"KEY_SUBTITLE": 370, 
+"KEY_TEXT": 371, 
+"KEY_RED": 398, 
+"KEY_GREEN": 399, 
+"KEY_YELLOW": 400, 
+"KEY_BLUE": 401, 
+"KEY_PLAYPAUSE": 409, 
 "KEY_MINUS": 12, 
 "KEY_EQUAL": 13, 
 "KEY_BACKSPACE": 14, 
@@ -102,23 +156,12 @@
 "KEY_RIGHTALT": 100, 
 "KEY_LINEFEED": 101, 
 "KEY_HOME": 102, 
-"KEY_UP": 103, 
-"KEY_PAGEUP": 104, 
-"KEY_LEFT": 105, 
-"KEY_RIGHT": 106, 
 "KEY_END": 107, 
-"KEY_DOWN": 108, 
-"KEY_PAGEDOWN": 109, 
 "KEY_INSERT": 110, 
 "KEY_DELETE": 111, 
 "KEY_MACRO": 112, 
-"KEY_MUTE": 113, 
-"KEY_VOLUMEDOWN": 114, 
-"KEY_VOLUMEUP": 115, 
-"KEY_POWER": 116, 
 "KEY_KPEQUAL": 117, 
 "KEY_KPPLUSMINUS": 118, 
-"KEY_PAUSE": 119, 
 "KEY_F21": 120, 
 "KEY_F22": 121, 
 "KEY_F23": 122, 
@@ -127,8 +170,6 @@
 "KEY_LEFTMETA": 125, 
 "KEY_RIGHTMETA": 126, 
 "KEY_COMPOSE": 127, 
-"KEY_STOP": 128, 
-"KEY_AGAIN": 129, 
 "KEY_PROPS": 130, 
 "KEY_UNDO": 131, 
 "KEY_FRONT": 132, 
@@ -138,18 +179,16 @@
 "KEY_FIND": 136, 
 "KEY_CUT": 137, 
 "KEY_HELP": 138, 
-"KEY_MENU": 139, 
+"KEY_PREVIOUS": 138, 
 "KEY_CALC": 140, 
 "KEY_SETUP": 141, 
 "KEY_SLEEP": 142, 
 "KEY_WAKEUP": 143, 
-"KEY_FILE": 144, 
 "KEY_SENDFILE": 145, 
 "KEY_DELETEFILE": 146, 
 "KEY_XFER": 147, 
 "KEY_PROG1": 148, 
 "KEY_PROG2": 149, 
-"KEY_WWW": 150, 
 "KEY_MSDOS": 151, 
 "KEY_COFFEE": 152, 
 "KEY_DIRECTION": 153, 
@@ -157,23 +196,16 @@
 "KEY_MAIL": 155, 
 "KEY_BOOKMARKS": 156, 
 "KEY_COMPUTER": 157, 
-"KEY_BACK": 158, 
-"KEY_FORWARD": 159, 
 "KEY_CLOSECD": 160, 
 "KEY_EJECTCD": 161, 
 "KEY_EJECTCLOSECD": 162, 
-"KEY_NEXTSONG": 163, 
-"KEY_PLAYPAUSE": 164, 
-"KEY_PREVIOUSSONG": 165, 
 "KEY_STOPCD": 166, 
-"KEY_RECORD": 167, 
-"KEY_REWIND": 168, 
 "KEY_PHONE": 169, 
 "KEY_ISO": 170, 
 "KEY_CONFIG": 171, 
 "KEY_HOMEPAGE": 172, 
 "KEY_REFRESH": 173, 
-"KEY_EXIT": 174, 
+"KEY_BACK": 174, 
 "KEY_MOVE": 175, 
 "KEY_EDIT": 176, 
 "KEY_SCROLLUP": 177, 
@@ -204,8 +236,6 @@
 "KEY_PROG4": 203, 
 "KEY_SUSPEND": 205, 
 "KEY_CLOSE": 206, 
-"KEY_PLAY": 207, 
-"KEY_FASTFORWARD": 208, 
 "KEY_BASSBOOST": 209, 
 "KEY_PRINT": 210, 
 "KEY_HP": 211, 
@@ -225,32 +255,30 @@
 "KEY_BRIGHTNESSUP": 225, 
 "KEY_MEDIA": 226, 
 "KEY_UNKNOWN": 240, 
-"KEY_OK": 352, 
-"KEY_SELECT": 353, 
+"BTN_0": 256, 
+"BTN_1": 257, 
+"BTN_2": 258, 
+"BTN_3": 259, 
+"BTN_4": 260, 
 "KEY_GOTO": 354, 
 "KEY_CLEAR": 355, 
 "KEY_POWER2": 356, 
 "KEY_OPTION": 357, 
-"KEY_INFO": 358, 
 "KEY_TIME": 359, 
 "KEY_VENDOR": 360, 
 "KEY_ARCHIVE": 361, 
 "KEY_PROGRAM": 362, 
 "KEY_CHANNEL": 363, 
-"KEY_FAVORITES": 364, 
-"KEY_EPG": 365, 
 "KEY_PVR": 366, 
 "KEY_MHP": 367, 
 "KEY_LANGUAGE": 368, 
 "KEY_TITLE": 369, 
-"KEY_SUBTITLE": 370, 
 "KEY_ANGLE": 371, 
 "KEY_ZOOM": 372, 
 "KEY_MODE": 373, 
 "KEY_KEYBOARD": 374, 
 "KEY_SCREEN": 375, 
 "KEY_PC": 376, 
-"KEY_TV": 377, 
 "KEY_TV2": 378, 
 "KEY_VCR": 379, 
 "KEY_VCR2": 380, 
@@ -258,34 +286,24 @@
 "KEY_SAT2": 382, 
 "KEY_CD": 383, 
 "KEY_TAPE": 384, 
-"KEY_RADIO": 385, 
 "KEY_TUNER": 386, 
-"KEY_PLAYER": 387, 
-"KEY_TEXT": 388, 
 "KEY_DVD": 389, 
 "KEY_AUX": 390, 
 "KEY_MP3": 391, 
-"KEY_AUDIO": 392, 
 "KEY_VIDEO": 393, 
 "KEY_DIRECTORY": 394, 
 "KEY_LIST": 395, 
 "KEY_MEMO": 396, 
 "KEY_CALENDAR": 397, 
-"KEY_RED": 398, 
-"KEY_GREEN": 399, 
-"KEY_YELLOW": 400, 
-"KEY_BLUE": 401, 
 "KEY_CHANNELUP": 402, 
 "KEY_CHANNELDOWN": 403, 
 "KEY_FIRST": 404, 
 "KEY_LAST": 405, 
 "KEY_AB": 406, 
-"KEY_NEXT": 407, 
 "KEY_RESTART": 408, 
 "KEY_SLOW": 409, 
 "KEY_SHUFFLE": 410, 
 "KEY_BREAK": 411, 
-"KEY_PREVIOUS": 412, 
 "KEY_DIGITS": 413, 
 "KEY_TEEN": 414, 
 "KEY_TWEN": 415, 
@@ -293,8 +311,10 @@
 "KEY_DEL_EOS": 449, 
 "KEY_INS_LINE": 450, 
 "KEY_DEL_LINE": 451, 
-"KEY_ASCII": 510,
+"KEY_ASCII": 510, 
 "KEY_MAX": 511, 
-"BTN_0": 256, 
-"BTN_1": 257, 
+"KEY_PREVIOUSSONG": 512, 
+"KEY_FORWARD": 513, 
+"KEY_NEXTSONG": 514, 
+"KEY_NEXT": 515, 
 }
diff -urN enigma2-nightly.org/lib/driver/rcinput.cpp enigma2-nightly/lib/driver/rcinput.cpp
--- enigma2-nightly.org/lib/driver/rcinput.cpp	2009-12-13 18:46:44.000000000 +0100
+++ enigma2-nightly/lib/driver/rcinput.cpp	2009-12-14 10:55:12.000000000 +0100
@@ -5,12 +5,97 @@
 #include <sys/ioctl.h>
 #include <linux/input.h>
 #include <sys/stat.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/un.h>
+
+#include <stdio.h>
+#include <fcntl.h>
+#include <stdlib.h>
 
 #include <lib/base/ebase.h>
 #include <lib/base/init.h>
 #include <lib/base/init_num.h>
 #include <lib/driver/input_fake.h>
 
+static int sockethandle = -1;
+//static bool isTuxTxt = false;
+#define SocketName "/tmp/rc.socket"
+
+int checkTuxTxt(int code)
+{
+    int tmp_o;
+    /*bool prev=true;
+
+    if ( isTuxTxt==false )
+    {
+	if(code==KEY_TEXT)
+	{
+		isTuxTxt=true;
+		prev=false;
+	}
+    }*/
+    
+    if /*( isTuxTxt )*/ ((tmp_o = open("/tmp/block.tmp", O_RDONLY)) >= 0)
+    {
+	close(tmp_o);
+
+ 	fprintf(stderr, "Forwarding to Socket-> %u\n", sockethandle);
+	
+	if (sockethandle <= 0)
+	{
+		struct sockaddr_un addr;
+		addr.sun_family = AF_UNIX;
+		strcpy( addr.sun_path, SocketName );
+		sockethandle = socket( PF_UNIX, SOCK_STREAM, 0 );
+		
+		if ( sockethandle <= 0 )  
+		{
+			fprintf(stderr, "No RemoteControlSocket attached!\n");
+			return 0;
+		};
+
+		if(connect(sockethandle,(struct sockaddr *)&addr,sizeof(addr))!=0)  
+		{
+			fprintf(stderr, "connect failed!\n");
+			return 0;
+		}
+	}
+		
+	if ( sockethandle > 0 )
+	{
+		char * tmp_s = (char*) malloc(sizeof("00000000"));
+		sprintf(tmp_s, "%08d", code);
+
+		if (write(sockethandle, (void *) tmp_s, sizeof("00000000")) <= 0)
+			fprintf(stderr, "Error while forwarding!");
+	} else
+		fprintf(stderr, "Error while forwarding!-");
+
+	/*switch(code)
+	{
+		case KEY_TEXT:
+			if(prev==true)
+				isTuxTxt=false;
+			break;			
+		case KEY_HOME:
+		case 158: //KEY_EXIT:
+		case KEY_POWER:
+			isTuxTxt=false;
+			break;
+	}*/
+
+	return 1;
+    }
+
+    if (sockethandle != -1)
+    {
+	close (sockethandle);
+	sockethandle = -1;
+    }
+    return 0;
+}
+
 void eRCDeviceInputDev::handleCode(long rccode)
 {
 	struct input_event *ev = (struct input_event *)rccode;
@@ -19,6 +104,11 @@
 
 //	eDebug("%x %x %x", ev->value, ev->code, ev->type);
 
+	if(checkTuxTxt(ev->code)==1)
+	{
+		return;
+	}
+
 	if (ev->type!=EV_KEY)
 		return;
 
diff -urN enigma2-nightly.org/lib/driver/vfd.cpp enigma2-nightly/lib/driver/vfd.cpp
--- enigma2-nightly.org/lib/driver/vfd.cpp	2009-12-13 18:46:44.000000000 +0100
+++ enigma2-nightly/lib/driver/vfd.cpp	2009-09-27 18:20:57.000000000 +0200
@@ -12,25 +12,34 @@
 #include <lib/base/eerror.h>
 #include <lib/driver/vfd.h>
 
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD) || defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+#define PLATFORM_FRONT_VFD
+#endif
+
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_MINI_FTA) || defined(PLATFORM_CUBEREVO_250HD) || defined(PLATFORM_CUBEREVO_2000HD) || defined(PLATFORM_CUBEREVO_9500HD)
+#define VFD_DEVICE "/dev/dbox/fp0"
+#else
 #ifdef PLATFORM_TF7700
 #include "../../misc/tools/tffpctl/frontpanel.h"
 #endif
-
 #define VFD_DEVICE "/dev/vfd"
 #define VFDICONDISPLAYONOFF	0xc0425a0a
 #define	VFDDISPLAYCHARS 	0xc0425a00
 #define VFDBRIGHTNESS           0xc0425a03
 //light on off
 #define VFDDISPLAYWRITEONOFF    0xc0425a05
+#endif
 
 bool startloop_running = false;
-static bool icon_onoff[32];
-static pthread_t thread_start_loop = 0;
+int scroll = 0;
+static bool icon_onoff[CLEAR-USB+1];
+static pthread_t thread_start_loop = NULL;
+static pthread_t thread_vfd_scrollText = NULL;
 void * start_loop (void *arg);
 bool blocked = false;
 bool requested = false;
 char chars[64];
-
+char g_str[64];
 
 struct vfd_ioctl_data {
 	unsigned char start;
@@ -58,13 +67,12 @@
 
 void evfd::init()
 {
-	pthread_create (&thread_start_loop, NULL, &start_loop, NULL);	
+	pthread_create (&thread_start_loop, NULL, start_loop, NULL);	
 	return;
 }
 
 evfd::~evfd()
 {
-	//close (file_vfd);
 }
 
 #ifdef PLATFORM_TF7700
@@ -179,7 +179,7 @@
 	vfd.vfd_write_string(" Team Ducktales ", true);
 	//run 2 times through all icons 
 	for  (int vloop = 0; vloop < 128; vloop++) {
-		/*if (vloop%14 == 0 )
+		if (vloop%14 == 0 )
 			vfd.vfd_set_brightness(1);
 		else if (vloop%14 == 1 )
 			vfd.vfd_set_brightness(2);
@@ -206,20 +214,20 @@
 		else if (vloop%14 == 12 )
 			vfd.vfd_set_brightness(1);
 		else if (vloop%14 == 13 )
-			vfd.vfd_set_brightness(0);*/
-		if (vloop%2 == 1) {
+			vfd.vfd_set_brightness(0);
+		/*if (vloop%2 == 1) {
 			vfd.vfd_set_icon( (tvfd_icon) (((vloop%32)/2)%16), ICON_OFF, true);
 			//usleep(1000);
 			vfd.vfd_set_icon( (tvfd_icon) ((((vloop%32)/2)%16)+1), ICON_ON, true);
-		}
+		}*/
 		usleep(75000);
 	}
 	vfd.vfd_set_brightness(7);
 	
 	//set all blocked icons
-	for (int id = 0x10; id < 0x20; id++) {
+	/*for (int id = USB; id < (CLEAR+1); id++) {
 		vfd.vfd_set_icon((tvfd_icon)id, icon_onoff[id]);	
-	}
+	}*/
 
 	blocked = false;
 	return NULL;
@@ -229,68 +237,77 @@
 
 //////////////////////////////////////////////////////////////////////////////////////
 
+void *vfd_write_string_scrollText(void *arg) {
+	pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
+	blocked = true;
+	char *text = (char *) arg;
+	char out[13+1];
+	evfd vfd;
+	bool do_it=true;
+	int i, len;
+
+	text=g_str;
+	len = strlen(text);
+
+	memset(out,0,14);
+	while((do_it) && (len>13))
+	{
+		usleep(500000);
+		for (i=0; i<=(len-13); i++) { // scroll text till end
+			memset(out, ' ', 13);
+			memcpy(out, text+i, 13);
+			vfd.vfd_write_string(out,true);
+			usleep(500000);
+		}
+		memcpy(out, text, 13); // display first 13 chars after scrolling
+		vfd.vfd_write_string(out,true);
+		if(scroll==1)
+			do_it=false;
+	}
+	blocked = false;
+	return NULL;
+}
+
 void evfd::vfd_write_string(char * str)
 {
+	if(blocked)
+	{
+		pthread_cancel(thread_vfd_scrollText);
+		pthread_join(thread_vfd_scrollText, NULL);
+		blocked=false;
+	}
+	memset(g_str,0,64);
+	strcpy(g_str,str);
 	vfd_write_string(str, false);
+	if(scroll>0)
+	{
+		pthread_create(&thread_vfd_scrollText, NULL, vfd_write_string_scrollText, (void *)str);
+	}
 }
 
 void evfd::vfd_write_string(char * str, bool force)
 {
 	int i;
 	i = strlen ( str );
-	if ( i > 63 ) i = 63;
-	memset ( chars, ' ', 63 );
+	if ( i > 13 ) i = 13;
+	memset ( chars, ' ', 13 );
 	memcpy ( chars, str, i);	
 
-#ifdef PLATFORM_TF7700
-
-	// request the display to cancel the start loop
-	requested = true;
-	while(blocked) usleep(200000);
-
-	{
-#else
 	if (!blocked || force) {
-#endif
 		struct vfd_ioctl_data data;
-		memset ( data.data, ' ', 63 );
+		memset ( data.data, ' ', 13 );
 		memcpy ( data.data, str, i );	
 
 		data.start = 0;
 		data.length = i;
 
 		file_vfd = open (VFD_DEVICE, O_WRONLY);
-		ioctl ( file_vfd, VFDDISPLAYCHARS, &data );
+		write(file_vfd,data.data,data.length);
 		close (file_vfd);
 	}
 	return;
 }
 
-void evfd::vfd_write_string_scrollText(char* text) {
-	if (!blocked) {
-		int i, len = strlen(text);
-		char* out = (char *) malloc(16);
-
-		for (i=0; i<=(len-16); i++) { // scroll text till end
-			memset(out, ' ', 16);
-			memcpy(out, text+i, 16);
-			vfd_write_string(out);
-			usleep(200000);
-		}
-		for (i=1; i<16; i++) { // scroll text with whitespaces from right
-			memset(out, ' ', 16);
-			memcpy(out, text+len+i-16, 16-i);
-			vfd_write_string(out);
-			usleep(200000);
-		}
-
-		memcpy(out, text, 16); // display first 16 chars after scrolling
-		vfd_write_string(out);
-		free (out);
-	}
-	return;
-}
-
 void evfd::vfd_clear_string()
 {
 	vfd_write_string("                ");
@@ -299,14 +316,551 @@
 
 //////////////////////////////////////////////////////////////////////////////////////
 
+static int vfd_symbol( unsigned char grid, unsigned char bit, unsigned char onoff )
+{
+#if defined(PLATFORM_FRONT_VFD)
+	int micom_fd = open("/dev/dbox/fp0", O_RDWR);
+	symbol_t symbol;
+
+	symbol.onoff = onoff; 
+	symbol.grid = grid;
+	symbol.bit = bit;
+
+	if( ioctl(micom_fd,FRONT_IOCS_SYMBOL, &symbol) < 0 )
+	{ 
+		//debug_Error( "ioctl failed(%s)\n", strerror(errno) ); 
+		return -1; 
+	}
+	close(micom_fd);
+#endif
+	return 0;
+}
+
+static int vfd_1gsymbol( unsigned char grid, unsigned char bit, unsigned char onoff )
+{
+#if defined(PLATFORM_FRONT_VFD)
+	int micom_fd = open("/dev/dbox/fp0", O_RDWR);
+	symbol_t symbol;
+
+	symbol.onoff = onoff; 
+	symbol.grid = grid;
+	symbol.bit = bit;
+
+	if( ioctl(micom_fd,FRONT_IOCS_1GSYMBOL, &symbol) < 0 )
+	{ 
+		//debug_Error( "ioctl failed(%s)\n", strerror(errno) ); 
+		return -1; 
+	}
+	close(micom_fd);
+#endif
+	return 0;
+}
+
+int vfd_symbol_clear()
+{
+#if defined(PLATFORM_FRONT_VFD)
+	int micom_fd = open("/dev/dbox/fp0", O_RDWR);
+
+	if( ioctl(micom_fd,FRONT_IOCC_CLRSYMBOL) < 0 )
+	{ 
+		//debug_Error( "ioctl failed(%s)\n", strerror(errno) ); 
+		return -1; 
+	}
+	close(micom_fd);
+#endif
+	return 0;
+}
+
+void vfd_symbol_sat(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+	vfd_1gsymbol( 
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+			0, 2, 
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+			1, 3,
+#endif
+			onoff );
+#endif
+}
+
+void vfd_symbol_ter(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+	vfd_1gsymbol(
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+			1, 2, 
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+			1, 3,
+#endif
+			onoff);
+#endif
+}
+
+void vfd_symbol_480i(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(4, 6, onoff);
+	vfd_1gsymbol(3, 6, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_480p(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(4, 6, onoff);
+	vfd_1gsymbol(2, 6, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_576i(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(1, 6, onoff);
+	vfd_1gsymbol(0, 6, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_576p(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(1, 6, onoff);
+	vfd_1gsymbol(4, 5, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_720p(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(3, 5, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_1080i(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(2, 5, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_1080p(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(1, 5, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_power(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+	//debug_Warning("[VFD] never use the power symbol\n");
+	return;
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(0, 3, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_radio(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(4, 2, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_tv(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(3, 2, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_file(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(2, 2, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_rec(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(0, 0, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+	vfd_1gsymbol(0, 2, onoff);
+#endif
+#endif
+}
+
+void vfd_symbol_timeshift(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(1, 1, onoff);
+	vfd_symbol(2, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_play(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_1gsymbol(4, 0, onoff);
+	vfd_1gsymbol(0, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+	vfd_1gsymbol(1, 2, onoff);
+#endif
+#endif
+}
+
+void vfd_symbol_schedule(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(3, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+	vfd_1gsymbol(0, 3, onoff);
+#endif
+#endif
+}
+
+void vfd_symbol_hd(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(4, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+	vfd_1gsymbol(4, 2, onoff);
+#endif
+#endif
+}
+
+void vfd_symbol_usb(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(5, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_lock(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(6, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_dolby(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(7, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+	vfd_1gsymbol(3, 2, onoff);
+#endif
+#endif
+}
+
+void vfd_symbol_pause(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+	vfd_1gsymbol(2, 2, onoff);
+#endif
+#endif
+}
+
+void vfd_symbol_mute(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(8, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_t1(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(9, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_t2(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(10, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_mp3(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(11, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+void vfd_symbol_repeat(int onoff)
+{
+#if defined(PLATFORM_FRONT_VFD)
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_symbol(12, 1, onoff);
+#elif defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_2000HD)
+
+#endif
+#endif
+}
+
+int do_vfd_sym( char *s, int onoff )
+{	
+	if(!strcmp(s, "sat"))
+		vfd_symbol_sat(onoff);
+	else if(!strcmp(s, "ter"))
+		vfd_symbol_ter(onoff);
+	else if(!strcmp(s, "480i"))
+		vfd_symbol_480i(onoff);
+	else if(!strcmp(s, "480p"))
+		vfd_symbol_480p(onoff);
+	else if(!strcmp(s, "576i"))
+		vfd_symbol_576i(onoff);
+	else if(!strcmp(s, "576p"))
+		vfd_symbol_576p(onoff);
+	else if(!strcmp(s, "720p"))
+		vfd_symbol_720p(onoff);
+	else if(!strcmp(s, "1080i"))
+		vfd_symbol_1080i(onoff);
+	else if(!strcmp(s, "1080p"))
+		vfd_symbol_1080p(onoff);
+	else if(!strcmp(s, "power"))
+		vfd_symbol_power(onoff);
+	else if(!strcmp(s, "radio"))
+		vfd_symbol_radio(onoff);
+	else if(!strcmp(s, "tv"))
+		vfd_symbol_tv(onoff);
+	else if(!strcmp(s, "file"))
+		vfd_symbol_file(onoff);
+	else if(!strcmp(s, "rec"))
+		vfd_symbol_rec(onoff);
+	else if(!strcmp(s, "timeshift"))
+		vfd_symbol_timeshift(onoff);
+	else if(!strcmp(s, "play"))
+		vfd_symbol_play(onoff);
+	else if(!strcmp(s, "schedule"))
+		vfd_symbol_schedule(onoff);
+	else if(!strcmp(s, "hd"))
+		vfd_symbol_hd(onoff);
+	else if(!strcmp(s, "usb"))
+		vfd_symbol_usb(onoff);
+	else if(!strcmp(s, "lock"))
+		vfd_symbol_lock(onoff);
+	else if(!strcmp(s, "dolby"))
+		vfd_symbol_dolby(onoff);
+	else if(!strcmp(s, "mute"))
+		vfd_symbol_mute(onoff);
+	else if(!strcmp(s, "t1"))
+		vfd_symbol_t1(onoff);
+	else if(!strcmp(s, "t2"))
+		vfd_symbol_t2(onoff);
+	else if(!strcmp(s, "mp3"))
+		vfd_symbol_mp3(onoff);
+	else if(!strcmp(s, "repeat"))
+		vfd_symbol_repeat(onoff);
+	else if(!strcmp(s, "clear"))
+		vfd_symbol_clear();
+	else if(!strcmp(s, "pause"))
+		vfd_symbol_pause(onoff);
+
+}
+
 void evfd::vfd_set_icon(tvfd_icon id, bool onoff)
 {
 	vfd_set_icon(id, onoff, false);
 	return;
 }
 
+void evfd::vfd_set_ani(int id)
+{
+	scroll=id;
+#if 0
+	int micom_fd = open("/dev/dbox/fp0", O_RDWR);
+	vfd_ani_t ani;
+			//    on/off | kind(0x7f:no property set)
+	if(on)	ani.kind_cnt = 0x80  | 		0x7f;
+	else	ani.kind_cnt = 0x00  | 		0x7f;
+
+	if(ioctl(micom_fd,FRONT_IOCS_ANIMATION,&ani)<0)
+	{
+		printf("ioctl failed(%s)\n", strerror(errno));
+		return -1;
+	}
+	close(micom_fd);
+#endif
+}
+
 void evfd::vfd_set_icon(tvfd_icon id, bool onoff, bool force)
 {
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_MINI_FTA) || defined(PLATFORM_CUBEREVO_250HD) || defined(PLATFORM_CUBEREVO_2000HD) || defined(PLATFORM_CUBEREVO_9500HD)
+	//icon_onoff[id] = onoff;
+
+	switch(id)
+	{
+		case USB:
+			do_vfd_sym("usb",onoff);
+			break;
+		case HD:
+			do_vfd_sym("hd",onoff);
+			break;
+		case LOCK:
+			do_vfd_sym("lock",onoff);
+			break;
+		case MP3:
+			do_vfd_sym("mp3",onoff);
+			break;
+		case MUSIC:
+			do_vfd_sym("radio",onoff);
+			break;
+		case DD:
+			do_vfd_sym("dolby",onoff);
+			break;
+		case MUTE:
+			do_vfd_sym("mute",onoff);
+			break;
+		case PLAY:
+			do_vfd_sym("play",onoff);
+			break;
+		case PAUSE:
+			do_vfd_sym("pause",onoff);
+			break;
+		case REC:
+			do_vfd_sym("rec",onoff);
+			break;
+		case TIMESHIFT:
+			do_vfd_sym("timeshift",onoff);
+			break;
+		case SAT:
+			do_vfd_sym("sat",onoff);
+			break;
+		case TER:
+			do_vfd_sym("ter",onoff);
+			break;
+		case _480i:
+			do_vfd_sym("480i",onoff);
+			break;
+		case _480p:
+			do_vfd_sym("480p",onoff);
+			break;
+		case _576i:
+			do_vfd_sym("576i",onoff);
+			break;
+		case _576p:
+			do_vfd_sym("576p",onoff);
+			break;
+		case _720p:
+			do_vfd_sym("720p",onoff);
+			break;
+		case _1080i:
+			do_vfd_sym("1080i",onoff);
+			break;
+		case _1080p:
+			do_vfd_sym("1080p",onoff);
+			break;
+		case POWER:
+			do_vfd_sym("power",onoff);
+			break;
+		case TV:
+			do_vfd_sym("tv",onoff);
+			break;
+		case FILE_:
+			do_vfd_sym("file",onoff);
+			break;
+		case SCHEDULE:
+			do_vfd_sym("schedule",onoff);
+			break;
+		case T1:
+			do_vfd_sym("t1",onoff);
+			break;
+		case T2:
+			do_vfd_sym("t2",onoff);
+			break;
+		case REPEAT:
+			do_vfd_sym("repeat",onoff);
+			break;
+		case CLEAR:
+			do_vfd_sym("clear",onoff);
+			break;
+		default:
+			break;
+	}
+#else
 	icon_onoff[id] = onoff;
 
 	if (!blocked || force) {
@@ -325,14 +879,29 @@
 			close (file_vfd);
 		}
 	}
+#endif
 	return;
 }
 
 void evfd::vfd_clear_icons()
 {
-	for (int id = 0x10; id < 0x20; id++) {
-		vfd_set_icon((tvfd_icon)id, false);	
+	int micom_fd = open("/dev/dbox/fp0", O_RDWR);
+
+	if(ioctl(micom_fd, FRONT_IOCC_BUFCLR)<0)
+	{
+		//debug_Error("ioctl failed(%s)\n",strerror(errno));
+		return;
+	}
+	if(ioctl(micom_fd, FRONT_IOCC_SEGUPDATE)<0)
+	{
+		//debug_Error("ioctl failed(%s)\n",strerror(errno));
+		return;
 	}
+	close(micom_fd);
+
+	/*for (int id = USB; id < (CLEAR+1); id++) {
+		vfd_set_icon((tvfd_icon)id, false);	
+	}*/
 	return;
 }
 
@@ -340,6 +909,12 @@
 
 void evfd::vfd_set_brightness(unsigned char setting)
 {
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_MINI_FTA) || defined(PLATFORM_CUBEREVO_250HD) || defined(PLATFORM_CUBEREVO_2000HD) || defined(PLATFORM_CUBEREVO_9500HD)
+	setting &= 0x07;
+
+	file_vfd = open (VFD_DEVICE, O_WRONLY);
+	ioctl ( file_vfd, FRONT_IOCS_VFDBRIGHT, &setting );
+#else
 	struct vfd_ioctl_data data;
 
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
@@ -349,14 +924,23 @@
 
 	file_vfd = open (VFD_DEVICE, O_WRONLY);
 	ioctl ( file_vfd, VFDBRIGHTNESS, &data );
+#endif
 	close (file_vfd);
 
 	return;
 }
 
-
 void evfd::vfd_set_light(bool onoff)
 {
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_MINI_FTA) || defined(PLATFORM_CUBEREVO_250HD) || defined(PLATFORM_CUBEREVO_2000HD) || defined(PLATFORM_CUBEREVO_9500HD)
+	vfd_led_t led;
+	led.type=LED_SET;
+	led.val=(onoff)?1:0;
+	
+	file_vfd = open (VFD_DEVICE, O_WRONLY);
+
+    	ioctl(file_vfd, FRONT_IOCS_VFDLED, &led);
+#else
 	struct vfd_ioctl_data data;
 
 	memset(&data, 0, sizeof(struct vfd_ioctl_data));
@@ -370,7 +954,23 @@
 	file_vfd = open (VFD_DEVICE, O_WRONLY);
 
     	ioctl(file_vfd, VFDDISPLAYWRITEONOFF, &data);
+#endif
+
+	close (file_vfd);
+	return;
+}
+
+void evfd::vfd_set_fan(bool onoff)
+{
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_MINI_FTA) || defined(PLATFORM_CUBEREVO_250HD) || defined(PLATFORM_CUBEREVO_2000HD) || defined(PLATFORM_CUBEREVO_9500HD)
+	file_vfd = open (VFD_DEVICE, O_WRONLY);
+
+	if(onoff)
+	    	ioctl(file_vfd, FRONT_IOCC_FAN_ON);
+	else
+	    	ioctl(file_vfd, FRONT_IOCC_FAN_OFF);
 
 	close (file_vfd);
+#endif
 	return;
 }
diff -urN enigma2-nightly.org/lib/driver/vfd.h enigma2-nightly/lib/driver/vfd.h
--- enigma2-nightly.org/lib/driver/vfd.h	2009-12-13 18:46:44.000000000 +0100
+++ enigma2-nightly/lib/driver/vfd.h	2009-12-14 10:56:10.000000000 +0100
@@ -4,7 +4,94 @@
 #define ICON_ON  1
 #define ICON_OFF 0
 
-typedef enum { USB = 0x10, HD, HDD, LOCK, BT, MP3, MUSIC, DD, MAIL, MUTE, PLAY, PAUSE, FF, FR, REC, CLOCK } tvfd_icon;
+#if defined(PLATFORM_CUBEREVO) || defined(PLATFORM_CUBEREVO_MINI) || defined(PLATFORM_CUBEREVO_MINI2) || defined(PLATFORM_CUBEREVO_MINI_FTA) || defined(PLATFORM_CUBEREVO_250HD) || defined(PLATFORM_CUBEREVO_2000HD) || defined(PLATFORM_CUBEREVO_9500HD)
+struct _symbol
+{
+	unsigned char onoff;
+	unsigned char grid;
+	unsigned char bit;
+};
+typedef struct _symbol symbol_t;
+
+struct _vfd_ani
+{
+	unsigned char kind_cnt;	// 7 : on/off, 6 ~ 0 : kind
+	unsigned char dir;
+	unsigned char width;
+	unsigned char delay;
+	
+};
+typedef struct _vfd_ani vfd_ani_t;
+
+enum vfd_led_type
+{
+	LED_SET,			/* LED엸 켜고 끈롕. */
+	LED_WAIT,			/* LED에 기롕옶라렺 표쑦엸 한롕. */
+	LED_WARN,			/* LED에 경고 표쑦엸 한롕. */
+};
+struct _vfd_led
+{
+	enum vfd_led_type type;
+	char val;
+};
+typedef struct _vfd_led vfd_led_t;
+
+#define TYPE_FRONT		'f'
+enum
+{
+	FRONT_SET_TIME,
+	FRONT_GET_TIME,
+	FRONT_SET_ALARM,
+	FRONT_CLR_ALARM,
+	FRONT_GET_ALARM,
+	FRONT_GET_ALARM_TIME,
+	FRONT_SET_SEG,
+	FRONT_SET_SEGS,
+	FRONT_GET_AWAKE_ALARM,
+	FRONT_GET_AWAKE_AC,
+	FRONT_SET_REBOOT,
+	FRONT_SET_POWEROFF,
+	FRONT_SET_WARMON,
+	FRONT_SET_WARMOFF,
+	FRONT_SET_WRITERAM,
+	FRONT_GET_READRAM,
+	FRONT_GET_MICOMVER,
+	FRONT_SET_VFDLED,
+	FRONT_SET_VFDTIME,
+	FRONT_SET_VFDBRIGHT,
+
+	FRONT_FAN_ON,
+	FRONT_FAN_OFF,
+	FRONT_RFMOD_ON,
+	FRONT_RFMOD_OFF,
+	FRONT_SET_SEGBUF,
+	FRONT_SET_SEGUPDATE,
+	FRONT_SET_SCROLLPT,
+	FRONT_SET_SEGSCROLL,
+	FRONT_SET_ANIMATION,
+	FRONT_SET_BUFCLR,
+
+	FRONT_SET_TIMEMODE,
+	FRONT_SET_SYMBOL,	/* supported from dotmatrix type front */
+	FRONT_SET_1GSYMBOL,
+	FRONT_SET_CLRSYMBOL,
+};
+#define FRONT_IOCS_SYMBOL	_IOW	(TYPE_FRONT,FRONT_SET_SYMBOL,		symbol_t)
+#define FRONT_IOCS_1GSYMBOL	_IOW	(TYPE_FRONT,FRONT_SET_1GSYMBOL,		symbol_t)
+#define FRONT_IOCC_CLRSYMBOL	_IO	(TYPE_FRONT,FRONT_SET_CLRSYMBOL		)
+#define FRONT_IOCS_ANIMATION	_IOW (TYPE_FRONT,FRONT_SET_ANIMATION,	vfd_ani_t)
+#define FRONT_IOCS_VFDLED		_IOW (TYPE_FRONT,FRONT_SET_VFDLED,		vfd_led_t)
+#define FRONT_IOCS_VFDBRIGHT	_IOW (TYPE_FRONT,FRONT_SET_VFDBRIGHT,	unsigned char)
+#define FRONT_IOCC_FAN_ON		_IO  (TYPE_FRONT,FRONT_FAN_ON			)
+#define FRONT_IOCC_FAN_OFF		_IO  (TYPE_FRONT,FRONT_FAN_OFF			)
+#define FRONT_IOCC_SEGUPDATE	_IO  (TYPE_FRONT,FRONT_SET_SEGUPDATE	)
+#define FRONT_IOCC_BUFCLR		_IO	 (TYPE_FRONT,FRONT_SET_BUFCLR		)
+#endif
+
+typedef enum { USB = 0x10, HD, LOCK, MP3, MUSIC, DD, MUTE, PLAY, PAUSE, REC,
+// Cuberevo personal icon
+TIMESHIFT, SAT, TER, _480i, _480p, _576i, _576p, _720p, _1080i, _1080p, 
+POWER, TV, FILE_, SCHEDULE, T1, T2, REPEAT, CLEAR } tvfd_icon;
 
 class evfd
 {
@@ -23,17 +110,18 @@
 	void init();
 	static evfd* getInstance();
 
+	void vfd_set_ani(int id);
 	void vfd_set_icon(tvfd_icon id, bool onoff);
 	void vfd_set_icon(tvfd_icon id, bool onoff, bool force);
 	void vfd_clear_icons();
 
 	void vfd_write_string(char * string);
 	void vfd_write_string(char * str, bool force);
-	void vfd_write_string_scrollText(char* text);
 	void vfd_clear_string();
 	
 	void vfd_set_brightness(unsigned char setting);
 	void vfd_set_light(bool onoff);
+	void vfd_set_fan(bool onoff);
 };


diff -urN enigma2-nightly/lib/python/Plugins/SystemPlugins/VFD-Icons/plugin.py enigma2-nightly/lib/python/Plugins/SystemPlugins/VFD-Icons/plugin.py
--- enigma2-nightly/lib/python/Plugins/SystemPlugins/VFD-Icons/plugin.py	1970-01-01 01:00:00.000000000 +0100
+++ enigma2-nightly/lib/python/Plugins/SystemPlugins/VFD-Icons/plugin.py	2009-12-13 10:15:06.057264790 +0100
@@ -69,9 +69,9 @@
 			info=service.info()
 			crypted = info and info.getInfo(iServiceInformation.sIsCrypted) or -1
 			if crypted == 1 : #set crypt symbol
-				evfd.getInstance().vfd_set_icon(0x13,1)
+				evfd.getInstance().vfd_set_icon(0x12,True)
 			else:
-				evfd.getInstance().vfd_set_icon(0x13,0)
+				evfd.getInstance().vfd_set_icon(0x12,False)
 	
 	def checkAudioTracks(self):
 		self.dolbyAvailable = False
@@ -92,16 +92,16 @@
 	def showDolby(self):
 		print "[showDolby]"
 		if self.dolbyAvailable:
-			evfd.getInstance().vfd_set_icon(0x17,1)
+			evfd.getInstance().vfd_set_icon(0x15,True)
 		else:
-			evfd.getInstance().vfd_set_icon(0x17,0)
+			evfd.getInstance().vfd_set_icon(0x15,False)
 		
 	def showMp3(self):
 		print "[showMp3]"
 		if self.mp3Available:
-			evfd.getInstance().vfd_set_icon(0x15,1)
+			evfd.getInstance().vfd_set_icon(0x13,True)
 		else:
-			evfd.getInstance().vfd_set_icon(0x15,0)
+			evfd.getInstance().vfd_set_icon(0x13,False)
 		
 	def __evUpdatedEventInfo(self):
 		print "[__evUpdatedEventInfo]"
@@ -118,9 +118,9 @@
 	def __evSeekableStatusChanged(self):
 		print "[__evSeekableStatusChanged]"
 		if self.getSeekState():
-			evfd.getInstance().vfd_set_icon(0x1A,1)
+			evfd.getInstance().vfd_set_icon(0x18,True)
 		else:
-			evfd.getInstance().vfd_set_icon(0x1A,0)
+			evfd.getInstance().vfd_set_icon(0x18,False)
 		
 	def __evVideoSizeChanged(self):
 		print "[__evVideoSizeChanged]"
@@ -128,18 +128,30 @@
 		if service is not None:
 			info=service.info()
 			height = info and info.getInfo(iServiceInformation.sVideoHeight) or -1
-			if height > 576 : #set HD symbol
-				evfd.getInstance().vfd_set_icon(0x11,1)
+			if height == 480:
+				evfd.getInstance().vfd_set_icon(0x1D,True)
 			else:
-				evfd.getInstance().vfd_set_icon(0x11,0)
+				evfd.getInstance().vfd_set_icon(0x1D,False)
+			if height == 576:
+				evfd.getInstance().vfd_set_icon(0x1F,True)
+			else:
+				evfd.getInstance().vfd_set_icon(0x1F,False)
+			if height == 720:
+				evfd.getInstance().vfd_set_icon(0x21,True)
+			else:
+				evfd.getInstance().vfd_set_icon(0x21,False)
+			if height == 1080:
+				evfd.getInstance().vfd_set_icon(0x22,True)
+			else:
+				evfd.getInstance().vfd_set_icon(0x22,False)
 		
 	def gotRecordEvent(self, service, event):
 		recs = self.session.nav.getRecordings()
 		nrecs = len(recs)
 		if nrecs > 0: #set rec symbol
-			evfd.getInstance().vfd_set_icon(0x1e,1)
+			evfd.getInstance().vfd_set_icon(0x19,True)
 		else:
-			evfd.getInstance().vfd_set_icon(0x1e,0)
+			evfd.getInstance().vfd_set_icon(0x19,False)
 		
 
 VFDIconsInstance = None
--- enigma2-nightly/lib/python/Components/Timezones.py	2010-05-03 13:45:15.006899015 +0200
+++ enigma2-nightly/lib/python/Components/Timezones.py	2010-05-02 14:06:52.000000000 +0200
@@ -50,6 +50,17 @@
 		except:
 			from enigma import e_tzset
 			e_tzset()
+#+++>
+		import os
+		try:
+			os.popen("/bin/cubefpctl --setgmtoffset")
+		except OSError:
+			print "no memory"
+		try:
+			os.popen("/bin/cubefpctl --syncfptime")
+		except OSError:
+			print "no memory"
+#+++<
 		
 	def getTimezoneList(self):
 		return [ str(x[0]) for x in self.timezones ]
--- enigma2-nightly/lib/python/Plugins/SystemPlugins/Tuxtxt/tuxtxt/tuxtxt.c	2010-05-03 13:45:15.266902593 +0200
+++ enigma2-nightly/lib/python/Plugins/SystemPlugins/Tuxtxt/tuxtxt/tuxtxt.c	2010-05-03 15:26:26.934973004 +0200
@@ -1541,7 +1541,7 @@
 
 void * rcThread(void *vData)
 {
-    int key;
+    int key, oldkey/*, count_time_elapsed*/;
     char stmp[9];
     struct sockaddr_un cliadr, srvadr;
     int srvfd;
@@ -1560,6 +1560,10 @@
     
     debugf (1, "Thread started !!!!!!!!!!!!!!!!!!!!!!!!!!!!\n");
 
+    oldkey = -2;
+    key = -1;
+    //count_time_elapsed = 0;
+
     while (rc == -1)
     {
     	rc = accept( srvfd, (struct sockaddr *) &cliadr, &len );
@@ -1579,44 +1583,56 @@
 	   sscanf(stmp, "%d", &key);
 	   debugf(1, "TT_KEY->: %d\n", key);
 
-           switch (key)
-               {
-                   case KEY_UP:        RCCode = RC_UP;        break;
-                   case KEY_DOWN:        RCCode = RC_DOWN;    break;
-                   case KEY_LEFT:        RCCode = RC_LEFT;    break;
-                   case KEY_RIGHT:        RCCode = RC_RIGHT;    break;
-                   case KEY_OK:        RCCode = RC_OK;        break;
-                   case KEY_0:        RCCode = RC_0;        break;
-                   case KEY_1:        RCCode = RC_1;        break;
-                   case KEY_2:        RCCode = RC_2;        break;
-                   case KEY_3:        RCCode = RC_3;        break;
-                   case KEY_4:        RCCode = RC_4;        break;
-                   case KEY_5:        RCCode = RC_5;        break;
-                   case KEY_6:        RCCode = RC_6;        break;
-                   case KEY_7:        RCCode = RC_7;        break;
-                   case KEY_8:        RCCode = RC_8;        break;
-                   case KEY_9:        RCCode = RC_9;        break;
-                   case KEY_RED:        RCCode = RC_RED;    break;
-                   case KEY_GREEN:        RCCode = RC_GREEN;    break;
-                   case KEY_YELLOW:    RCCode = RC_YELLOW;    break;
-                   case KEY_BLUE:        RCCode = RC_BLUE;    break;
-                   case KEY_VOLUMEUP:    RCCode = RC_PLUS;    break;
-                   case KEY_VOLUMEDOWN:    RCCode = RC_MINUS;    break;
-                   case KEY_MUTE:        RCCode = RC_MUTE;    break;
-                   case KEY_HELP:        RCCode = RC_HELP;    break;
-                   case KEY_MENU:        RCCode = RC_DBOX;    break;
-                   case KEY_HOME:        RCCode = RC_HOME;    break;
-                   case KEY_POWER:        RCCode = RC_STANDBY;    break;
-                   default:  RCCode = -1;
-	       }
-       
-       	   if (RCCode != -1)
-	   	newKey = 1;
-	   else
-	        newKey = 0;
-       }
-       
-       usleep(100000);       	
+	   if((oldkey!=key) /*|| (count_time_elapsed>0)*/)
+	   {
+		   switch (key)
+		       {
+		           case KEY_UP:        RCCode = RC_UP;        break;
+		           case KEY_DOWN:      RCCode = RC_DOWN;    break;
+		           case KEY_LEFT:      RCCode = RC_LEFT;    break;
+		           case KEY_RIGHT:     RCCode = RC_RIGHT;    break;
+		           case KEY_OK:        RCCode = RC_OK;        break;
+		           case KEY_0:         RCCode = RC_0;        break;
+		           case KEY_1:         RCCode = RC_1;        break;
+		           case KEY_2:         RCCode = RC_2;        break;
+		           case KEY_3:         RCCode = RC_3;        break;
+		           case KEY_4:         RCCode = RC_4;        break;
+		           case KEY_5:         RCCode = RC_5;        break;
+		           case KEY_6:         RCCode = RC_6;        break;
+		           case KEY_7:         RCCode = RC_7;        break;
+		           case KEY_8:         RCCode = RC_8;        break;
+		           case KEY_9:         RCCode = RC_9;        break;
+		           case KEY_RED:       RCCode = RC_RED;    break;
+		           case KEY_GREEN:     RCCode = RC_GREEN;    break;
+		           case KEY_YELLOW:    RCCode = RC_YELLOW;    break;
+		           case KEY_BLUE:      RCCode = RC_BLUE;    break;
+		           case KEY_FASTFORWARD:
+	           case KEY_PAGEUP:    RCCode = RC_PLUS;    break;
+		           case KEY_REWIND:
+		           case KEY_PAGEDOWN:  RCCode = RC_MINUS;    break;
+		           case KEY_MUTE:      RCCode = RC_MUTE;    break;
+		           case KEY_HELP:      RCCode = RC_HELP;    break;
+		           case KEY_MENU:        RCCode = RC_DBOX;    break;
+		           case KEY_HOME:
+			   case KEY_TEXT:
+			   case 158/*KEY_EXIT*/:
+		           case KEY_POWER:        RCCode = RC_HOME;    break;
+		           default:  RCCode = -1;
+		       }
+	       
+	       	   if (RCCode != -1) {
+               		fprintf(stderr, "TT_KEY->: %d\n", key);
+               		newKey = 1;
+	   	   } else
+	        	newKey = 0;
+		   //count_time_elapsed = 0;
+		   oldkey = key;
+           }
+	   else oldkey = -1;
+        }
+ 
+       usleep(100000);
+       //count_time_elapsed++;
 
     }
     
@@ -6733,7 +6749,7 @@
        while(1)
        {
            /* block until rcThread says there is a new key */
-	   usleep(1000000/100);
+	   usleep(350*1000/1000);
 	   if (newKey == 1)
 	   {   
 	        newKey = 0;
@@ -6746,7 +6762,8 @@
         int tmp = newKey;
 	newKey = 0;
 	
-	usleep(1000000/100);
+	//350*HZ*10/1000
+	usleep(350*1000/1000);
 	
 	return tmp;
     }   

--- enigma2-nightly.org/lib/python/Screens/ChannelSelection.py	2010-05-03 18:46:44.000000000 +0100
+++ enigma2-nightly/lib/python/Screens/ChannelSelection.py	2010-05-03 10:45:07.000000000 +0100
@@ -1130,11 +1130,11 @@
 
 	def asciiOn(self):
 		rcinput = eRCInput.getInstance()
-		rcinput.setKeyboardMode(rcinput.kmAscii)
+		#rcinput.setKeyboardMode(rcinput.kmAscii)
 
 	def asciiOff(self):
 		rcinput = eRCInput.getInstance()
-		rcinput.setKeyboardMode(rcinput.kmNone)
+		#rcinput.setKeyboardMode(rcinput.kmNone)
 
 	def multibouquet_config_changed(self, val):
 		self.recallBouquetMode()

--- enigma2-nightly.org/lib/python/Components/ConfigList.py	2010-05-03 18:46:44.000000000 +0100
+++ enigma2-nightly/lib/python/Components/ConfigList.py	2010-05-03 10:33:56.000000000 +0100
@@ -18,12 +18,12 @@
 
 	def execBegin(self):
 		rcinput = eRCInput.getInstance()
-		rcinput.setKeyboardMode(rcinput.kmAscii)
+		#rcinput.setKeyboardMode(rcinput.kmAscii)
 		self.timer.callback.append(self.timeout)
 
 	def execEnd(self):
 		rcinput = eRCInput.getInstance()
-		rcinput.setKeyboardMode(rcinput.kmNone)
+		#rcinput.setKeyboardMode(rcinput.kmNone)
 		self.timer.callback.remove(self.timeout)
 
 	def toggle(self):

--- enigma2-nightly.org/lib/python/Plugins/SystemPlugins/CommonInterfaceAssignment/plugin.py	2010-05-03 10:52:45.000000000 +0100
+++ enigma2-nightly/lib/python/Plugins/SystemPlugins/CommonInterfaceAssignment/plugin.py	2010-05-03 10:38:57.000000000 +0100
@@ -27,8 +27,10 @@
 		</screen>"""
 
 	def __init__(self, session, args = 0):
-
 		Screen.__init__(self, session)
+#+++>
+		appname = ""
+#+++<
 
 		self["key_red"] = StaticText(_("Close"))
 		self["key_green"] = StaticText(_("Edit"))
@@ -51,7 +53,12 @@
 		if NUM_CI > 0:
 			for slot in range(NUM_CI):
 				state = eDVBCI_UI.getInstance().getState(slot)
-				if state == 0:
+#--->
+#-				if state == 0:
+#---<
+#+++>
+				if state == 0 or state == -1:
+#+++<
 					appname = _("Slot %d") %(slot+1) + " - " + _("no module found")
 				elif state == 1:	
 					appname = _("Slot %d") %(slot+1) + " - " + _("init modules")


