--- a/libavformat/mpegts.c
+++ b/libavformat/mpegts.c
@@ -1365,11 +1365,24 @@ int ff_parse_mpeg2_descriptor(AVFormatContext *fc, AVStream *st, int stream_type
         }
         break;
     case 0x56: /* DVB teletext descriptor */
-        language[0] = get8(pp, desc_end);
-        language[1] = get8(pp, desc_end);
-        language[2] = get8(pp, desc_end);
-        language[3] = 0;
-        av_dict_set(&st->metadata, "language", language, 0);
+	{
+		int ttx_count = desc_len / 5;
+		for (i = 0; i < ttx_count; i++) {
+			char tmp_key[20];
+			char tmp_val[40];
+			language[0] = get8(pp, desc_end);
+			language[1] = get8(pp, desc_end);
+			language[2] = get8(pp, desc_end);
+			language[3] = 0;
+			av_dict_set(&st->metadata, "language", language, 0);
+			AVDictionaryEntry *t = NULL;
+			unsigned char b3 = get8(pp, desc_end);
+			unsigned char b4 = get8(pp, desc_end);
+			snprintf(tmp_key, sizeof(tmp_key), "teletext_%d", i);
+			snprintf(tmp_val, sizeof(tmp_val), "%d %s %u %u %u", pid, language, b3 >> 3, b3 & 0x7, b4); // pid lang type mag page
+			av_dict_set(&st->metadata, tmp_key, tmp_val, 0);
+		}
+	}
         break;
     case 0x59: /* subtitling descriptor */
         language[0] = get8(pp, desc_end);
--- a/cmdutils.c-orig	2013-02-25 20:26:44.974681651 +0100
+++ b/cmdutils.c	2013-02-25 20:27:21.414681657 +0100
@@ -992,8 +992,7 @@
         av_log(NULL, level, " Copyright (c) %d-%d the FFmpeg developers",
                program_birth_year, this_year);
     av_log(NULL, level, "\n");
-    av_log(NULL, level, "%sbuilt on %s %s with %s\n",
-           indent, __DATE__, __TIME__, CC_IDENT);
+    av_log(NULL, level, "%sbuilt with %s\n", indent, CC_IDENT);
 
     av_log(NULL, level, "%sconfiguration: " FFMPEG_CONFIGURATION "\n", indent);
 }
--- a/ffprobe.c-orig	2013-02-25 20:27:01.214681641 +0100
+++ b/ffprobe.c	2013-02-25 20:27:21.414681657 +0100
@@ -1925,8 +1925,6 @@
     print_str("version", FFMPEG_VERSION);
     print_fmt("copyright", "Copyright (c) %d-%d the FFmpeg developers",
               program_birth_year, this_year);
-    print_str("build_date", __DATE__);
-    print_str("build_time", __TIME__);
     print_str("compiler_ident", CC_IDENT);
     print_str("configuration", FFMPEG_CONFIGURATION);
     writer_print_section_footer(w);
