--- freetype-2.1.3/include/freetype/config/ftoption.h.orig	Mon Feb 24 11:52:24 2003
+++ freetype-2.1.3/include/freetype/config/ftoption.h	Mon Feb 24 11:52:53 2003
@@ -378,7 +378,7 @@
   /*   do not #undef this macro here, since the build system might         */
   /*   define for certain configurations                                   */
   /*                                                                       */
-/* #define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
