diff -upNr -X ../notdiff.txt linux-2.6.17.14_stm22_0041.3/drivers/i2c/algos/i2c-algo-bit.c linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/algos/i2c-algo-bit.c
--- linux-2.6.17.14_stm22_0041.3/drivers/i2c/algos/i2c-algo-bit.c	2008-07-14 09:30:13.000000000 +0200
+++ linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/algos/i2c-algo-bit.c	2008-07-14 09:44:30.000000000 +0200
@@ -28,8 +28,10 @@
 #include <linux/init.h>
 #include <linux/errno.h>
 #include <linux/sched.h>
+#include <linux/mutex.h>
 #include <linux/i2c.h>
 #include <linux/i2c-algo-bit.h>
+#include "../busses/i2c-stm.h"
 
 
 /* ----- global defines ----------------------------------------------- */
@@ -39,6 +41,26 @@
 #define DEBPROTO(x) if (i2c_debug>=9) { x; }
  	/* debug the protocol by showing transferred bits */
 
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+#define NO_STOP_ENABLED                        0x1
+#define NO_STOP_ENABLED_MASK           0x1
+
+#define LAST_I2C_WAS_NO_STOP           0x2
+#define LAST_I2C_WAS_NO_STOP_MASK      0x2
+
+#define check_master_bus(adap)         ((adap)->buslock_task == get_current())
+#define check_nostop(adap)             (((adap)->config & \
+                                                NO_STOP_ENABLED_MASK ) ? 1 : 0 )
+
+#define set_nostop(adap)               ((adap)->config |=  NO_STOP_ENABLED)
+#define clear_nostop(adap)                     ((adap)->config &= ~NO_STOP_ENABLED_MASK)
+
+#define check_lasti2cwas_nostop(adap) (((adap)->config & \
+                                               LAST_I2C_WAS_NO_STOP_MASK) ? 1 : 0 )
+
+#define set_lasti2cwas_nostop(adap)     ((adap)->config |=  LAST_I2C_WAS_NO_STOP)
+#define clear_lasti2cwas_nostop(adap)   ((adap)->config &= ~LAST_I2C_WAS_NO_STOP_MASK)
+#endif
 
 /* ----- global variables ---------------------------------------------	*/
 
@@ -128,8 +150,23 @@ static void i2c_repstart(struct i2c_algo
 }
 
 
-static void i2c_stop(struct i2c_algo_bit_data *adap) 
+static void i2c_stop(struct i2c_algo_bit_data *adap, int error)
 {
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+/*
+ * To use the NoStop condition we need:
+ * 1. a registered buslock_task
+ * 2. a no stop operation (read or write )
+ * 3. no error on transaction
+ */
+       if (adap->buslock_task &&
+           check_nostop(adap) &&
+           error>0) {
+               set_lasti2cwas_nostop(adap);
+               return ;
+       }
+       clear_lasti2cwas_nostop(adap);
+#endif
 	DEBPROTO(printk("P\n"));
 	/* assert: scl is low */
 	sdalo(adap);
@@ -315,7 +352,7 @@ static inline int try_address(struct i2c
 		ret = i2c_outb(i2c_adap,addr);
 		if (ret==1)
 			break;	/* success! */
-		i2c_stop(adap);
+		i2c_stop(adap,-1);
 		udelay(5/*adap->udelay*/);
 		if (i==retries)  /* no success */
 			break;
@@ -350,7 +387,7 @@ static int sendbytes(struct i2c_adapter 
 			wrcount++;
 		} else { /* arbitration or no acknowledge */
 			dev_err(&i2c_adap->dev, "sendbytes: error - bailout.\n");
-			i2c_stop(adap);
+			i2c_stop(adap,-1);
 			return (retval<0)? retval : -EFAULT;
 			        /* got a better one ?? */
 		}
@@ -474,7 +511,24 @@ static int bit_xfer(struct i2c_adapter *
 	int i,ret;
 	unsigned short nak_ok;
 
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+       if (check_master_bus(adap)) {
+/* master process... */
+               if (check_lasti2cwas_nostop(adap)) {
+                       clear_lasti2cwas_nostop(adap);
+                       i2c_repstart(adap);
+                       }
+               else
+                       i2c_start(adap);
+       }
+       else {
+/* normal process...  */
+               mutex_lock(&i2c_adap->bus_lock);
+               i2c_start(adap);
+       }
+#else
 	i2c_start(adap);
+#endif
 	for (i=0;i<num;i++) {
 		pmsg = &msgs[i];
 		nak_ok = pmsg->flags & I2C_M_IGNORE_NAK; 
@@ -486,6 +540,10 @@ static int bit_xfer(struct i2c_adapter *
 			if ((ret != 0) && !nak_ok) {
 			    DEB2(printk(KERN_DEBUG "i2c-algo-bit.o: NAK from device addr %2.2x msg #%d\n"
 					,msgs[i].addr,i));
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+                           if (!check_master_bus(adap) )
+                               mutex_unlock(&i2c_adap->bus_lock);
+#endif
 			    return (ret<0) ? ret : -EREMOTEIO;
 			}
 		}
@@ -494,6 +552,10 @@ static int bit_xfer(struct i2c_adapter *
 			ret = readbytes(i2c_adap, pmsg);
 			DEB2(printk(KERN_DEBUG "i2c-algo-bit.o: read %d bytes.\n",ret));
 			if (ret < pmsg->len ) {
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+                               if (!check_master_bus(adap))
+                                       mutex_unlock(&i2c_adap->bus_lock);
+#endif
 				return (ret<0)? ret : -EREMOTEIO;
 			}
 		} else {
@@ -501,11 +563,20 @@ static int bit_xfer(struct i2c_adapter *
 			ret = sendbytes(i2c_adap, pmsg);
 			DEB2(printk(KERN_DEBUG "i2c-algo-bit.o: wrote %d bytes.\n",ret));
 			if (ret < pmsg->len ) {
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+                               if (!check_master_bus(adap))
+                                       mutex_unlock(&i2c_adap->bus_lock);
+#endif
 				return (ret<0) ? ret : -EREMOTEIO;
 			}
 		}
 	}
-	i2c_stop(adap);
+
+       i2c_stop(adap,1);
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+       if (!check_master_bus(adap))
+               mutex_unlock(&i2c_adap->bus_lock);
+#endif
 	return num;
 }
 
@@ -517,10 +588,74 @@ static u32 bit_func(struct i2c_adapter *
 
 
 /* -----exported algorithm data: -------------------------------------	*/
+#if defined(CONFIG_I2C_STM_NOSTOP_API) && !defined(CONFIG_I2C_STM)
+int iic_nostop_operation(struct i2c_adapter *adapter, int enable)
+{
+       struct i2c_algo_bit_data *bit_adap = adapter->algo_data;
+       if (bit_adap->buslock_task != get_current() )
+               return -1;
+       if (enable)
+               set_nostop(bit_adap);
+       else
+               clear_nostop(bit_adap);
+       return 0;
+}
+
+int iic_register_buslock_task(struct task_struct* this,struct i2c_adapter *adapter)
+{
+       struct i2c_algo_bit_data *bit_adap = adapter->algo_data;
+       if (bit_adap->buslock_task)
+               return -1; /* already locked */
+       mutex_lock(&adapter->bus_lock);
+       bit_adap->buslock_task = this;
+       clear_nostop(bit_adap);
+       clear_lasti2cwas_nostop(bit_adap);
+/*
+ * At this point the iic_adap->buslock_task has the bus!
+ */
+       return 0;
+}
+
+int iic_unregister_buslock_task(struct task_struct* this,struct i2c_adapter *adapter)
+{
+       struct i2c_algo_bit_data *bit_adap = adapter->algo_data;
+       if (bit_adap->buslock_task != this)
+               return -1; /* it isn\ufffdt the right process */
+       bit_adap->buslock_task = NULL;
+       clear_nostop(bit_adap);
+       clear_lasti2cwas_nostop(bit_adap);
+       mutex_unlock(&adapter->bus_lock);
+       return 0;
+}
+
+static int bit_control(struct i2c_adapter *adapter,
+                                unsigned int cmd, unsigned long arg)
+{
+       struct i2c_algo_bit_data *bit_adap = adapter->algo_data;
+       switch (cmd) {
+       case I2C_STM_IOCTL_MASTER_BUS:
+               if (bit_adap->buslock_task)
+                       return iic_unregister_buslock_task(get_current(),adapter);
+               return  iic_register_buslock_task(get_current(),adapter);
+               break;
+       case I2C_STM_IOCTL_NOSTOP:
+               return iic_nostop_operation(adapter, arg);
+               break;
+       case I2C_STM_IOCTL_FAST:
+               printk(KERN_INFO "INFO: The i2c fast mode isn't supported on soft i2c\n");
+               break;
+       }
+
+       return 0;
+}
+#endif
 
 static struct i2c_algorithm i2c_bit_algo = {
 	.master_xfer	= bit_xfer,
 	.functionality	= bit_func,
+#if defined(CONFIG_I2C_STM_NOSTOP_API) && !defined(CONFIG_I2C_STM)
+       .algo_control = bit_control,
+#endif
 };
 
 /* 
diff -upNr -X ../notdiff.txt linux-2.6.17.14_stm22_0041.3/drivers/i2c/busses/i2c-stm.c linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/busses/i2c-stm.c
--- linux-2.6.17.14_stm22_0041.3/drivers/i2c/busses/i2c-stm.c	2008-07-14 09:31:50.000000000 +0200
+++ linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/busses/i2c-stm.c	2008-07-14 09:44:30.000000000 +0200
@@ -118,6 +118,13 @@
 
 #define IIC_STM_CONFIG_BAUDRATE_MASK       0xffff0000
 
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+#define NO_STOP_ENABLED_MASK              0x2
+#define NO_STOP_ENABLED                   0x2
+#define LAST_I2C_WAS_NO_STOP_MASK         0x4
+#define LAST_I2C_WAS_NO_STOP              0x4
+#endif
+
 typedef enum _iic_state_machine_e {
 	IIC_FSM_VOID = 0,
 	IIC_FSM_PREPARE,
@@ -144,6 +151,9 @@ struct iic_ssc {
 	struct i2c_adapter adapter;
 	unsigned long config;
 	struct list_head list;
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+       struct task_struct *buslock_task;       /* which task asked for NoStop mode */
+#endif
 };
 
 /*
@@ -183,6 +193,35 @@ struct iic_transaction {
 #define check_fastmode(adap)	(((adap)->config & \
                                  IIC_STM_CONFIG_SPEED_MASK ) ? 1 : 0 )
 
+/*
+ * Defines to support the NoStop API
+ */
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+#define check_master_bus(adap) ((adap)->buslock_task == get_current())
+#define check_nostop(adap)     (((adap)->config & \
+                                NO_STOP_ENABLED_MASK ) ? 1 : 0 )
+
+#define set_nostop(adap)       ((adap)->config |=  NO_STOP_ENABLED)
+#define clear_nostop(adap)     ((adap)->config &= ~NO_STOP_ENABLED_MASK)
+
+#define check_lasti2cwas_nostop(adap) (((adap)->config & \
+                                LAST_I2C_WAS_NO_STOP_MASK) ? 1 : 0 )
+
+#define set_lasti2cwas_nostop(adap)    ((adap)->config |=  LAST_I2C_WAS_NO_STOP)
+#define clear_lasti2cwas_nostop(adap)  ((adap)->config &= ~LAST_I2C_WAS_NO_STOP_MASK)
+#else
+/*
+ * With this macro the compiler will be able
+ * to remove the 'dead' code
+ */
+#define check_master_bus(adap)                 (1==0)
+#define check_nostop(adap)                     (1==0)
+#define set_nostop(adap)                       {}
+#define clear_nostop(adap)                     {}
+#define check_lasti2cwas_nostop(adap)          (1==0)
+#define set_lasti2cwas_nostop(adap)            {}
+#define clear_lasti2cwas_nostop(adap)          {}
+#endif
 
 static void iic_stm_setup_timing(struct iic_ssc *adap);
 
@@ -231,6 +270,10 @@ static void iic_state_machine(struct iic
 	barrier();
 	switch (trsc->state) {
 	case IIC_FSM_PREPARE:
+               if (check_lasti2cwas_nostop(adap)) {
+                       clear_lasti2cwas_nostop(adap);
+                       jump_on_fsm_repstart(trsc);
+                       }
 		dgb_print2("-Prepare\n");
 		/*
 		 * Here we set the right Pio configuration
@@ -450,6 +493,22 @@ static void iic_state_machine(struct iic
 		    (++trsc->current_msg < trsc->queue_length)) {
 			jump_on_fsm_repstart(trsc);
 		}
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+/*
+ * To use the NoStop condition we need:
+ * 1. a registered buslock_task
+ * 2. an enabled no_stop operation (read or write)
+ * 3. no error on the transaction!!!
+ */
+               if (adap->buslock_task &&
+                   check_nostop(adap) &&
+                   !(status & SSC_STA_NACK) ) {
+                       set_lasti2cwas_nostop(adap);
+                       dgb_print2(" Stop not done\n");
+                       jump_on_fsm_complete(trsc);
+               }
+               clear_lasti2cwas_nostop(adap);
+#endif
 		dgb_print2(" Stop\n");
 		ssc_store32(ssc_bus, SSC_CLR, 0xdc0);
 		trsc->next_state = IIC_FSM_COMPLETE;
@@ -460,7 +519,7 @@ static void iic_state_machine(struct iic
 		break;
 
 	case IIC_FSM_COMPLETE:
-		/* be_fsm_complete: */
+		 be_fsm_complete:
 		dgb_print2(" Complete\n");
 		ssc_store32(ssc_bus, SSC_IEN, 0x0);
 		if (!(trsc->status_error & IIC_E_NOTACK))
@@ -586,11 +645,32 @@ static int iic_stm_xfer(struct i2c_adapt
 		};
 
 	dgb_print("\n");
-	ssc_request_bus(ssc_bus, (void*)iic_state_machine, &transaction);
 
+       if (check_master_bus(adap)) {
+       /*
+        * The ssc_bus->irq_private_data should not be modified
+        * but when the master bus requested the ssc_bus it had
+        * no 'irq_private_data' (because it had no transaction)
+        * now it has to update the 'irq_private_data' with the
+        * right transaction address.
+        */
+               ssc_bus->irq_private_data =(void*)&transaction;
+               if (!check_lasti2cwas_nostop(adap)){
+iic_xfer_retry_0:
+                       if(!iic_wait_free_bus(ssc_bus))
+                               iic_pio_stop(ssc_bus);
+                       }
+       } else {
+               ssc_request_bus(ssc_bus, iic_state_machine, &transaction);
+iic_xfer_retry:
+               if(!iic_wait_free_bus(ssc_bus))
+                       iic_pio_stop(ssc_bus);
+       }
+#if 0
 iic_xfer_retry:
 	if(!iic_wait_free_bus(ssc_bus))
 		iic_pio_stop(ssc_bus);
+#endif
 
 	iic_state_machine(&transaction);
 
@@ -613,7 +693,10 @@ iic_xfer_retry:
                                 printk(KERN_ERR "i2c-stm: error on address.  auto retry %d\n", transaction.attempt);
                                 transaction.status_error = 0;
                                 local_irq_restore(flag);
-                                goto iic_xfer_retry;
+				if(check_master_bus(adap))
+                                       goto iic_xfer_retry_0;
+                                else
+                                       goto iic_xfer_retry;
                 } else if (transaction.status_error == IIC_E_ARBL){
                         /* Arbitration error */
                         printk(KERN_ERR "i2c-stm: arbitration error\n");
@@ -659,7 +742,8 @@ iic_xfer_retry:
 	} else
 		local_irq_restore(flag);
 
-	ssc_release_bus(ssc_bus);
+	if (!check_master_bus(adap))
+               ssc_release_bus(ssc_bus);
 
 	return result;
 }
@@ -776,12 +860,75 @@ static void iic_stm_setup_timing(struct 
 	return;
 }
 
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+int iic_nostop_operation(struct i2c_adapter *adapter, int enable)
+{
+   struct iic_ssc *iic_adap =
+       container_of(adapter, struct iic_ssc, adapter);
+
+   if (iic_adap->buslock_task != get_current())
+       return -1;
+   if (enable)
+       set_nostop(iic_adap);
+   else
+       clear_nostop(iic_adap);
+
+   return 0;
+}
+
+int iic_register_buslock_task(struct task_struct* this,struct i2c_adapter *adapter)
+{
+       struct iic_ssc *iic_adap =
+               container_of(adapter, struct iic_ssc, adapter);
+       struct ssc_t *ssc_bus =
+                       container_of(adapter->dev.parent, struct ssc_t, dev);
+
+        if (iic_adap->buslock_task)
+                return -1; /* already locked */
+        iic_adap->buslock_task = this;
+       ssc_request_bus(ssc_bus, iic_state_machine, iic_adap);
+       clear_lasti2cwas_nostop(iic_adap);
+/*
+ * At this point the iic_adap->buslock_task has the bus!
+ */
+       return 0;
+}
+
+int iic_unregister_buslock_task(struct task_struct* this,struct i2c_adapter *adapter)
+{
+       struct iic_ssc *iic_adap =
+               container_of(adapter, struct iic_ssc, adapter);
+       struct ssc_t *ssc_bus =
+                       container_of(adapter->dev.parent, struct ssc_t, dev);
+        if ( iic_adap->buslock_task != this)
+                return -1; /* it isn√<82>¬¥t the right process */
+        iic_adap->buslock_task = NULL;
+       clear_nostop(iic_adap);
+       clear_lasti2cwas_nostop(iic_adap);
+       ssc_release_bus(ssc_bus);
+       return 0;
+}
+#endif
+
 static int iic_stm_control(struct i2c_adapter *adapter,
 				unsigned int cmd, unsigned long arg)
 {
 	struct iic_ssc *iic_adap =
 	    container_of(adapter, struct iic_ssc, adapter);
 	switch (cmd) {
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+       case I2C_STM_IOCTL_MASTER_BUS:
+               dgb_print("Master bus\n");
+               if (iic_adap->buslock_task)
+                       return iic_unregister_buslock_task(get_current(),adapter);
+               return iic_register_buslock_task(get_current(),adapter);
+               break;
+       case I2C_STM_IOCTL_NOSTOP:
+               dgb_print("nostop operation\n");
+               return iic_nostop_operation(adapter, arg);
+               break;
+#endif
+
 	case I2C_STM_IOCTL_FAST:
 		dgb_print("ioctl fast 0x%x\n",arg);
 		iic_adap->config &= ~IIC_STM_CONFIG_SPEED_MASK;
diff -upNr -X ../notdiff.txt linux-2.6.17.14_stm22_0041.3/drivers/i2c/busses/i2c-stm.h linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/busses/i2c-stm.h
--- linux-2.6.17.14_stm22_0041.3/drivers/i2c/busses/i2c-stm.h	2008-07-14 09:30:13.000000000 +0200
+++ linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/busses/i2c-stm.h	2008-07-14 09:44:30.000000000 +0200
@@ -2,5 +2,19 @@
 #define I2C_STM
 
 #define I2C_STM_IOCTL_FAST           0xfa57
+#define I2C_STM_IOCTL_MASTER_BUS     0xfa20
+#define I2C_STM_IOCTL_NOSTOP        0xfa21
+
+#ifdef __KERNEL__
+/*
+ *  The kernel function interface to manage
+ *  the NoStop operations for
+ *  - the bit-banging algorithm        (I2C_ST40_PIO)
+ *  - the ssc hardware         (I2C_STM)
+ */
+int iic_nostop_operation(struct i2c_adapter *adapter, int enable);
+int iic_register_buslock_task(struct task_struct* this,struct i2c_adapter *adapter);
+int iic_unregister_buslock_task(struct task_struct* this,struct i2c_adapter *adapter);
+#endif
 
 #endif
diff -upNr -X ../notdiff.txt linux-2.6.17.14_stm22_0041.3/drivers/i2c/busses/Kconfig linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/busses/Kconfig
--- linux-2.6.17.14_stm22_0041.3/drivers/i2c/busses/Kconfig	2008-07-14 09:30:13.000000000 +0200
+++ linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/busses/Kconfig	2008-07-14 09:44:30.000000000 +0200
@@ -585,4 +585,11 @@ config HW_GLITCH_WIDTH
 	int "Glitch suppression width in uS"
 	default "1"
 
+config  I2C_STM_NOSTOP_API
+	bool "I2C add the NoStop operation API"
+	depends on I2C_ST40_PIO || I2C_STM
+	default n
+	---help---
+	Enable this option to use the NoStop operation API
+
 endmenu
diff -upNr -X ../notdiff.txt linux-2.6.17.14_stm22_0041.3/drivers/i2c/i2c-core.c linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/i2c-core.c
--- linux-2.6.17.14_stm22_0041.3/drivers/i2c/i2c-core.c	2008-07-14 09:30:13.000000000 +0200
+++ linux-2.6.17.14_stm22_0041.3_noStop/drivers/i2c/i2c-core.c	2008-07-14 09:44:30.000000000 +0200
@@ -569,9 +569,13 @@ int i2c_transfer(struct i2c_adapter * ad
 		}
 #endif
 
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+               ret = adap->algo->master_xfer(adap,msgs,num);
+#else
 		mutex_lock(&adap->bus_lock);
 		ret = adap->algo->master_xfer(adap,msgs,num);
 		mutex_unlock(&adap->bus_lock);
+#endif
 
 		return ret;
 	} else {
diff -upNr -X ../notdiff.txt linux-2.6.17.14_stm22_0041.3/include/linux/i2c-algo-bit.h linux-2.6.17.14_stm22_0041.3_noStop/include/linux/i2c-algo-bit.h
--- linux-2.6.17.14_stm22_0041.3/include/linux/i2c-algo-bit.h	2008-07-14 09:30:08.000000000 +0200
+++ linux-2.6.17.14_stm22_0041.3_noStop/include/linux/i2c-algo-bit.h	2008-07-14 09:44:30.000000000 +0200
@@ -42,6 +42,10 @@ struct i2c_algo_bit_data {
 				/* i.e. clock is (500 / udelay) KHz */
 	int mdelay;		/* in millisecs, unused */
 	int timeout;		/* in jiffies */
+#if defined(CONFIG_I2C_STM_NOSTOP_API)
+       int config;
+       struct task_struct *buslock_task;
+#endif
 };
 
 int i2c_bit_add_bus(struct i2c_adapter *);

