--- libvpx-v1.2.0/build/make/Makefile	2013-11-09 23:11:11.000000000 +0100
+++ libvpx-v1.2.0/build/make/Makefile.patched	2013-11-09 23:09:39.000000000 +0100
@@ -250,8 +250,8 @@
 #
 # This needs further abstraction for dealing with non-GNU linkers.
 $(1):
-	$(if $(quiet),@echo "    [LD] $$@")
-	$(qexec)$$(LD) -shared $$(LDFLAGS) \
+	$(if $(quiet),@echo "    [CC] $$@")
+	$(qexec)$$(CC) -shared $$(LDFLAGS) \
             -Wl,--no-undefined -Wl,-soname,$$(SONAME) \
             -Wl,--version-script,$$(SO_VERSION_SCRIPT) -o $$@ \
             $$(filter %.o,$$?) $$(extralibs)

--- libvpx-v1.2.0/configure	2013-11-09 22:25:01.000000000 +0100
+++ libvpx-v1.2.0/configure.patched	2013-11-09 22:24:38.000000000 +0100
@@ -136,6 +136,7 @@
 all_platforms="${all_platforms} universal-darwin11-gcc"
 all_platforms="${all_platforms} universal-darwin12-gcc"
 all_platforms="${all_platforms} generic-gnu"
+all_platforms="${all_platforms} sh4-linux"
 
 # all_targets is a list of all targets that can be configured
 # note that these should be in dependency order for now.
@@ -203,6 +204,7 @@
     x86_64
     ppc32
     ppc64
+    sh4
 "
 ARCH_EXT_LIST="
     edsp
