*** a/arch/sh/kernel/cpu/sh4/setup-stb7100.c	2011-01-03 11:13:35.000000000 +0100
--- b/setup-stb7100.c	2011-01-03 16:28:45.578307315 +0100
***************
*** 514,520 ****
  	mac_speed_sc = sysconf_claim(SYS_CFG, 7, 20, 20, "stmmac");
  
  	/* Remove the PHY clk */
! 	stpio_request_pin(3, 7, "stmmac EXTCLK", STPIO_ALT_OUT);
  
  	/* Configure the ethernet MAC PBL depending on the cut of the chip */
  	if (chip_revision == 1){
--- 514,520 ----
  	mac_speed_sc = sysconf_claim(SYS_CFG, 7, 20, 20, "stmmac");
  
  	/* Remove the PHY clk */
! //	stpio_request_pin(3, 7, "stmmac EXTCLK", STPIO_ALT_OUT);
  
  	/* Configure the ethernet MAC PBL depending on the cut of the chip */
  	if (chip_revision == 1){
***************
*** 677,685 ****
  		STPIO_ALT_OUT, STPIO_IN, STPIO_IN, STPIO_ALT_OUT), /* oe pin: 6 */
  	STASC_DEVICE(0x18031000, 122, -1, -1, 1, 0, 1, 4, 5,
  		STPIO_ALT_OUT, STPIO_IN, STPIO_IN, STPIO_ALT_OUT), /* oe pin: 6 */
! 	STASC_DEVICE(0x18032000, 121, -1, -1, 4, 3, 2, 4, 5,
  		STPIO_ALT_OUT, STPIO_IN, STPIO_IN, STPIO_ALT_OUT),
! 	STASC_DEVICE(0x18033000, 120, -1, -1, 5, 0, 1, 2, 3,
  		STPIO_ALT_OUT, STPIO_IN, STPIO_IN, STPIO_ALT_OUT),
  };
  
--- 677,685 ----
  		STPIO_ALT_OUT, STPIO_IN, STPIO_IN, STPIO_ALT_OUT), /* oe pin: 6 */
  	STASC_DEVICE(0x18031000, 122, -1, -1, 1, 0, 1, 4, 5,
  		STPIO_ALT_OUT, STPIO_IN, STPIO_IN, STPIO_ALT_OUT), /* oe pin: 6 */
! 	STASC_DEVICE(0x18032000, 121, -1, -1, 4, 3, 2, -1, -1,
  		STPIO_ALT_OUT, STPIO_IN, STPIO_IN, STPIO_ALT_OUT),
! 	STASC_DEVICE(0x18033000, 120, -1, -1, 5, 0, 1, -1, -1,
  		STPIO_ALT_OUT, STPIO_IN, STPIO_IN, STPIO_ALT_OUT),
  };
  
*** a/drivers/stm/pio.c	2011-01-03 11:13:42.000000000 +0100
--- b/pio.c	2011-01-03 17:15:15.958555041 +0100
***************
*** 154,159 ****
--- 154,168 ----
  		stpio_configure_pin(pin, direction);
  		pin->name = name;
  	}
+ 	else
+ 	{
+ 		pin = &stpio_ports[portno].pins[pinno];
+ 
+ 		if( pin )
+ 		{
+ 			printk( "GPIO %d %d is occupied by %s\n", portno, pinno, pin->name );
+ 		}
+ 	}
  
  	spin_unlock(&stpio_lock);
  
***************
*** 163,169 ****
  
  void stpio_free_pin(struct stpio_pin *pin)
  {
! 	stpio_configure_pin(pin, STPIO_IN);
  	pin->name = NULL;
  	pin->func = 0;
  	pin->dev  = 0;
--- 172,178 ----
  
  void stpio_free_pin(struct stpio_pin *pin)
  {
! //	stpio_configure_pin(pin, STPIO_IN);
  	pin->name = NULL;
  	pin->func = 0;
  	pin->dev  = 0;
--- linux-sh4/drivers/net/stmmac/stmmac_main.c	2011-01-04 16:02:05.000000000 +0100
+++ linux-sh4/drivers/net/stmmac/stmmac_main.c.patched	2011-01-12 17:21:54.000000000 +0100
@@ -389,6 +389,96 @@
 	return 0;
 }
 
+#include <linux/ctype.h>
+
+static unsigned long simple_strtoul2(const char *cp,char **endp,unsigned int base)
+{
+	unsigned long result = 0,value;
+
+	if (*cp == '0') {
+		cp++;
+		if ((*cp == 'x') && isxdigit(cp[1])) {
+			base = 16;
+			cp++;
+		}
+		if (!base) {
+			base = 8;
+		}
+	}
+	if (!base) {
+		base = 10;
+	}
+	while (isxdigit(*cp) && (value = isdigit(*cp) ? *cp-'0' : (islower(*cp)
+	    ? toupper(*cp) : *cp)-'A'+10) < base) {
+		result = result*base + value;
+		cp++;
+	}
+	if (endp)
+		*endp = (char *)cp;
+	return result;
+}
+
+/**
+ * get_mac_addr
+ * @ioaddr: device I/O address
+ * @addr: mac address
+ * @high: High register offset
+ * @low: low register offset
+ * Description: the function gets the hardware MAC address
+ */
+
+char g_env_mac[]="00:50:FD:00:00:00";
+
+static void get_mac_address(unsigned long ioaddr, unsigned char *addr,
+			    unsigned int high, unsigned int low)
+{
+	unsigned int hi_addr, lo_addr;
+
+    int env_size,reg,env_present=0, tmp_count;
+    char *s = NULL, *e,  es[] = "00:00:00:00:00:00";
+    unsigned char s_env_mac[64], v_env_mac[6];
+
+	/* Read the MAC address from the hardware */
+	env_size = 17;
+	memcpy(s_env_mac, g_env_mac, 17);
+
+	if (env_size > 0) {
+			env_present = 1;
+			s = s_env_mac;
+		}
+
+	//by doliyu ...patch for mac address write error!
+	for(tmp_count =0 ; tmp_count < 17; tmp_count++)
+	{
+		if(s_env_mac[tmp_count] == ' ')
+			s_env_mac[tmp_count]='0';
+		printk("s_env_mac[%d] = %c\n", tmp_count, s_env_mac[tmp_count]);
+	}
+	
+	for (reg = 0; reg < 6; ++reg) { /* turn string into mac value */
+		v_env_mac[reg] = s ? simple_strtoul2 (s, &e, 16) : 0;
+		if (s)
+			s = (*e) ? e + 1 : e;
+	}
+
+    printk("get ether addr = %x, %x, %x, %x, %x, %x \n", v_env_mac[0],v_env_mac[1],v_env_mac[2],v_env_mac[3],v_env_mac[4],v_env_mac[5]);
+    hi_addr = (int)(v_env_mac[5]) <<8 | (int)(v_env_mac[4]);
+    lo_addr = (int)(v_env_mac[3]) << 24 | (int)(v_env_mac[2]) << 16 | (int)(v_env_mac[1]) <<8 | v_env_mac[0];
+
+//	hi_addr = readl(ioaddr + high);
+//	lo_addr = readl(ioaddr + low);
+
+	/* Extract the MAC address from the high and low words */
+	addr[0] = lo_addr & 0xff;
+	addr[1] = (lo_addr >> 8) & 0xff;
+	addr[2] = (lo_addr >> 16) & 0xff;
+	addr[3] = (lo_addr >> 24) & 0xff;
+	addr[4] = hi_addr & 0xff;
+	addr[5] = (hi_addr >> 8) & 0xff;
+
+	return;
+}
+
 /**
  * stmmac_mac_enable_rx
  * @dev: net device structure
@@ -2348,6 +2438,9 @@
 			tmrate = simple_strtoul(opt + 7, NULL, 0);
 #endif
 		}
+		else if( !strncmp(opt,"ethaddr:", 8)) {
+			memcpy(g_env_mac, opt + 8, 17);
+		}
 	}
 	return 0;
 }
*** a/arch/sh/boards/st/mb448/setup.c	2011-01-30 10:43:58.000000000 +0100
--- b/setup.c	2011-01-30 14:37:47.912176246 +0100
***************
*** 18,23 ****
--- 18,24 ----
  #include <linux/mtd/mtd.h>
  #include <linux/mtd/physmap.h>
  #include <linux/mtd/partitions.h>
+ #include <linux/bpa2.h>
  #include <linux/phy.h>
  #include <asm/irl.h>
  
***************
*** 25,42 ****
  
  static int ascs[2] __initdata = { 2, 3 };
  
  void __init mb448_setup(char** cmdline_p)
  {
  	printk("STMicroelectronics STb7109E Reference board initialisation\n");
  
  	stx7100_early_device_init();
! 	stb7100_configure_asc(ascs, 2, 0);
  }
  
  static struct plat_ssc_data ssc_private_info = {
  	.capability  =
  		ssc0_has(SSC_I2C_CAPABILITY) |
! 		ssc1_has(SSC_SPI_CAPABILITY) |
  		ssc2_has(SSC_I2C_CAPABILITY),
  };
  
--- 26,65 ----
  
  static int ascs[2] __initdata = { 2, 3 };
  
+ const char *LMI_VID_partalias[] = { "BPA2_Region1", "coredisplay-video", "gfx-memory", "v4l2-video-buffers", NULL };
+ const char *LMI_SYS_partalias[] = { "BPA2_Region0", "bigphysarea", "v4l2-coded-video-buffers", NULL };
+ 
+ static struct bpa2_partition_desc bpa2_parts_table[] = {
+ 	{
+ 		.name  = "LMI_VID",
+ 		.start = 0x10800000,
+ 		.size  = 0x03800000,
+ 		.flags = 0,
+ 		.aka   = LMI_VID_partalias
+ 	},
+ 	{
+ 		.name  = "LMI_SYS",
+ 		.start = 0,
+ 		.size  = 0x00F00000,
+ 		.flags = 0,
+ 		.aka   = LMI_SYS_partalias
+ 	}
+ };
+ 
+ 
  void __init mb448_setup(char** cmdline_p)
  {
  	printk("STMicroelectronics STb7109E Reference board initialisation\n");
  
  	stx7100_early_device_init();
! 	stb7100_configure_asc(ascs, 2, 1);
!   	bpa2_init(bpa2_parts_table, ARRAY_SIZE(bpa2_parts_table));
  }
  
  static struct plat_ssc_data ssc_private_info = {
  	.capability  =
  		ssc0_has(SSC_I2C_CAPABILITY) |
! 		ssc1_has(SSC_I2C_CAPABILITY) |
  		ssc2_has(SSC_I2C_CAPABILITY),
  };
  
***************
*** 62,84 ****
  
  static void set_vpp(struct map_info * info, int enable)
  {
! 	stpio_set_pin(vpp_pio, enable);
  }
  
! static struct mtd_partition mtd_parts_table[3] = {
  	{
! 		.name = "Boot firmware",
! 		.size = 0x00040000,
! 		.offset = 0x00000000,
! 	}, {
! 		.name = "Kernel",
! 		.size = 0x00100000,
! 		.offset = 0x00040000,
! 	}, {
! 		.name = "Root FS",
! 		.size = MTDPART_SIZ_FULL,
! 		.offset = 0x00140000,
! 	}
  };
  
  static struct physmap_flash_data physmap_flash_data = {
--- 85,200 ----
  
  static void set_vpp(struct map_info * info, int enable)
  {
! //	stpio_set_pin(vpp_pio, enable);
  }
  
! static struct mtd_partition mtd_parts_table[] = 
! {
! #if 1
  	{
! 	 .name = "Boot firmware : 	0xA000.0000-0xA003.FFFF",
! 	 .size = 0x00040000,
! 	 .offset = 0x00000000,
! 	 /* mask_flags: MTD_WRITEABLE *//* force read-only */
! 	 },
! 	 {
! 	 .name = "Kernel - 		0xA004.0000-0xA01F.FFFF",
! 	 .size = 0x001C0000,
! 	 .offset = 0x00040000,
! 	 },
! 	 {
! 	 .name = "Config FS - 		0xA020.0000-0xA029.FFFF",
! 	 .size = 0x0A0000,	
! 	 .offset = 0x00200000,
! 	 },
! 	 {
! 	 .name = "Root FS-		0xA02A.0000-0xA04D.FFFF",
! 	 .size = 0x240000,	/* will expand to the end of the flash */
! 	 .offset = 0x002A0000,
! 	 },
! 	 {
! 	 .name = "App + GTK 1		0xA04E.0000-0xA0FF.FFFF",
! 	 .size = 0x00B20000,	/* will expand to the end of the flash */
! 	 .offset = 0x004E0000,
! 	 },
! 	 {
! 	 .name = "App + GTK 2		0xA04E.0000-0xA0FF.FFFF",
! 	 .size = 0x00B20000,	/* will expand to the end of the flash */
! 	 .offset = 0x004E0000,
! 	 },
! 	 {
! 	 .name = "App + GTK 3		0xA04E.0000-0xA0FF.FFFF",
! 	 .size = 0x00B20000,	/* will expand to the end of the flash */
! 	 .offset = 0x004E0000,
! 	 },
! 	 {
! 	 .name = "All Flash Block	0xA000.0000-0xA0FF.FFFF",
! 	 .size = 0x01000000,	/* will expand to the end of the flash */
! 	 .offset = 0x00000000,
! 	 },
! 	 {
! 	 .name = "Update Flash Block	0xA004.0000-0xA0FF.FFFF",
! 	 .size = 0x00fC0000,	/* will expand to the end of the flash */
! 	 .offset = 0x00040000,
! 	 },
! 	 {
! 	 .name = "Update Flash Block	0xA004.0000-0xA0FF.FFFF",
! 	 .size = 0x00fC0000,	/* will expand to the end of the flash */
! 	 .offset = 0x00040000,
! 	 }	 
! #else
! 	{
! 	 .name = "Boot firmware : 	0xA000.0000-0xA003.FFFF",
! 	 .size = 0x00040000,
! 	 .offset = 0x00000000,
! 	 /* mask_flags: MTD_WRITEABLE *//* force read-only */
! 	 },
! 	 {
! 	 .name = "Kernel - 		0xA004.0000-0xA01F.FFFF",
! 	 .size = 0x001C0000,
! 	 .offset = 0x00040000,
! 	 },
! 	 {
! 	 .name = "Config FS - 		0xA020.0000-0xA029.FFFF",
! 	 .size = 0x0A0000,	
! 	 .offset = 0x00200000,
! 	 },
! 	 {
! 	 .name = "Root FS-		0xA02A.0000-0xA04D.FFFF",
! 	 .size = 0x240000,	/* will expand to the end of the flash */
! 	 .offset = 0x002A0000,
! 	 },
! 	 {
! 	 .name = "APP_Modules		0xA04E.0000-0xA0AF.FFFF",
! 	 .size = 0x0620000,	/* will expand to the end of the flash */
! 	 .offset = 0x004E0000,
! 	 },
! 	 {
! 	 .name = "App + GTK		0xA04E.0000-0xA0FF.FFFF",
! 	 .size = 0x00B20000,	/* will expand to the end of the flash */
! 	 .offset = 0x004E0000,
! 	 },
! 	 {
! 	 .name = "GTK Lib		0xA0B0.0000-0xA0FF.FFFF",
! 	 .size = 0x00500000,	/* will expand to the end of the flash */
! 	 .offset = 0x00B00000,
! 	 },
! 	 {
! 	 .name = "All Flash Block	0xA000.0000-0xA0FF.FFFF",
! 	 .size = 0x01000000,	/* will expand to the end of the flash */
! 	 .offset = 0x00000000,
! 	 },
! 	 {
! 	 .name = "Update Flash Block	0xA004.0000-0xA0FF.FFFF",
! 	 .size = 0x00fC0000,	/* will expand to the end of the flash */
! 	 .offset = 0x00040000,
! 	 },
! 	 {
! 	 .name = "Update Flash Block	0xA004.0000-0xA0FF.FFFF",
! 	 .size = 0x00fC0000,	/* will expand to the end of the flash */
! 	 .offset = 0x00040000,
! 	 }	 
! #endif	 
  };
  
  static struct physmap_flash_data physmap_flash_data = {
***************
*** 90,96 ****
  
  static struct resource physmap_flash_resource = {
  	.start		= 0x00000000,
! 	.end		= 0x00800000 - 1,
  	.flags		= IORESOURCE_MEM,
  };
  
--- 206,212 ----
  
  static struct resource physmap_flash_resource = {
  	.start		= 0x00000000,
! 	.end		= 0x01000000 - 1,
  	.flags		= IORESOURCE_MEM,
  };
  
***************
*** 106,112 ****
  
  static struct plat_stmmacphy_data phy_private_data = {
  	.bus_id = 0,
! 	.phy_addr = 14,
  	.phy_mask = 1,
  	.interface = PHY_INTERFACE_MODE_MII,
  	.phy_reset = NULL,
--- 222,228 ----
  
  static struct plat_stmmacphy_data phy_private_data = {
  	.bus_id = 0,
! 	.phy_addr = 2,
  	.phy_mask = 1,
  	.interface = PHY_INTERFACE_MODE_MII,
  	.phy_reset = NULL,
***************
*** 119,126 ****
  	.resource	= (struct resource[]) {
                  {
  			.name	= "phyirq",
! 			.start	= IRL0_IRQ,
! 			.end	= IRL0_IRQ,
  			.flags	= IORESOURCE_IRQ,
  		},
  	},
--- 235,242 ----
  	.resource	= (struct resource[]) {
                  {
  			.name	= "phyirq",
! 			.start	= IRL3_IRQ,
! 			.end	= IRL3_IRQ,
  			.flags	= IORESOURCE_IRQ,
  		},
  	},
***************
*** 135,159 ****
  	&mb448_phy_device,
  };
  
  static int __init device_init(void)
  {
  	struct stpio_pin *smc91x_reset;
  
  	stx7100_configure_sata();
  	stx7100_configure_ssc(&ssc_private_info);
  	stx7100_configure_usb();
! 	stx7100_configure_ethernet(0, 0, 0);
  
! 	vpp_pio = stpio_request_pin(2,7, "VPP", STPIO_OUT);
  
! 	/* Reset the SMSC 91C111 Ethernet chip */
! 	smc91x_reset = stpio_request_set_pin(2, 6, "smc91x_reset",
  					     STPIO_OUT, 0);
  	udelay(1);
  	stpio_set_pin(smc91x_reset, 1);
  	udelay(1);
  	stpio_set_pin(smc91x_reset, 0);
  
  	return platform_add_devices(mb448_devices,
  				    ARRAY_SIZE(mb448_devices));
  }
--- 251,283 ----
  	&mb448_phy_device,
  };
  
+ static struct plat_stm_pwm_data pwm_private_info = {
+ 	.flags		= PLAT_STM_PWM_OUT1,
+ };
+ 
  static int __init device_init(void)
  {
  	struct stpio_pin *smc91x_reset;
  
  	stx7100_configure_sata();
+ //	stx7100_configure_pwm(&pwm_private_info);	
  	stx7100_configure_ssc(&ssc_private_info);
  	stx7100_configure_usb();
! 	stx7100_configure_ethernet(0, 1, 0);
  
! //	vpp_pio = stpio_request_pin(2,7, "VPP", STPIO_OUT);
  
! 	/* Reset the USB POWER 
! 	smc91x_reset = stpio_request_set_pin(5, 7, "smc91x_reset",
  					     STPIO_OUT, 0);
  	udelay(1);
  	stpio_set_pin(smc91x_reset, 1);
  	udelay(1);
  	stpio_set_pin(smc91x_reset, 0);
  
+ 	udelay( 10 );
+ 	stpio_free_pin( smc91x_reset );
+ */
  	return platform_add_devices(mb448_devices,
  				    ARRAY_SIZE(mb448_devices));
  }
