*** a/arch/sh/boards/st/mb448/setup.c	2011-04-24 11:27:10.000000000 +0200
--- b/setup.c	2011-04-24 16:48:31.641054509 +0200
***************
*** 15,26 ****
--- 15,47 ----
  #include <linux/stm/pio.h>
  #include <linux/stm/soc.h>
  #include <linux/delay.h>
+ #include <linux/bpa2.h>
  #include <linux/mtd/mtd.h>
  #include <linux/mtd/physmap.h>
  #include <linux/mtd/partitions.h>
  #include <linux/phy.h>
  #include <asm/irl.h>
  
+ const char *LMI_VID_partalias[] = { "BPA2_Region1", "coredisplay-video", "v4l2-video-buffers", "gfx-memory", NULL };
+ const char *LMI_SYS_partalias[] = { "BPA2_Region0", "bigphysarea",  NULL };
+ 
+ static struct bpa2_partition_desc bpa2_parts_table[] = {
+         {
+                 .name  = "LMI_VID",
+                 .start = 0x10800000,
+                 .size  = 0x03800000,
+                 .flags = 0,
+                 .aka   = LMI_VID_partalias
+         },
+         {
+                 .name  = "LMI_SYS",
+                 .start = 0,
+                 .size  = 0x01200000,
+                 .flags = 0,
+                 .aka   = LMI_SYS_partalias
+         }
+ };
+ 
  static struct stpio_pin *vpp_pio;
  
  static int ascs[2] __initdata = { 2, 3 };
***************
*** 31,42 ****
  
  	stx7100_early_device_init();
  	stb7100_configure_asc(ascs, 2, 0);
  }
  
  static struct plat_ssc_data ssc_private_info = {
  	.capability  =
  		ssc0_has(SSC_I2C_CAPABILITY) |
! 		ssc1_has(SSC_SPI_CAPABILITY) |
  		ssc2_has(SSC_I2C_CAPABILITY),
  };
  
--- 52,64 ----
  
  	stx7100_early_device_init();
  	stb7100_configure_asc(ascs, 2, 0);
+     bpa2_init(bpa2_parts_table, ARRAY_SIZE(bpa2_parts_table));
  }
  
  static struct plat_ssc_data ssc_private_info = {
  	.capability  =
  		ssc0_has(SSC_I2C_CAPABILITY) |
! 		ssc1_has(SSC_I2C_CAPABILITY) |
  		ssc2_has(SSC_I2C_CAPABILITY),
  };
  
***************
*** 62,68 ****
  
  static void set_vpp(struct map_info * info, int enable)
  {
! 	stpio_set_pin(vpp_pio, enable);
  }
  
  static struct mtd_partition mtd_parts_table[3] = {
--- 84,90 ----
  
  static void set_vpp(struct map_info * info, int enable)
  {
! //	stpio_set_pin(vpp_pio, enable);
  }
  
  static struct mtd_partition mtd_parts_table[3] = {
***************
*** 106,112 ****
  
  static struct plat_stmmacphy_data phy_private_data = {
  	.bus_id = 0,
! 	.phy_addr = 14,
  	.phy_mask = 1,
  	.interface = PHY_INTERFACE_MODE_MII,
  	.phy_reset = NULL,
--- 128,134 ----
  
  static struct plat_stmmacphy_data phy_private_data = {
  	.bus_id = 0,
! 	.phy_addr = 2,
  	.phy_mask = 1,
  	.interface = PHY_INTERFACE_MODE_MII,
  	.phy_reset = NULL,
***************
*** 144,160 ****
  	stx7100_configure_usb();
  	stx7100_configure_ethernet(0, 0, 0);
  
! 	vpp_pio = stpio_request_pin(2,7, "VPP", STPIO_OUT);
  
  	/* Reset the SMSC 91C111 Ethernet chip */
  	smc91x_reset = stpio_request_set_pin(2, 6, "smc91x_reset",
  					     STPIO_OUT, 0);
  	udelay(1);
  	stpio_set_pin(smc91x_reset, 1);
  	udelay(1);
  	stpio_set_pin(smc91x_reset, 0);
! 
! 	return platform_add_devices(mb448_devices,
  				    ARRAY_SIZE(mb448_devices));
  }
  
--- 166,183 ----
  	stx7100_configure_usb();
  	stx7100_configure_ethernet(0, 0, 0);
  
! //	vpp_pio = stpio_request_pin(2,7, "VPP", STPIO_OUT);
  
  	/* Reset the SMSC 91C111 Ethernet chip */
  	smc91x_reset = stpio_request_set_pin(2, 6, "smc91x_reset",
  					     STPIO_OUT, 0);
+ 	stpio_set_pin(smc91x_reset, 0);
  	udelay(1);
  	stpio_set_pin(smc91x_reset, 1);
  	udelay(1);
  	stpio_set_pin(smc91x_reset, 0);
! 	
!     return platform_add_devices(mb448_devices,
  				    ARRAY_SIZE(mb448_devices));
  }
  
