--- linux/drivers/serial/stasc.h.orig	2010-03-31 23:49:40.000000000 +0200
+++ linux/drivers/serial/stasc.h	2010-03-31 23:49:53.000000000 +0200
@@ -40,7 +40,8 @@
 #elif defined(CONFIG_CPU_SUBTYPE_STI5528)
 #define ASC_NPORTS             2
 #elif defined(CONFIG_CPU_SUBTYPE_STB7100)
-#define ASC_NPORTS             2
+//#define ASC_NPORTS             2
+#define ASC_NPORTS             1
 #else
 #error "Unknown CPU"
 #endif
