--- a/sound/stm/pcm_player.c	2011-06-29 20:24:54.000000000 +0200
+++ b/sound/stm/pcm_player.c	2011-07-01 17:12:20.992653000 +0200
@@ -673,10 +673,7 @@
 
 	/* Mute & shutdown DAC */
 
-	if (pcm_player->conv_group) {
-		snd_stm_conv_mute(pcm_player->conv_group);
-		snd_stm_conv_disable(pcm_player->conv_group);
-	}
+	// No we will not
 
 	/* Disable interrupts */
 
@@ -685,15 +682,17 @@
 	disable_irq_nosync(pcm_player->irq);
 
 	/* Stop PCM player */
-
-	set__AUD_PCMOUT_CTRL__MODE__OFF(pcm_player);
+	
+	// No, pause it because stopping causes cracking noise
+	set__AUD_PCMOUT_CTRL__MODE__MUTE(pcm_player);
 
 	/* Stop FDMA transfer */
 
 	dma_stop_channel(pcm_player->fdma_channel);
 
 	/* Reset PCM player */
-	set__AUD_PCMOUT_RST__SRSTP__RESET(pcm_player);
+	
+	// No we will not
 
 	return 0;
 }
