--- linux-sh4.orig/arch/sh/kernel/sh_ksyms.c	2010-05-23 15:11:21.000000000 +0200
+++ linux-sh4/arch/sh/kernel/sh_ksyms.c	2010-05-23 15:11:46.000000000 +0200
@@ -122,9 +122,10 @@
 DECLARE_EXPORT(__movmem_i4_odd);
 DECLARE_EXPORT(__movmemSI12_i4);
 
+DECLARE_EXPORT(__umoddi3);
-#if (__GNUC_MINOR__ == 2 || defined(__GNUC_STM_RELEASE__))
+#if (__GNUC_MINOR__ >= 2 || defined(__GNUC_STM_RELEASE__))
 /*
- * GCC 4.2 emits these for division, as do GCC 4.1.x versions of the ST
+ * GCC >= 4.2 emits these for division, as do GCC 4.1.x versions of the ST
  * compiler which include backported patches.
  */
 DECLARE_EXPORT(__sdivsi3_i4i);
