*** linux-sh4-2.6.32.28_stm24_0207_orig/scripts/kconfig/confdata.c	Mon May  3 15:58:22 2010
--- linux-sh4/scripts/kconfig/confdata.c	Mon May  3 16:24:53 2010
***************
*** 730,735 ****
--- 730,740 ----
  				fprintf(out_h, "#define CONFIG_%s_MODULE 1\n", sym->name);
  				break;
  			case yes:
+ 				if(!memcmp(sym->name, "32BIT",5))
+ 				{
+ 					sym->name="29BIT";
+ 					//printf("CONFIG_%s=y\n", sym->name);
+ 				}
  				fprintf(out, "CONFIG_%s=y\n", sym->name);
  				fprintf(out_h, "#define CONFIG_%s 1\n", sym->name);
  				break;
