diff -u --recursive --new-file linux-2.6.17.14_stm22_0041.org/drivers/char/lirc/Kconfig linux-2.6.17.14_stm22_0041/drivers/char/lirc/Kconfig
--- linux-2.6.17.14_stm22_0041.org/drivers/char/lirc/Kconfig	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.17.14_stm22_0040/drivers/char/lirc/Kconfig	2008-06-25 21:12:03.000000000 +0200
@@ -261,7 +261,7 @@
 			depends on LIRC_SIR
         config ST_LIRC
                 tristate "STM Lirc support"
-                depends on LIRC_SUPPORT && (SH_STM8000_DEMO || SH_ST220_EVAL || SH_STI5528_EVAL || SH_STI5528_ESPRESSO || SH_STB7100_MBOARD || SH_STB7100_REF || SH_STB7109E_REF || SH_HMS1)
+                depends on LIRC_SUPPORT && (SH_STM8000_DEMO || SH_ST220_EVAL || SH_STI5528_EVAL || SH_STI5528_ESPRESSO || SH_STB7100_MBOARD || SH_STB7100_REF || SH_STB7109E_REF || SH_HMS1 || SH_IPBOX9900 || SH_IPBOX99 || SH_IPBOX55)
 
                 choice
                         prompt "Receiver type"
@@ -273,7 +273,7 @@
 
                         config IRB_RECEIVER
                         bool "IRB Receiver"
-			depends on ST_LIRC && (SH_STM8000_DEMO || SH_ST220_EVAL || SH_STI5528_EVAL || SH_STB7100_MBOARD || SH_STB7100_REF || SH_STB7109E_REF || SH_HMS1)
+			depends on ST_LIRC && (SH_STM8000_DEMO || SH_ST220_EVAL || SH_STI5528_EVAL || SH_STB7100_MBOARD || SH_STB7100_REF || SH_STB7109E_REF || SH_HMS1 || SH_IPBOX9900 || SH_IPBOX99 || SH_IPBOX55)
 
                         config UHF_RECEIVER
                         bool "UHF Receiver"
