*** a/arch/sh/boards/mach-mb448/setup.c	2011-07-11 21:21:16.000000000 +0200
--- b/setup.c	2011-07-13 22:15:15.134402345 +0200
***************
*** 17,22 ****
--- 17,24 ----
  #include <linux/phy.h>
  #include <linux/mtd/mtd.h>
  #include <linux/mtd/physmap.h>
+ #include <linux/mtd/partitions.h>
+ #include <linux/bpa2.h>
  #include <linux/stm/platform.h>
  #include <linux/stm/stx7100.h>
  #include <asm/irl.h>
***************
*** 26,31 ****
--- 28,53 ----
  #define MB448_PIO_SMC91X_RESET stm_gpio(2, 6)
  #define MB448_PIO_FLASH_VPP stm_gpio(2, 7)
  
+ const char *LMI_VID_partalias[] = { "BPA2_Region1", "coredisplay-video", "gfx-memory", "v4l2-video-buffers", NULL };
+ const char *LMI_SYS_partalias[] = { "BPA2_Region0", "bigphysarea", "v4l2-coded-video-buffers", NULL };
+ 
+ 
+ static struct bpa2_partition_desc bpa2_parts_table[] = {
+    {
+        .name  = "LMI_VID",
+        .start = 0x10800000,
+        .size  = 0x03800000,
+        .flags = 0,
+        .aka   = LMI_VID_partalias
+    },
+    {
+        .name  = "LMI_SYS",
+        .start = 0,
+        .size  = 0x01200000,
+        .flags = 0,
+        .aka   = LMI_SYS_partalias
+    }
+ };
  
  
  void __init mb448_setup(char **cmdline_p)
***************
*** 35,45 ****
  	stx7100_early_device_init();
  
  	stx7100_configure_asc(2, &(struct stx7100_asc_config) {
! 			.hw_flow_control = 0,
  			.is_console = 1, });
  	stx7100_configure_asc(3, &(struct stx7100_asc_config) {
! 			.hw_flow_control = 0,
  			.is_console = 0, });
  }
  
  
--- 57,70 ----
  	stx7100_early_device_init();
  
  	stx7100_configure_asc(2, &(struct stx7100_asc_config) {
! 			.hw_flow_control = 1,
  			.is_console = 1, });
+ 
  	stx7100_configure_asc(3, &(struct stx7100_asc_config) {
! 			.hw_flow_control = 1,
  			.is_console = 0, });
+     
+     bpa2_init(bpa2_parts_table, ARRAY_SIZE(bpa2_parts_table));
  }
  
  
***************
*** 68,74 ****
  
  static void mb448_set_vpp(struct map_info *info, int enable)
  {
! 	gpio_set_value(MB448_PIO_FLASH_VPP, enable);
  }
  
  static struct platform_device mb448_physmap_flash = {
--- 93,99 ----
  
  static void mb448_set_vpp(struct map_info *info, int enable)
  {
! //	gpio_set_value(MB448_PIO_FLASH_VPP, enable);
  }
  
  static struct platform_device mb448_physmap_flash = {
***************
*** 86,92 ****
  
  
  
! #define STMMAC_PHY_ADDR 14
  static int stmmac_phy_irqs[PHY_MAX_ADDR] = {
  	[STMMAC_PHY_ADDR] = IRL0_IRQ,
  };
--- 111,117 ----
  
  
  
! #define STMMAC_PHY_ADDR 2
  static int stmmac_phy_irqs[PHY_MAX_ADDR] = {
  	[STMMAC_PHY_ADDR] = IRL0_IRQ,
  };
***************
*** 106,112 ****
  	stx7100_configure_sata();
  
  	stx7100_configure_ssc_i2c(0);
! 	stx7100_configure_ssc_spi(1, NULL);
  	stx7100_configure_ssc_i2c(2);
  
  	stx7100_configure_usb();
--- 131,137 ----
  	stx7100_configure_sata();
  
  	stx7100_configure_ssc_i2c(0);
! 	stx7100_configure_ssc_i2c(1);
  	stx7100_configure_ssc_i2c(2);
  
  	stx7100_configure_usb();
***************
*** 119,130 ****
  			.mdio_bus_data = &stmmac_mdio_bus,
  		});
  
! 	gpio_request(MB448_PIO_FLASH_VPP, "Flash VPP");
  	gpio_direction_output(MB448_PIO_FLASH_VPP, 0);
! 
  	/* Reset the SMSC 91C111 Ethernet chip */
  	gpio_request(MB448_PIO_SMC91X_RESET, "SMC91x reset");
  	gpio_direction_output(MB448_PIO_SMC91X_RESET, 0);
  	udelay(1);
  	gpio_set_value(MB448_PIO_SMC91X_RESET, 1);
  	udelay(1);
--- 144,156 ----
  			.mdio_bus_data = &stmmac_mdio_bus,
  		});
  
! /*	gpio_request(MB448_PIO_FLASH_VPP, "Flash VPP");
  	gpio_direction_output(MB448_PIO_FLASH_VPP, 0);
! */
  	/* Reset the SMSC 91C111 Ethernet chip */
  	gpio_request(MB448_PIO_SMC91X_RESET, "SMC91x reset");
  	gpio_direction_output(MB448_PIO_SMC91X_RESET, 0);
+ 	gpio_set_value(MB448_PIO_SMC91X_RESET, 0);
  	udelay(1);
  	gpio_set_value(MB448_PIO_SMC91X_RESET, 1);
  	udelay(1);
*** a/drivers/stm/stx7100_comms.c	2011-07-11 21:21:19.000000000 +0200
--- b/stx7100_comms.c	2011-07-14 19:51:23.729808261 +0200
***************
*** 240,251 ****
  			STM_PAD_PIO_BIDIR_NAMED(3, 0, 1, "SCL"),
  			STM_PAD_PIO_BIDIR_NAMED(3, 1, 1, "SDA"),
  		},
! 		.sysconfs_num = 2,
  		.sysconfs = (struct stm_pad_sysconf []) {
  			/* SSC1_MUX_SEL = 0 (default assignment) */
  			STM_PAD_SYS_CFG(7, 2, 2, 0),
! 			/* DVO_OUT_ON = 0 (SSC not DVO) */
! 			STM_PAD_SYS_CFG(7, 10, 10, 0),
  		},
  	},
  	[2] = {
--- 240,251 ----
  			STM_PAD_PIO_BIDIR_NAMED(3, 0, 1, "SCL"),
  			STM_PAD_PIO_BIDIR_NAMED(3, 1, 1, "SDA"),
  		},
! 		.sysconfs_num = 1,
  		.sysconfs = (struct stm_pad_sysconf []) {
  			/* SSC1_MUX_SEL = 0 (default assignment) */
  			STM_PAD_SYS_CFG(7, 2, 2, 0),
! 			/* DVO_OUT_ON = 0 (SSC not DVO) 
! 			STM_PAD_SYS_CFG(7, 10, 10, 0),*/
  		},
  	},
  	[2] = {
