--- linux-sh4-2.6.23.17_stm23_0123_orig/arch/sh/kernel/cpu/sh4/setup-stb7100.c	2011-05-27 18:30:22.000000000 +0200
+++ linux-sh4/arch/sh/kernel/cpu/sh4/setup-stb7100.c	2011-05-27 18:54:01.000000000 +0200
@@ -296,9 +296,9 @@
 static char i2c_st[] = "i2c_st";
 static char spi_st[] = "spi_st_ssc";
 static struct platform_device stssc_devices[] = {
-	STSSC_DEVICE(0x18040000, 119, 2, 0, 1, 2),
-	STSSC_DEVICE(0x18041000, 118, 3, 0, 1, 2),
-	STSSC_DEVICE(0x18042000, 117, 4, 0, 1, 0xff),
+	STSSC_DEVICE(0x18040000, 119, 2, 0, 1, 0xff),
+	STSSC_DEVICE(0x18041000, 118, 3, 0, 1, 0xff),
+
 };
 
 static int __initdata num_i2c;
@@ -592,24 +592,6 @@
 		.dir  = STPIO_IN,
                 .pinof= 0x00 | LIRC_IR_RX | LIRC_PIO_ON
 	},
-	[1] = {
-		.bank = 3,
-		.pin  = 4,
-		.dir  = STPIO_IN,
-                .pinof= 0x00 | LIRC_UHF_RX | LIRC_PIO_ON
-	},
-	[2] = {
-		.bank = 3,
-		.pin  = 5,
-		.dir  = STPIO_ALT_OUT,
-                .pinof= 0x00 | LIRC_IR_TX /* | LIRC_PIO_ON not available */
-	},
-	[3] = {
-		.bank = 3,
-		.pin  = 6,
-		.dir  = STPIO_ALT_OUT,
-                .pinof= 0x00 | LIRC_IR_TX /* | LIRC_PIO_ON not available */
-	}
 };
 
 static struct plat_lirc_data lirc_private_info = {
