diff -u --recursive --new-file oldlinux-2.6.17.14_stm22_0040/drivers/media/dvb/dvb-core/dvb_ca_en50221.c linux-2.6.17.14_stm22_0040/drivers/media/dvb/dvb-core/dvb_ca_en50221.c
--- oldlinux-2.6.17.14_stm22_0040/drivers/media/dvb/dvb-core/dvb_ca_en50221.c	2008-07-19 17:47:55.000000000 +0200
+++ linux-2.6.17.14_stm22_0040/drivers/media/dvb/dvb-core/dvb_ca_en50221.c	2008-07-20 19:25:31.000000000 +0200
@@ -1256,7 +1256,15 @@
 		}
 		break;
 	}
-
+	/* Dagobert */
+	case CA_SET_DESCR:
+	{
+		ca_descr_t *descr = (ca_descr_t*) parg;
+ 
+		if (ca->pub->descrambler != NULL)
+			ca->pub->descrambler(ca->pub, descr);
+		break;
+	}
 	default:
 		err = -EINVAL;
 		break;
diff -u --recursive --new-file oldlinux-2.6.17.14_stm22_0040/drivers/media/dvb/dvb-core/dvb_ca_en50221.h linux-2.6.17.14_stm22_0040/drivers/media/dvb/dvb-core/dvb_ca_en50221.h
--- oldlinux-2.6.17.14_stm22_0040/drivers/media/dvb/dvb-core/dvb_ca_en50221.h	2006-06-18 03:49:35.000000000 +0200
+++ linux-2.6.17.14_stm22_0040/drivers/media/dvb/dvb-core/dvb_ca_en50221.h	2008-07-20 19:26:20.000000000 +0200
@@ -60,7 +60,8 @@
 	int (*slot_reset)(struct dvb_ca_en50221* ca, int slot);
 	int (*slot_shutdown)(struct dvb_ca_en50221* ca, int slot);
 	int (*slot_ts_enable)(struct dvb_ca_en50221* ca, int slot);
-
+         /* Dagobert */
+	int (*descrambler)(struct dvb_ca_en50221* ca, ca_descr_t* descr);
 	/*
 	* Poll slot status.
 	* Only necessary if DVB_CA_FLAG_EN50221_IRQ_CAMCHANGE is not set
