--- minidlna-1.0.24/Makefile	2012-01-17 23:49:01.000000000 +0100
+++ minidlna-1.0.24/Makefile.patched	2012-05-17 00:51:15.000000000 +0200
@@ -13,12 +13,15 @@
 #CFLAGS = -Wall -O -D_GNU_SOURCE -g -DDEBUG
 #CFLAGS = -Wall -g -Os -D_GNU_SOURCE
 CFLAGS = -Wall -g -O3 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 \
-	 -I/usr/include/ffmpeg \
-	 -I/usr/include/libavutil -I/usr/include/libavcodec -I/usr/include/libavformat \
-	 -I/usr/include/ffmpeg/libavutil -I/usr/include/ffmpeg/libavcodec -I/usr/include/ffmpeg/libavformat
+	 -I. \
+	 -I$(DESTDIR)/usr/include \
+	 -I$(DESTDIR)/usr/include/ffmpeg \
+	 -I$(DESTDIR)/usr/include/libavformat \
+	 -I$(DESTDIR)/usr/include/libavutil \
+	 -I$(DESTDIR)/usr/include/libavcodec \
 #STATIC_LINKING: CFLAGS += -DSTATIC
 #STATIC_LINKING: LDFLAGS = -static
-CC = gcc
+
 RM = rm -f
 INSTALL = install
 
@@ -37,7 +40,7 @@
 
 ALLOBJS = $(BASEOBJS) $(LNXOBJS)
 
-LIBS = -lpthread -lexif -ljpeg -lsqlite3 -lavformat -lavutil -lavcodec -lid3tag -lFLAC -logg -lvorbis
+LIBS = -lavformat -lavcodec -lavutil -lexif -ljpeg -lsqlite3 -lid3tag -lFLAC -logg -lvorbis -lm -lpthread
 #STATIC_LINKING: LIBS = -lvorbis -logg -lm -lsqlite3 -lpthread -lexif -ljpeg -lFLAC -lm -lid3tag -lz -lavformat -lavutil -lavcodec -lm
 
 TESTUPNPDESCGENOBJS = testupnpdescgen.o upnpdescgen.o
@@ -64,7 +67,7 @@
 	$(INSTALL) -d $(ETCINSTALLDIR)
 	$(INSTALL) --mode=0644 minidlna.conf $(ETCINSTALLDIR)
 
-minidlna:	$(BASEOBJS) $(LNXOBJS) $(LIBS)
+minidlna:	$(BASEOBJS) $(LNXOBJS) #$(LIBS)
 	@echo Linking $@
 	@$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(BASEOBJS) $(LNXOBJS) $(LIBS)
 
--- minidlna-1.0.24/genconfig.sh	1970-01-01 01:00:00.000000000 +0100
+++ minidlna-1.0.24/genconfig.sh.patched	2012-05-17 00:47:37.000000000 +0200
@@ -1,4 +1,5 @@
 #! /bin/sh
+# $Id: genconfig.sh,v 1.20 2011/02/17 23:17:24 jmaggard Exp $
 # MiniDLNA project
 # http://sourceforge.net/projects/minidlna/
 #
@@ -29,8 +30,14 @@
 LOG_PATH="${DB_PATH}"
 
 # detecting the OS name and version
-OS_NAME=`uname -s`
-OS_VERSION=`uname -r`
+if [ -z "$BR2_DEFAULT_KERNEL_HEADERS" ]; then
+    OS_NAME=`uname -s`
+    OS_VERSION=`uname -r`
+else
+    OS_NAME=Linux
+    OS_VERSION=$BR2_DEFAULT_KERNEL_HEADERS
+    echo SD=$DESTDIR
+fi
 TIVO="/*#define TIVO_SUPPORT*/"
 NETGEAR="/*#define NETGEAR*/"
 READYNAS="/*#define READYNAS*/"
@@ -38,28 +45,29 @@
 
 ${RM} ${CONFIGFILE}
 
-# Detect if there are missing headers
-# NOTE: This check only works with a normal distro
-[ ! -e "/usr/include/sqlite3.h" ] && MISSING="libsqlite3 $MISSING"
-[ ! -e "/usr/include/jpeglib.h" ] && MISSING="libjpeg $MISSING"
-[ ! -e "/usr/include/libexif/exif-loader.h" ] && MISSING="libexif $MISSING"
-[ ! -e "/usr/include/id3tag.h" ] && MISSING="libid3tag $MISSING"
-[ ! -e "/usr/include/ogg/ogg.h" ] && MISSING="libogg $MISSING"
-[ ! -e "/usr/include/vorbis/codec.h" ] && MISSING="libvorbis $MISSING"
-[ ! -e "/usr/include/FLAC/metadata.h" ] && MISSING="libflac $MISSING"
-[ ! -e "/usr/include/ffmpeg/avutil.h" -a \
-  ! -e "/usr/include/libavutil/avutil.h" -a \
-  ! -e "/usr/include/ffmpeg/libavutil/avutil.h" ] && MISSING="libavutil $MISSING"
-[ ! -e "/usr/include/ffmpeg/avformat.h" -a \
-  ! -e "/usr/include/libavformat/avformat.h" -a \
-  ! -e "/usr/include/ffmpeg/libavformat/avformat.h" ] && MISSING="libavformat $MISSING"
-[ ! -e "/usr/include/ffmpeg/avcodec.h" -a \
-  ! -e "/usr/include/libavcodec/avcodec.h" -a \
-  ! -e "/usr/include/ffmpeg/libavcodec/avcodec.h" ] && MISSING="libavcodec $MISSING"
+## Detect if there are missing headers
+## NOTE: This check only works with a normal distro
+[ ! -e "${DESTDIR}/usr/include/sqlite3.h" ] && MISSING="libsqlite3 $MISSING"
+[ ! -e "${DESTDIR}/usr/include/jpeglib.h" ] && MISSING="libjpeg $MISSING"
+[ ! -e "${DESTDIR}/usr/include/libexif/exif-loader.h" ] && MISSING="libexif $MISSING"
+[ ! -e "${DESTDIR}/usr/include/id3tag.h" ] && MISSING="libid3tag $MISSING"
+[ ! -e "${DESTDIR}/usr/include/ogg/ogg.h" ] && MISSING="libogg $MISSING"
+[ ! -e "${DESTDIR}/usr/include/vorbis/codec.h" ] && MISSING="libvorbis $MISSING"
+[ ! -e "${DESTDIR}/usr/include/FLAC/metadata.h" ] && MISSING="libflac $MISSING"
+[ ! -e "${DESTDIR}/usr/include/ffmpeg/avutil.h" -a \
+  ! -e "${DESTDIR}/usr/include/libavutil/avutil.h" -a \
+  ! -e "${DESTDIR}/usr/include/ffmpeg/libavutil/avutil.h" ] && MISSING="libavutil $MISSING"
+[ ! -e "${DESTDIR}/usr/include/ffmpeg/avformat.h" -a \
+  ! -e "${DESTDIR}/usr/include/libavformat/avformat.h" -a \
+  ! -e "${DESTDIR}/usr/include/ffmpeg/libavformat/avformat.h" ] && MISSING="libavformat $MISSING"
+[ ! -e "${DESTDIR}/usr/include/ffmpeg/avcodec.h" -a \
+  ! -e "${DESTDIR}/usr/include/libavcodec/avcodec.h" -a \
+  ! -e "${DESTDIR}/usr/include/ffmpeg/libavcodec/avcodec.h" ] && MISSING="libavcodec $MISSING"
 if [ -n "$MISSING" ]; then
 	echo -e "\nERROR!  Cannot continue."
 	echo -e "The following required libraries are either missing, or are missing development headers:\n"
 	echo -e "$MISSING\n"
+	echo -e "2222\n"
 	exit 1
 fi
 
@@ -119,7 +127,7 @@
 		KERNVERD=`echo $OS_VERSION | awk -F. '{print $4}'`
 		#echo "$KERNVERA.$KERNVERB.$KERNVERC.$KERNVERD"
 		# NETGEAR ReadyNAS special case
-		if [ -f /etc/raidiator_version ]; then
+		if [ -f "${DESTDIR}/etc/raidiator_version" ]; then
 			OS_NAME=$(awk -F'!!|=' '{ print $1 }' /etc/raidiator_version)
 			OS_VERSION=$(awk -F'!!|[=,.]' '{ print $3"."$4 }' /etc/raidiator_version)
 			OS_URL="http://www.readynas.com/"
@@ -130,7 +138,7 @@
 			READYNAS="#define READYNAS"
 			PNPX="#define PNPX 5"
 		# Debian GNU/Linux special case
-		elif [ -f /etc/debian_version ]; then
+		elif [ -f "${DESTDIR}/etc/debian_version" ]; then
 			OS_NAME=Debian
 			OS_VERSION=`cat /etc/debian_version`
 			OS_URL=http://www.debian.org/
@@ -141,7 +149,7 @@
 				OS_NAME=`${LSB_RELEASE} -i -s`
 				OS_VERSION=`${LSB_RELEASE} -r -s`
 			fi
-		else
+		elif [ -z "${DESTDIR}" ]; then
 			# use lsb_release (Linux Standard Base) when available
 			LSB_RELEASE=`which lsb_release 2>/dev/null`
 			if [ 0 -eq $? ]; then
@@ -175,7 +183,7 @@
 echo "" >> ${CONFIGFILE}
 
 echo "/* Enable if the system inotify.h exists.  Otherwise our own inotify.h will be used. */" >> ${CONFIGFILE}
-if [ -f /usr/include/sys/inotify.h ]; then
+if [ -f ${DESTDIR}/usr/include/sys/inotify.h ]; then
 echo "#define HAVE_INOTIFY_H" >> ${CONFIGFILE}
 else
 echo "/*#define HAVE_INOTIFY_H*/" >> ${CONFIGFILE}
@@ -183,7 +191,7 @@
 echo "" >> ${CONFIGFILE}
 
 echo "/* Enable if the system iconv.h exists.  ID3 tag reading in various character sets will not work properly otherwise. */" >> ${CONFIGFILE}
-if [ -f /usr/include/iconv.h ]; then
+if [ -f ${DESTDIR}/usr/include/iconv.h ]; then
 echo "#define HAVE_ICONV_H" >> ${CONFIGFILE}
 else
 echo -e "\nWARNING!!  Iconv support not found.  ID3 tag reading may not work."
@@ -192,7 +200,7 @@
 echo "" >> ${CONFIGFILE}
 
 echo "/* Enable if the system libintl.h exists for NLS support. */" >> ${CONFIGFILE}
-if [ -f /usr/include/libintl.h ]; then
+if [ -f ${DESTDIR}/usr/include/libintl.h ]; then
 echo "#define ENABLE_NLS" >> ${CONFIGFILE}
 else
 echo "/*#define ENABLE_NLS*/" >> ${CONFIGFILE}
