--- mtd/util/Makefile	2005-02-09 00:00:17.000000000 +0100
+++ mtd/util/Makefile-new	2007-06-16 12:36:58.000000000 +0200
@@ -59,8 +59,8 @@
 
 install: ${TARGETS}
 	mkdir -p ${DESTDIR}/${SBINDIR}
-	install -m0755 -oroot -groot ${TARGETS} ${DESTDIR}/${SBINDIR}/
+	install -m0755 ${TARGETS} ${DESTDIR}/${SBINDIR}/
 	mkdir -p ${DESTDIR}/${MANDIR}/man1
 	gzip -c mkfs.jffs2.1 > ${DESTDIR}/${MANDIR}/man1/mkfs.jffs2.1.gz
 	mkdir -p ${DESTDIR}/${INCLUDEDIR}/mtd
-	install -m0644 -oroot -groot ../include/mtd/*.h ${DESTDIR}/${INCLUDEDIR}/mtd/
+	install -m0644 ../include/mtd/*.h ${DESTDIR}/${INCLUDEDIR}/mtd/
