diff -NEbur '--exclude-from=diff-exclude' neutrino-hd2-exp.org/lib/libdvbsub/dvbsubtitle.cpp neutrino-hd2-exp/lib/libdvbsub/dvbsubtitle.cpp
--- neutrino-hd2-exp.org/lib/libdvbsub/dvbsubtitle.cpp	2013-11-04 08:57:06.603494566 +0100
+++ neutrino-hd2-exp/lib/libdvbsub/dvbsubtitle.cpp	2013-11-04 09:03:25.127510258 +0100
@@ -193,7 +193,7 @@
 		return;
 	}
 
-	avctx = avcodec_alloc_context();
+	avctx = avcodec_alloc_context3(avcodec);
 
 	if (!avctx) 
 	{
@@ -201,7 +201,7 @@
 		return;
 	}
 
-	if (avcodec_open(avctx, avcodec) < 0)
+	if (avcodec_open2(avctx, avcodec, NULL) < 0)
 		dbgconverter("cDvbSubtitleConverter: unable to open codec !\n");
 
 #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(52, 64, 0)
diff -NEbur '--exclude-from=diff-exclude' neutrino-hd2-exp.org/lib/libeplayer3/container/container_ffmpeg.c neutrino-hd2-exp/lib/libeplayer3/container/container_ffmpeg.c
--- neutrino-hd2-exp.org/lib/libeplayer3/container/container_ffmpeg.c	2013-11-04 08:57:06.255494551 +0100
+++ neutrino-hd2-exp/lib/libeplayer3/container/container_ffmpeg.c	2013-11-04 09:03:25.123510258 +0100
@@ -38,7 +38,9 @@
 #include <pthread.h>
 
 #include <libavutil/avutil.h>
+#include <libavutil/time.h>
 #include <libavformat/avformat.h>
+#include <libavutil/opt.h>
 
 #include "common.h"
 #include "misc.h"
@@ -48,6 +50,10 @@
 #include "ffmpeg_metadata.h"
 #include "subtitle.h"
 
+#if LIBAVCODEC_VERSION_MAJOR > 54
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000 // 1 second of 48khz 32bit audio
+#endif
+
 /* ***************************** */
 /* Makros/Constants              */
 /* ***************************** */
@@ -131,9 +137,11 @@
 	ffmpeg_printf(100, "::%d released mutex\n", line);
 }
 
-static char *Codec2Encoding(enum CodecID id, int* version)
+//static char *Codec2Encoding(enum CodecID id, int* version)
+static char* Codec2Encoding(AVCodecContext *codec, int* version)
 {
-	switch (id)
+//	switch (id)
+	switch (codec->codec_id)
 	{
 		case CODEC_ID_MPEG1VIDEO:
 			return "V_MPEG1";
@@ -255,7 +263,8 @@
 #endif 
 
 		default:
-			ffmpeg_err("ERROR! CODEC NOT FOUND -> %d\n",id);
+//			ffmpeg_err("ERROR! CODEC NOT FOUND -> %d\n",id);
+			ffmpeg_err("ERROR! CODEC NOT FOUND -> %d\n",codec->codec_id);
 	}
 	
 	return NULL;
@@ -637,7 +646,7 @@
 							avOut.len        = decoded_data_size;
 
 							avOut.pts        = pts;
-							avOut.extradata  = &extradata;
+							avOut.extradata  = (unsigned char *) &extradata;
 							avOut.extralen   = sizeof(extradata);
 							avOut.frameRate  = 0;
 							avOut.timeScale  = 0;
@@ -805,7 +814,7 @@
 							SubtitleData_t data;
 							data.data      = line;
 							data.len       = strlen((char*)line);
-							data.extradata = DEFAULT_ASS_HEAD;
+							data.extradata = (unsigned char *) DEFAULT_ASS_HEAD;
 							data.extralen  = strlen(DEFAULT_ASS_HEAD);
 							data.pts       = pts;
 							data.duration  = duration;
@@ -901,6 +910,9 @@
 	
 	avContext->flags |= AVFMT_FLAG_GENPTS;
 
+	if ( strstr(filename, ".ts") )
+		avContext->max_analyze_duration = 5;
+
 	ffmpeg_printf(20, "find_streaminfo\n");
 
 #if LIBAVCODEC_VERSION_MAJOR < 54
@@ -940,7 +952,8 @@
 		AVStream * stream = avContext->streams[n];
 		int version = 0;
 
-		char * encoding = Codec2Encoding(stream->codec->codec_id, &version);
+//		char * encoding = Codec2Encoding(stream->codec->codec_id, &version);
+		char * encoding = Codec2Encoding(stream->codec, &version);
 
 		if (encoding != NULL)
 			ffmpeg_printf(1, "%d. encoding = %s - version %d\n", n, encoding, version);
@@ -1089,7 +1102,8 @@
 
 					AVCodec *codec = avcodec_find_decoder(stream->codec->codec_id);
 
-					if(codec != NULL && !avcodec_open(stream->codec, codec))
+//					if(codec != NULL && !avcodec_open(stream->codec, codec))
+					if(codec != NULL && !avcodec_open2(stream->codec, codec, NULL))
 						printf("AVCODEC__INIT__SUCCESS\n");
 					else
 						printf("AVCODEC__INIT__FAILED\n");
Binärdateien neutrino-hd2-exp.org/lib/libeplayer3/eplayer3 und neutrino-hd2-exp/lib/libeplayer3/eplayer3 sind verschieden.
Binärdateien neutrino-hd2-exp.org/lib/libeplayer3/meta und neutrino-hd2-exp/lib/libeplayer3/meta sind verschieden.
diff -NEbur '--exclude-from=diff-exclude' neutrino-hd2-exp.org/src/gui/infoviewer.cpp neutrino-hd2-exp/src/gui/infoviewer.cpp
--- neutrino-hd2-exp.org/src/gui/infoviewer.cpp	2013-11-04 08:57:10.335494720 +0100
+++ neutrino-hd2-exp/src/gui/infoviewer.cpp	2013-11-04 09:06:08.091517014 +0100
@@ -35,6 +35,7 @@
 #include <config.h>
 #endif
 #include <string>
+#include <unistd.h>
 #include <algorithm>
 
 #include <sys/types.h>
diff -NEbur '--exclude-from=diff-exclude' neutrino-hd2-exp.org/src/gui/Makefile.am neutrino-hd2-exp/src/gui/Makefile.am
--- neutrino-hd2-exp.org/src/gui/Makefile.am	2013-11-04 08:57:10.347494721 +0100
+++ neutrino-hd2-exp/src/gui/Makefile.am	2013-11-04 09:03:25.123510258 +0100
@@ -3,7 +3,7 @@
 BUILT_SOURCES  = svn_version.h
 
 svn_version.h:
-	@if test -d .svn ; then \
+	@if test -d ../../.svn ; then \
 		sleep 180 && rm svn_version.h & echo "#define SVNVERSION \"$$(svnversion -n || echo svn_oops!)\" " > svn_version.h ; \
 	else \
 		rm svn_version.h; echo '#define BUILT_DATE "'`date`'"' > svn_version.h ; \
