diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/acinclude.m4 neutrino-hd2/acinclude.m4
--- neutrino-hd2.org/acinclude.m4	2012-07-15 12:45:12.256796577 +0200
+++ neutrino-hd2/acinclude.m4	2012-07-15 12:45:12.516266139 +0200
@@ -225,10 +225,10 @@
 AC_DEFUN([_TUXBOX_APPS_LIB_PKGCONFIG],[
 AC_REQUIRE([TUXBOX_APPS_PKGCONFIG])
 AC_MSG_CHECKING(for package $2)
-if PKG_CONFIG_PATH="${prefix}/lib/pkgconfig" $PKG_CONFIG --exists "$2" ; then
+if $PKG_CONFIG --exists "$2" ; then
 	AC_MSG_RESULT(yes)
-	$1_CFLAGS=$(PKG_CONFIG_PATH="${prefix}/lib/pkgconfig" $PKG_CONFIG --cflags "$2")
-	$1_LIBS=$(PKG_CONFIG_PATH="${prefix}/lib/pkgconfig" $PKG_CONFIG --libs "$2")
+	$1_CFLAGS=$($PKG_CONFIG --cflags "$2")
+	$1_LIBS=$($PKG_CONFIG --libs "$2")
 else
 	AC_MSG_RESULT(no)
 fi
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/lib/libcoolstream/audio_cs.h neutrino-hd2/lib/libcoolstream/audio_cs.h
--- neutrino-hd2.org/lib/libcoolstream/audio_cs.h	2012-07-15 12:45:02.968766743 +0200
+++ neutrino-hd2/lib/libcoolstream/audio_cs.h	2012-07-15 12:45:12.516266139 +0200
@@ -25,7 +25,7 @@
 
 
 #ifdef __sh__
-#include <player2/linux/include/linux/dvb/stm_ioctls.h>
+#include <linux/dvb/stm_ioctls.h>
 #endif
 
 
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/lib/libcoolstream/Makefile.am neutrino-hd2/lib/libcoolstream/Makefile.am
--- neutrino-hd2.org/lib/libcoolstream/Makefile.am	2012-07-15 12:45:02.964766889 +0200
+++ neutrino-hd2/lib/libcoolstream/Makefile.am	2012-07-15 12:45:12.516266139 +0200
@@ -7,7 +7,7 @@
 	-I$(top_srcdir)/src/zapit/include/client \
 	-I$(top_srcdir)/lib/connection \
 	-I$(top_srcdir)/lib/libeventserver \
-	-I$(top_srcdir)/lib/libeplayer3
+	-I$(top_srcdir)/../misc/tools/libeplayer3/include
 
 noinst_LIBRARIES = libcoolstream.a
 
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/lib/libcoolstream/playback_cs.h neutrino-hd2/lib/libcoolstream/playback_cs.h
--- neutrino-hd2.org/lib/libcoolstream/playback_cs.h	2012-07-15 12:45:02.964766889 +0200
+++ neutrino-hd2/lib/libcoolstream/playback_cs.h	2012-07-15 12:45:12.516266139 +0200
@@ -28,8 +28,8 @@
 
 
 #if defined (ENABLE_LIBEPLAYER3)
-#include <libeplayer3/include/common.h>
-#include <libeplayer3/include/subtitle.h>
+#include <common.h>
+#include <subtitle.h>
 #include <linux/fb.h>
 
 
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/lib/libcoolstream/video_cs.h neutrino-hd2/lib/libcoolstream/video_cs.h
--- neutrino-hd2.org/lib/libcoolstream/video_cs.h	2012-07-15 12:45:02.960321929 +0200
+++ neutrino-hd2/lib/libcoolstream/video_cs.h	2012-07-15 12:45:12.516266139 +0200
@@ -25,7 +25,7 @@
 #include <driver/framebuffer.h>
 
 #ifdef __sh__
-#include <player2/linux/include/linux/dvb/stm_ioctls.h>
+#include <linux/dvb/stm_ioctls.h>
 #endif
 
 // bcm
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/lib/Makefile.am neutrino-hd2/lib/Makefile.am
--- neutrino-hd2.org/lib/Makefile.am	2012-07-15 12:45:03.952766840 +0200
+++ neutrino-hd2/lib/Makefile.am	2012-07-15 12:45:12.516266139 +0200
@@ -10,13 +10,12 @@
 	libtuxtxt \
 	libdvbsub \
 	libupnpclient \
-	liblcddisplay \
 	libcoolstream
 
 # libeplayer3
-if ENABLE_LIBEPLAYER3
-SUBDIRS += libeplayer3
-endif
+#if ENABLE_LIBEPLAYER3
+#SUBDIRS += libeplayer3
+#endif
 
 
 
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/driver/framebuffer.h neutrino-hd2/src/driver/framebuffer.h
--- neutrino-hd2.org/src/driver/framebuffer.h	2012-07-15 12:45:08.328266720 +0200
+++ neutrino-hd2/src/driver/framebuffer.h	2012-07-15 12:45:12.516266139 +0200
@@ -36,7 +36,7 @@
 
 // stmfb
 #ifdef __sh__
-#include <stgfb/stmfb/linux/video/stmfb.h>
+#include <linux/stmfb.h>
 #endif
 
 // 32 bit
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/gui/audioplayer.cpp neutrino-hd2/src/gui/audioplayer.cpp
--- neutrino-hd2.org/src/gui/audioplayer.cpp	2012-07-15 12:45:06.984766763 +0200
+++ neutrino-hd2/src/gui/audioplayer.cpp	2012-07-15 12:45:12.520266981 +0200
@@ -126,7 +126,6 @@
 }
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 struct MemoryStruct {
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/gui/filebrowser.cpp neutrino-hd2/src/gui/filebrowser.cpp
--- neutrino-hd2.org/src/gui/filebrowser.cpp	2012-07-15 12:45:06.984766763 +0200
+++ neutrino-hd2/src/gui/filebrowser.cpp	2012-07-15 12:45:12.520266981 +0200
@@ -59,7 +59,6 @@
 #include <sys/stat.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <driver/encoding.h>
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/gui/imageinfo.cpp neutrino-hd2/src/gui/imageinfo.cpp
--- neutrino-hd2.org/src/gui/imageinfo.cpp	2012-07-15 12:45:06.980766980 +0200
+++ neutrino-hd2/src/gui/imageinfo.cpp	2012-07-15 12:45:12.520266981 +0200
@@ -159,7 +159,7 @@
 
 
 	CConfigFile config('\t');
-	config.loadConfig("/.version");
+	config.loadConfig("/var/etc/.version");
 
 	const char * imagename = config.getString("imagename", "NeutrinoHD2").c_str();
 	const char * homepage  = config.getString("homepage",  "http://www.dgstation-forum.org").c_str();
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/gui/Makefile.am neutrino-hd2/src/gui/Makefile.am
--- neutrino-hd2.org/src/gui/Makefile.am	2012-07-15 12:45:06.992767102 +0200
+++ neutrino-hd2/src/gui/Makefile.am	2012-07-15 12:45:12.520266981 +0200
@@ -3,6 +3,7 @@
 SUBDIRS = widget bedit
 
 INCLUDES = \
+  -I$(top_srcdir)/../misc/tools/libeplayer3/include \
 	-I$(top_srcdir)/daemons \
 	-I$(top_srcdir)/src \
 	-I$(top_srcdir)/src/zapit/include \
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/gui/update.cpp neutrino-hd2/src/gui/update.cpp
--- neutrino-hd2.org/src/gui/update.cpp	2012-07-15 12:45:06.984766763 +0200
+++ neutrino-hd2/src/gui/update.cpp	2012-07-15 12:45:12.520266981 +0200
@@ -55,7 +55,6 @@
 #define SQUASHFS
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <stdio.h>
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/Makefile.am neutrino-hd2/src/Makefile.am
--- neutrino-hd2.org/src/Makefile.am	2012-07-15 12:45:08.360305338 +0200
+++ neutrino-hd2/src/Makefile.am	2012-07-15 12:48:47.628266364 +0200
@@ -17,7 +17,7 @@
 	@FREETYPE_CFLAGS@
 
 if ENABLE_LIBEPLAYER3
-INCLUDES += -I$(top_srcdir)/lib/libeplayer3
+INCLUDES += -I$(top_srcdir)/../misc/tools/libeplayer3/include
 endif
 
 if ENABLE_TEST_MENU
@@ -71,7 +71,7 @@
 	-ljpeg -lgif -lrt -lpthread -lblkid -ldl -lFLAC -lavformat -lavcodec -lavutil
 
 if ENABLE_LIBEPLAYER3
-neutrino_LDADD += $(top_builddir)/lib/libeplayer3/libeplayer3.a
+neutrino_LDADD += -leplayer3
 endif
 
 if ENABLE_LIBASS
@@ -79,8 +79,8 @@
 endif
 
 # LCD display
-neutrino_LDADD += \
-	$(top_builddir)/lib/liblcddisplay/liblcddisplay.a
+#neutrino_LDADD += \
+#	$(top_builddir)/lib/liblcddisplay/liblcddisplay.a
 
 
 
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/nhttpd/tuxboxapi/neutrinoapi.cpp neutrino-hd2/src/nhttpd/tuxboxapi/neutrinoapi.cpp
--- neutrino-hd2.org/src/nhttpd/tuxboxapi/neutrinoapi.cpp	2012-07-15 12:45:04.036767781 +0200
+++ neutrino-hd2/src/nhttpd/tuxboxapi/neutrinoapi.cpp	2012-07-15 12:45:12.520266981 +0200
@@ -64,7 +64,7 @@
 bool _initialize_iso639_map(void)
 {
 	std::string s, t, u, v;
-	std::ifstream in("/share/iso-codes/iso-639.tab");
+	std::ifstream in("/usr/local/share/iso-codes/iso-639.tab");
 	if (in.is_open())
 	{
 		while (in.peek() == '#')
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/nhttpd/web/scripts/Makefile.am neutrino-hd2/src/nhttpd/web/scripts/Makefile.am
--- neutrino-hd2.org/src/nhttpd/web/scripts/Makefile.am	2012-07-15 12:45:04.480766781 +0200
+++ neutrino-hd2/src/nhttpd/web/scripts/Makefile.am	2012-07-15 12:45:12.520266981 +0200
@@ -2,6 +2,3 @@
 
 install_DATA= api.sh _Y_Globals.sh _Y_Library.sh Y_Live.sh Y_Tools.sh
 
-install-data-hook:
-	chmod 0755 $(DATADIR)/neutrino/httpd/scripts/api.sh
-	chmod 0755 $(DATADIR)/neutrino/httpd/scripts/Y_*.sh
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/system/httptool.cpp neutrino-hd2/src/system/httptool.cpp
--- neutrino-hd2.org/src/system/httptool.cpp	2012-07-15 12:45:07.220767182 +0200
+++ neutrino-hd2/src/system/httptool.cpp	2012-07-15 12:45:12.520266981 +0200
@@ -22,7 +22,6 @@
 #include <system/httptool.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <global.h>
diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/system/localize.cpp neutrino-hd2/src/system/localize.cpp
--- neutrino-hd2.org/src/system/localize.cpp	2012-07-15 12:45:07.216768368 +0200
+++ neutrino-hd2/src/system/localize.cpp	2012-07-15 12:45:12.520266981 +0200
@@ -45,7 +45,7 @@
 #include <string>
 
 
-static const char * iso639filename = "/share/iso-codes/iso-639.tab";
+static const char * iso639filename = "/usr/local/share/iso-codes/iso-639.tab";
 
 
 #if 0
