diff -ur --exclude=.svn --exclude=.git --exclude=tuxtxt neutrino-hd2.org/src/gui/eventlist.cpp neutrino-hd2/src/gui/eventlist.cpp
--- neutrino-hd2.org/src/gui/eventlist.cpp	2012-07-09 17:19:01.688766972 +0200
+++ neutrino-hd2/src/gui/eventlist.cpp	2012-07-09 20:39:42.972266519 +0200
@@ -35,6 +35,7 @@
 
 #include <global.h>
 #include <gui/eventlist.h>
+#include <gui/epgplus.h>
 #include <gui/timerlist.h>
 
 #include <gui/widget/icons.h>
@@ -528,7 +529,14 @@
 				loop = false;
 			}
 		}
-		else if ( msg==CRCInput::RC_left || msg==CRCInput::RC_red )		  
+		else if (msg == CRCInput::RC_epg)
+	  {
+	    hide();
+	    CEPGplusHandler eplus;
+	    eplus.exec(NULL, "");
+	    loop = false;
+	  }
+    else if ( msg==CRCInput::RC_left || msg==CRCInput::RC_red )		  
 		{
 			loop = false;
 		}