diff --git a/include/ntp_machine.h b/include/ntp_machine.h
index 69dca68..91d233a 100644
--- a/include/ntp_machine.h
+++ b/include/ntp_machine.h
@@ -20,6 +20,9 @@
 # endif
 #endif
 
+#ifndef MOD_NANO
+#define MOD_NANO ADJ_NANO
+#endif
 #include "ntp_proto.h"
 
 /*
diff --git a/scripts/mkver b/scripts/mkver
index aa5f371..2c2fcd9 100755
--- a/scripts/mkver.in
+++ b/scripts/mkver.in
@@ -17,7 +17,7 @@ case "" in
  *) ConfStr="${ConfStr}-?" ;;
 esac
 
-ConfStr="$ConfStr `LC_TIME=C TZ=UTC date`"
+ConfStr="$ConfStr"
 
 if [ ! -f .version ]; then
   echo 0 > .version
