diff -Nur '--exclude-from=diff-ignore' python-2.7.3.org/Modules/_ctypes/libffi/configure.ac python-2.7.3/Modules/_ctypes/libffi/configure.ac
--- python-2.7.3.org/Modules/_ctypes/libffi/configure.ac	2012-04-10 01:07:33.000000000 +0200
+++ python-2.7.3/Modules/_ctypes/libffi/configure.ac	2012-11-16 20:14:43.770806447 +0100
@@ -409,7 +409,7 @@
 
 AC_CONFIG_LINKS(include/ffitarget.h:src/$TARGETDIR/ffitarget.h)
 
-AC_CONFIG_FILES(include/Makefile include/ffi.h Makefile testsuite/Makefile man/Makefile libffi.pc)
+AC_CONFIG_FILES(include/Makefile include/ffi.h Makefile )
 
 AC_CONFIG_LINKS(include/ffi_common.h:include/ffi_common.h)
 
diff -Nur '--exclude-from=diff-ignore' python-2.7.3.org/Modules/_ctypes/libffi/Makefile.am python-2.7.3/Modules/_ctypes/libffi/Makefile.am
--- python-2.7.3.org/Modules/_ctypes/libffi/Makefile.am	2012-04-10 01:07:33.000000000 +0200
+++ python-2.7.3/Modules/_ctypes/libffi/Makefile.am	2012-11-16 20:14:43.770806447 +0100
@@ -2,7 +2,7 @@
 
 AUTOMAKE_OPTIONS = foreign subdir-objects
 
-SUBDIRS = include testsuite man
+SUBDIRS = include
 
 EXTRA_DIST = LICENSE ChangeLog.v1 ChangeLog.libgcj configure.host \
 	src/alpha/ffi.c src/alpha/osf.S src/alpha/ffitarget.h \
@@ -34,8 +34,6 @@
 	libtool-version ChangeLog.libffi m4/libtool.m4 \
 	m4/lt~obsolete.m4 m4/ltoptions.m4 m4/ltsugar.m4 m4/ltversion.m4
 
-info_TEXINFOS = doc/libffi.texi
-
 ## ################################################################
 
 ##
@@ -81,8 +79,6 @@
 
 MAKEOVERRIDES=
 
-ACLOCAL_AMFLAGS=$(ACLOCAL_AMFLAGS) -I m4
-
 lib_LTLIBRARIES = libffi.la
 noinst_LTLIBRARIES = libffi_convenience.la
 
--- python-2.7.3.org/Modules/_ctypes/libffi/fficonfig.py.in	2010-01-16 19:05:15.000000000 +0100
+++ python-2.7.3/Modules/_ctypes/libffi/fficonfig.py.in	2012-04-09 19:38:57.000000000 +0200
@@ -30,4 +30,4 @@
 
 ffi_sources += ffi_platforms['@TARGET@']
 
-ffi_cflags = '@CFLAGS@'
+ffi_cflags = '@CFLAGS@ -DPIC'
