--- SPECS/stm-target-tar.spec	2006-12-22 02:00:01.000000000 +0100
+++ SPECS/stm-target-tar.spec-new	2007-06-11 21:05:12.000000000 +0200
@@ -1,5 +1,5 @@
 Summary: GNU tar
-Name: %{_stm_pkg_prefix}-target-tar
+Name: %{_hhl_pkg_prefix}-target-tar
 Version: 1.16.1
 Release: 7
 Source0: tar-%{version}.tar.bz2
@@ -8,15 +8,15 @@
 License: GPL
 Group: Applications/Archiving
 Buildroot: %{_tmppath}/%{_pkgname}-root
-%define targetroot %{_stm_cross_target_dir}
+%define targetroot %{_hhl_cross_target_dir}
 Prefix: %{targetroot}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-tar
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-tar
 
-%define _bindir		%{targetroot}%{_stm_target_bin_dir}
-%define _datadir	%{targetroot}%{_stm_target_sharedstate_dir}
-%define _infodir	%{targetroot}%{_stm_target_info_dir}
-%define _mandir		%{targetroot}%{_stm_target_man_dir}
-%define _docdir		%{targetroot}%{_stm_target_doc_dir}
+%define _bindir		%{targetroot}%{_hhl_target_bin_dir}
+%define _datadir	%{targetroot}%{_hhl_target_sharedstate_dir}
+%define _infodir	%{targetroot}%{_hhl_target_info_dir}
+%define _mandir		%{targetroot}%{_hhl_target_man_dir}
+%define _docdir		%{targetroot}%{_hhl_target_doc_dir}
 %define _rbindir	%{targetroot}/bin
 
 %description
@@ -95,6 +95,6 @@
 * Wed Dec 29 2004 Stuart Menefy <stuart.menefy@st.com>
 - Remove info dir file.
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Sat Jan 26 2002 Stuart Menefy <stuart.menefy@st.com>
 - Copied from MontaVista version 1.13.19-hhl2.0.2
