--- SPECS/stm-host-autotools.spec.orig	2010-05-15 07:33:36.000000000 +0200
+++ SPECS/stm-host-autotools.spec	2010-05-15 07:34:55.000000000 +0200
@@ -6,14 +6,11 @@
 Group: Development/Base
 Source0: http://ftp.gnu.org/gnu/autoconf/autoconf-2.59.tar.bz2
 Source1: http://ftp.gnu.org/gnu/automake/automake-1.8.5.tar.bz2
-Source2: http://ftp.gnu.org/gnu/libtool/libtool-1.5.10.tar.gz
 Source3: http://www.freedesktop.org/software/pkgconfig/releases/pkgconfig-0.21.tar.gz
 Patch0: automake-1.8.5-stm.patch
 Patch1: automake-1.8.5-nodefault_path.patch
 Patch2: autoconf-2.59-stm.patch
 Patch3: autoconf-2.59-nodefault_path.patch
-Patch20: libtool-1.5.10-stm.patch
-Patch21: libtool-1.5.10-path-order.patch
 Patch30: pkgconfig-0.21-cross.patch
 
 Requires	: %{_stm_pkg_prefix}-host-filesystem
@@ -34,12 +31,6 @@
  - Automake is a tool for automatically generating
    `Makefile.in' files compliant with the GNU Coding Standards.
 
- - The libtool package contains the GNU libtool, a set of shell scripts
-   which automatically configure UNIX and UNIX-like architectures to
-   generically build shared libraries.  Libtool provides a consistent,
-   portable interface which simplifies the process of using shared
-   libraries.
-
  - The pkgconfig package makes putting together all the build
    flags when compiling/linking a lot easier.
 
@@ -53,13 +44,11 @@
 to select the corect version of the tools as appropriate.
 
 %prep
-%setup -q -n autotools-%{version} -c -a0 -a 1 -a 2 -a 3
+%setup -q -n autotools-%{version} -c -a0 -a 1 -a 3
 patch -d automake-1.8.5 -p1 < %PATCH0
 patch -d automake-1.8.5 -p1 < %PATCH1
 patch -d autoconf-2.59 -p1 < %PATCH2
 patch -d autoconf-2.59 -p1 < %PATCH3
-patch -d libtool-1.5.10 -p1 < %PATCH20
-patch -d libtool-1.5.10 -p1 < %PATCH21
 patch -d pkg-config-0.21 -p1 < %PATCH30
 
 %build
@@ -92,7 +81,7 @@
 make install
 popd
 
-for d in autoconf-2.59 automake-1.8.5 libtool-1.5.10 ; do
+for d in autoconf-2.59 automake-1.8.5 ; do
   cd $d
   ./configure --prefix=%{_stm_host_dir}
   %make
@@ -111,7 +100,7 @@
 %install
 %host_setup
 
-for d in autoconf-2.59 automake-1.8.5 libtool-1.5.10 pkg-config-0.21; do
+for d in autoconf-2.59 automake-1.8.5 pkg-config-0.21; do
   cd $d
   %make prefix=%{buildroot}%{_stm_host_dir} install
   cd ..
@@ -127,10 +116,7 @@
 %compress_man %{buildroot}%{_stm_host_dir}/man
 
 # Remove unnecessary files
-rm -f %{buildroot}%{_stm_host_dir}/include/ltdl.h
 rm -f %{buildroot}%{_stm_host_dir}/info/*
-rm -f %{buildroot}%{_stm_host_dir}/lib/libltdl.a
-rm -f %{buildroot}%{_stm_host_dir}/lib/libltdl.la
 
 %clean
 rm -rf %{buildroot}
@@ -144,7 +130,6 @@
 %{_stm_host_dir}/man/man1/*.gz
 %{_stm_host_dir}/share/aclocal/pkg.m4
 %{_stm_host_dir}/lib/pkgconfig
-%{_stm_host_dir}/lib/libltdl.so*
 
 %changelog
 * Tue Nov 11 2008 Stuart Menefy <stuart.menefy@st.com> 14
