--- SPECS/stm-cross-binutils.spec.orig	2009-06-06 01:05:24.000000000 +0200
+++ SPECS/stm-cross-binutils.spec	2010-04-05 09:26:59.911267525 +0200
@@ -46,9 +46,6 @@
 binary and object files.  They may be used in conjunction with a compiler
 and various libraries to build programs for Linux.
 
-%if ! %{nohostbuilddeps}
-BuildRequires: autoconf, bison, flex, gettext, texinfo
-%endif
 BuildRequires: %{_stm_pkg_prefix}-host-rpmconfig
 
 # Cross Package
@@ -126,6 +123,7 @@
 %cross_configure \
        --enable-nls \
        --disable-multilib \
+       --disable-werror \
        --program-prefix=%{_stm_target_toolprefix} \
 %if_target_cpu st200
        --with-float=soft \
