--- SPECS/stm-target-strace.spec	2006-10-12 22:14:35.000000000 +0200
+++ SPECS/stm-target-strace.spec-new	2007-07-21 17:46:49.000000000 +0200
@@ -1,5 +1,5 @@
 Summary: system call tracer
-Name: %{_stm_pkg_prefix}-target-strace
+Name: %{_hhl_pkg_prefix}-target-strace
 Version: 4.5.14
 Release: 10
 Source: http://prdownloads.sourceforge.net/strace/strace-%{version}.tar.bz2
@@ -16,10 +16,10 @@
 Patch21: strace-4.5.14-st200.patch
 License: BSD
 Group: Development/Debuggers
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir		%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-strace
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir		%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-strace
 
 %description
 strace is a system call tracer, i.e. a debugging tool which prints out
@@ -66,25 +66,25 @@
 %install
 %target_setup
 rm -rf %{buildroot}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man1
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1
 %target_makeinstall
 
-gzip -9 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/strace.1
+gzip -9 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/strace.1
 cd ..
 cp COPYRIGHT LICENSE
 
 # Remove perl files
-rm -f %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}/strace-graph
+rm -f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/strace-graph
 
 %clean
 rm -rf %{buildroot}
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/strace
-# perl %{_stm_cross_target_dir}%{_stm_target_bin_dir}/strace-graph
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/strace.1.gz
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/strace
+# perl %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/strace-graph
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/strace.1.gz
 %doc ChangeLog NEWS TODO COPYRIGHT CREDITS LICENSE
 
 %changelog
@@ -113,7 +113,7 @@
 - Added ST200 support
 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Mon Sep 06 2004 Stuart Menefy <stuart.menefy@st.com>
 - Updated to 4.5.7
 
