--- SPECS/stm-host-gettext.spec	2007-06-21 11:29:33.000000000 +0200
+++ SPECS/stm-host-gettext.spec-new	2007-06-23 12:22:18.000000000 +0200
@@ -58,16 +58,17 @@
 rm -f $RPM_BUILD_ROOT%{_hhl_host_lib_dir}/*.a
 rm -rf $RPM_BUILD_ROOT%{_hhl_host_data_dir}/doc
 rm -rf $RPM_BUILD_ROOT%{_hhl_host_data_dir}/emacs
+rm -rf $RPM_BUILD_ROOT%{_hhl_host_lib_dir}/GNU.Gettext.dll
 
 %clean
 rm -rf %{buildroot}
 
 %post
-/sbin/install-info %{_hhl_host_info_dir}/gettext.info.gz %{_hhl_host_info_dir}/dir
+install-info --quiet --dir-file=%{_hhl_host_info_dir}/dir %{_hhl_host_info_dir}/gettext.info.gz
 
 %preun
 if [ "$1" = 0 ]; then
-    /sbin/install-info --delete %{_hhl_host_info_dir}/gettext.info.gz %{_hhl_host_info_dir}/dir
+    install-info --quiet --remove --dir-file=%{_hhl_host_info_dir}/dir %{_hhl_host_info_dir}/gettext.info.gz
 fi
 exit 0
 
