--- SPECS/stm-host-stimage.spec	2005-09-23 22:30:08.000000000 +0200
+++ SPECS/stm-host-stimage.spec-new	2007-06-16 13:17:07.000000000 +0200
@@ -28,13 +28,13 @@
 rm -rf %{buildroot}
 %target_setup
 mkdir -p %{buildroot}%{_hhl_devkit_dir}/sh4/bin
-mkdir -p %{buildroot}/usr/share/man/man8
+mkdir -p %{buildroot}%{_hhl_devkit_dir}/sh4/man/man8
 cp hex2img %{buildroot}%{_hhl_devkit_dir}/sh4/bin
 cp hex2raw %{buildroot}%{_hhl_devkit_dir}/sh4/bin
 cp elf2img %{buildroot}%{_hhl_devkit_dir}/sh4/bin
-cp man/elf2img.8.gz %{buildroot}/usr/share/man/man8/
-cp man/hex2img.8.gz %{buildroot}/usr/share/man/man8/
-cp man/hex2raw.8.gz %{buildroot}/usr/share/man/man8/
+cp man/elf2img.8.gz %{buildroot}%{_hhl_devkit_dir}/sh4/man/man8/
+cp man/hex2img.8.gz %{buildroot}%{_hhl_devkit_dir}/sh4/man/man8/
+cp man/hex2raw.8.gz %{buildroot}%{_hhl_devkit_dir}/sh4/man/man8/
 
 %clean
 rm -rf %{buildroot}
@@ -42,7 +42,7 @@
 %files
 %defattr(-,root,root)
 %{_hhl_devkit_dir}/sh4/bin
-/usr/share/man/man8
+%{_hhl_devkit_dir}/sh4/man/man8
 
 %changelog
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
