--- SPECS/stm-target-file.spec	2006-10-12 22:14:34.000000000 +0200
+++ SPECS/stm-target-file.spec-new	2007-06-16 13:13:40.000000000 +0200
@@ -1,5 +1,5 @@
 Summary   	: A utility for determining file types.
-Name	  	: %{_stm_pkg_prefix}-target-file
+Name	  	: %{_hhl_pkg_prefix}-target-file
 Version	  	: 4.17
 Release 	: 3
 Source	  	: file-%{version}.tar.gz
@@ -15,11 +15,11 @@
 Patch10		: file-4.17-mp3_flac.patch
 License 	: Other
 Group	  	: Applications/File
-Buildroot 	: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix	  	: %{_stm_cross_target_dir}
+Buildroot 	: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix	  	: %{_hhl_cross_target_dir}
 
-%define _defaultdocdir	%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-file
+%define _defaultdocdir	%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-file
 
 #
 #	SRPM
@@ -77,12 +77,12 @@
 
 %install
 %target_setup
-PATH=%{_stm_cross_bin_dir}:$PATH
+PATH=%{_hhl_cross_bin_dir}:$PATH
 rm -rf %{buildroot}
 
 %target_makeinstall
 
-%compress_man  %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
+%compress_man  %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}
 
 cd ..
 cp LEGAL.NOTICE LICENSE
@@ -92,22 +92,22 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/file
-%dir %{_stm_cross_target_dir}%{_stm_target_sharedstate_dir}/file
-%{_stm_cross_target_dir}%{_stm_target_sharedstate_dir}/file/magic
-%{_stm_cross_target_dir}%{_stm_target_sharedstate_dir}/file/magic.mgc
-%{_stm_cross_target_dir}%{_stm_target_sharedstate_dir}/file/magic.mime
-%{_stm_cross_target_dir}%{_stm_target_sharedstate_dir}/file/magic.mime.mgc
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libmagic.a
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libmagic.la
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libmagic.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libmagic.so.1
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libmagic.so.1.0.0
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/magic.h
-
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/file.1.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man3/libmagic.3.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man4/magic.4.gz
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/file
+%dir %{_hhl_cross_target_dir}%{_hhl_target_sharedstate_dir}/file
+%{_hhl_cross_target_dir}%{_hhl_target_sharedstate_dir}/file/magic
+%{_hhl_cross_target_dir}%{_hhl_target_sharedstate_dir}/file/magic.mgc
+%{_hhl_cross_target_dir}%{_hhl_target_sharedstate_dir}/file/magic.mime
+%{_hhl_cross_target_dir}%{_hhl_target_sharedstate_dir}/file/magic.mime.mgc
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libmagic.a
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libmagic.la
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libmagic.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libmagic.so.1
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libmagic.so.1.0.0
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/magic.h
+
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/file.1.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3/libmagic.3.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man4/magic.4.gz
 %doc LICENSE README
 
 %changelog
@@ -118,6 +118,6 @@
 - Automatically removed dependency opt-out line
 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Sat Jan 26 2002 Stuart Menefy <stuart.menefy@st.com>
 - Copied from MontaVista version 3.28-hhl2.0.2
