--- SPECS/stm-target-distributionutils.spec	2008-10-31 19:06:17.000000000 +0100
+++ SPECS/stm-target-distributionutils.spec-new	2008-10-31 19:12:17.000000000 +0100
@@ -32,7 +32,7 @@
 This package includes the documentation for distributionutils.
 
 %prep
-%setup -q -a 1 -a 2 -a 3 -a 4 -n debianutils-%{version}
+%setup -q -a 1 -a 2 -n debianutils-%{version}
 %autoremake
 
 %build
@@ -49,18 +49,8 @@
 %{_stm_target_toolprefix}gcc -DTARGETDIR=\"\" -o initdconfig initdconfig.c
 cd ..
 
-cd start-stop-daemon
-CC=%{_stm_target_toolprefix}gcc make
-cd ..
-
-cd usleep
-%{_stm_target_toolprefix}gcc -o usleep usleep.c
-cd ..
-
 %install
-%target_setup
-%target_makeinstall
-cd ..
+install -d %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
 
 install -m 755 shellconfig/shellconfig %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/.
 ln -s shellconfig %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/%{_stm_target_toolprefix}shellconfig
@@ -68,31 +58,6 @@
 install -m 755 initdconfig/initdconfig %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/.
 ln -s initdconfig %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/%{_stm_target_toolprefix}initdconfig
 
-mkdir -p %{buildroot}%{_stm_cross_target_dir}/sbin
-install -m 755 start-stop-daemon/start-stop-daemon %{buildroot}%{_stm_cross_target_dir}/sbin/.
-install -m 644 start-stop-daemon/start-stop-daemon.8 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-
-mkdir -p %{buildroot}%{_stm_cross_target_dir}/bin
-install -m 755 usleep/usleep %{buildroot}%{_stm_cross_target_dir}/bin/.
-
-# Remove files we install but don't distribute
-rm %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}/which
-rm %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/which.1
-rm %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/fr/man1/which.1
-
-rm %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/installkernel
-rm %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/mkboot
-find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir} -name "installkernel*" | xargs rm
-find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir} -name "mkboot*" | xargs rm
-
-rm %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/add-shell
-rm %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/remove-shell
-find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir} -name "add-shell*" | xargs rm
-find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir} -name "remove-shell*" | xargs rm
-
-rm %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}/sensible-*
-find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir} -name "sensible-*" | xargs rm
-
 %target_install_fixup
 
 cp debian/copyright LICENSE
@@ -102,12 +67,6 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}/bin/usleep
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/run-parts
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/tempfile
-%{_stm_cross_target_dir}/sbin/start-stop-daemon
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/savelog
-
 %{_stm_cross_target_dir}%{_stm_target_sbin_dir}/shellconfig
 %{_stm_cross_target_dir}%{_stm_target_sbin_dir}/%{_stm_target_toolprefix}shellconfig
 
@@ -117,13 +76,6 @@
 
 %files -n %{_pkgname}-doc
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/tempfile.1.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/run-parts.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/savelog.8.gz
-%doc %lang(fr) %{_stm_cross_target_dir}%{_stm_target_man_dir}/fr/man1/tempfile.1.gz
-%doc %lang(fr) %{_stm_cross_target_dir}%{_stm_target_man_dir}/fr/man8/run-parts.8.gz
-%doc %lang(fr) %{_stm_cross_target_dir}%{_stm_target_man_dir}/fr/man8/savelog.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/start-stop-daemon.8.gz
 %doc LICENSE
 
 %changelog
