--- SPECS/stm-target-libtermcap.spec	2005-09-23 22:30:09.000000000 +0200
+++ SPECS/stm-target-libtermcap.spec-new	2007-06-16 12:11:50.000000000 +0200
@@ -1,7 +1,7 @@
 Name: %{_hhl_pkg_prefix}-target-libtermcap
 Version: 2.0.8
 Release: 8
-Copyright: LGPL
+License: LGPL
 Source: ftp://sunsite.unc.edu/pub/Linux/GCC/termcap-2.0.8.tar.gz
 Patch0: termcap-2.0.8-shared.patch
 Patch1: termcap-2.0.8-setuid.patch
