--- a/gcc/Makefile.in.org	2011-11-02 10:34:37.000000000 +0100
+++ b/gcc/Makefile.in	2013-07-03 17:12:06.658799684 +0200
@@ -4534,7 +4534,7 @@
 # We keep the directory structure for files in config and .def files. All
 # other files are flattened to a single directory.
 	$(mkinstalldirs) $(DESTDIR)$(plugin_includedir)
-	headers=`echo $(PLUGIN_HEADERS) | tr ' ' '\012' | sort -u`; \
+	headers="$(sort $(PLUGIN_HEADERS))"; \
 	srcdirstrip=`echo "$(srcdir)" | sed 's/[].[^$$\\*|]/\\\\&/g'`; \
 	for file in $$headers; do \
 	  if [ -f $$file ] ; then \
