--- SPECS/stm-cross-binutils-sh4processed.spec	2006-10-12 22:27:32.000000000 +0200
+++ SPECS/stm-cross-binutils-sh4processed.spec-new	2007-06-16 16:13:02.000000000 +0200
@@ -3,15 +3,15 @@
 Version: %{binutils_version}
 Release: 13
 License: GPL
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_install_prefix}
-%define _docdir %{_stm_cross_doc_dir}
-BuildArch: %{_stm_host_arch}
-%define _pkgname %{_stm_pkg_prefix}-cross-%{_stm_target_arch}-binutils
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_install_prefix}
+%define _docdir %{_hhl_cross_doc_dir}
+BuildArch: %{_hhl_host_arch}
+%define _pkgname %{_hhl_pkg_prefix}-cross-%{_hhl_target_arch}-binutils
 
 # Source Package
 Summary: The GNU assembler, linker and binary utilities.
-Name: %{_stm_pkg_prefix}-cross-binutils
+Name: %{_hhl_pkg_prefix}-cross-binutils
 Source0: binutils-%{binutils_version}.tar.bz2
 
 Group: Development/Languages 
@@ -21,14 +21,14 @@
 and various libraries to build programs for Linux.
 
 BuildRequires: autoconf, bison, flex, gettext, texinfo
-BuildRequires: %{_stm_pkg_prefix}-host-rpmconfig
+BuildRequires: %{_hhl_pkg_prefix}-host-rpmconfig
 
 # Cross Package
 %package -n %{_pkgname}
-Summary: The GNU assembler, linker and binary utilities for the %{_stm_target_arch}.
+Summary: The GNU assembler, linker and binary utilities for the %{_hhl_target_arch}.
 Group: Development/Languages
-Requires: %{_stm_pkg_prefix}-host-filesystem,%{_stm_pkg_prefix}-cross-%{_stm_target_arch}-filesystem
-Provides: %{_stm_pkg_prefix}-generic-%{_stm_target_arch}-binutils
+Requires: %{_hhl_pkg_prefix}-host-filesystem,%{_hhl_pkg_prefix}-cross-%{_hhl_target_arch}-filesystem
+Provides: %{_hhl_pkg_prefix}-generic-%{_hhl_target_arch}-binutils
 %description -n %{_pkgname}
 The programs in this package are used to assemble, link and manipulate
 binary and object files.  They may be used in conjunction with a compiler
@@ -76,7 +76,7 @@
        --enable-nls \
        --disable-multilib \
        --enable-install-libbfd \
-       --with-sysroot=%{_stm_cross_target_dir}
+       --with-sysroot=%{_hhl_cross_target_dir}
 make
 
 %install
@@ -84,38 +84,38 @@
 rm -rf %{buildroot}
 srcdir=$PWD
 
-mkdir -p %{buildroot}%{_stm_cross_dir}/include
-mkdir -p %{buildroot}%{_stm_cross_dir}/lib
+mkdir -p %{buildroot}%{_hhl_cross_dir}/include
+mkdir -p %{buildroot}%{_hhl_cross_dir}/lib
 
 cd %{_objdir}
 %cross_makeinstall install-info
 make -C libiberty \
-        prefix=%{buildroot}%{_stm_cross_dir} \
+        prefix=%{buildroot}%{_hhl_cross_dir} \
         target_header_dir=libiberty \
         install_to_libdir
 
 # need to add some links to keep the st231 compiler happy
 
 # Remove unnecessary info pages
-rm -f %{buildroot}%{_stm_cross_info_dir}/dir
-rm %{buildroot}%{_stm_cross_info_dir}/{configure,standards}.*
+rm -f %{buildroot}%{_hhl_cross_info_dir}/dir
+rm %{buildroot}%{_hhl_cross_info_dir}/{configure,standards}.*
 
 # Remove unnecessary man pages
-rm %{buildroot}%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}dlltool.1
-rm %{buildroot}%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}nlmconv.1
-rm %{buildroot}%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}windres.1
+rm %{buildroot}%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}dlltool.1
+rm %{buildroot}%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}nlmconv.1
+rm %{buildroot}%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}windres.1
 
 # Add missing links to tools
-mkdir -p %{buildroot}%{_stm_cross_targetconf_dir}/bin
+mkdir -p %{buildroot}%{_hhl_cross_targetconf_dir}/bin
 for tool in addr2line c++filt objcopy objdump readelf size strings ; do
-  ln -f %{buildroot}%{_stm_cross_bin_dir}/%{_stm_target_toolprefix}${tool} \
-    %{buildroot}%{_stm_cross_targetconf_dir}/bin/${tool}
+  ln -f %{buildroot}%{_hhl_cross_bin_dir}/%{_hhl_target_toolprefix}${tool} \
+    %{buildroot}%{_hhl_cross_targetconf_dir}/bin/${tool}
 done
 
 # --enable-install-libbfd puts the files in an odd place when cross compiling
-mv %{buildroot}/%{_stm_cross_dir}/%{_stm_host_config}/%{_stm_target_config}/include/* %{buildroot}%{_stm_cross_include_dir}
-mv %{buildroot}/%{_stm_cross_dir}/%{_stm_host_config}/%{_stm_target_config}/lib/*     %{buildroot}%{_stm_cross_lib_dir}
-rm -rf %{buildroot}/%{_stm_cross_dir}/%{_stm_host_config}
+mv %{buildroot}/%{_hhl_cross_dir}/%{_hhl_host_config}/%{_hhl_target_config}/include/* %{buildroot}%{_hhl_cross_include_dir}
+mv %{buildroot}/%{_hhl_cross_dir}/%{_hhl_host_config}/%{_hhl_target_config}/lib/*     %{buildroot}%{_hhl_cross_lib_dir}
+rm -rf %{buildroot}/%{_hhl_cross_dir}/%{_hhl_host_config}
 
 cd $srcdir
 %find_lang binutils
@@ -132,61 +132,61 @@
 cat gprof.lang >> binutils.lang
 
 #  Remove some gprof info files not required
-rm -f %{buildroot}/%{_stm_cross_man_dir}/man1/st231-linux-gprof.1
+rm -f %{buildroot}/%{_hhl_cross_man_dir}/man1/st231-linux-gprof.1
 
 %clean
 rm -rf %{buildroot}
 
 %files -n %{_pkgname} -f binutils.lang
 %defattr(-,root,root)
-%{_stm_cross_bin_dir}/*
-%{_stm_cross_targetconf_dir}/bin/*
-%dir %{_stm_cross_targetconf_dir}/lib/ldscripts
-%{_stm_cross_targetconf_dir}/lib/ldscripts/*
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}addr2line.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}ar.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}as.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}c++filt.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}ld.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}nm.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}objcopy.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}objdump.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}ranlib.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}readelf.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}size.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}strings.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}strip.1
-%{_stm_cross_man_dir}/man1/%{_stm_target_toolprefix}gprof.1
+%{_hhl_cross_bin_dir}/*
+%{_hhl_cross_targetconf_dir}/bin/*
+%dir %{_hhl_cross_targetconf_dir}/lib/ldscripts
+%{_hhl_cross_targetconf_dir}/lib/ldscripts/*
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}addr2line.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}ar.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}as.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}c++filt.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}ld.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}nm.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}objcopy.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}objdump.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}ranlib.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}readelf.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}size.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}strings.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}strip.1
+%{_hhl_cross_man_dir}/man1/%{_hhl_target_toolprefix}gprof.1
 %doc README COPYING COPYING.LIB
 
 %files -n %{_pkgname}-dev
 %defattr(-,root,root)
-%{_stm_cross_lib_dir}/*
-%{_stm_cross_include_dir}/*
-%doc %{_stm_cross_info_dir}/bfd.info*
+%{_hhl_cross_lib_dir}/*
+%{_hhl_cross_include_dir}/*
+%doc %{_hhl_cross_info_dir}/bfd.info*
 
 %files -n %{_pkgname}-doc
 %defattr(-,root,root)
-%doc %{_stm_cross_info_dir}/*
+%doc %{_hhl_cross_info_dir}/*
 
 %post -n %{_pkgname}-dev
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir \
-	%{_stm_cross_info_dir}/bfd.info
+install-info --quiet --dir-file=%{_hhl_cross_info_dir}/dir \
+	%{_hhl_cross_info_dir}/bfd.info
 
 %preun -n %{_pkgname}-dev
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir --remove \
-	%{_stm_cross_info_dir}/bfd.info
+install-info --quiet --dir-file=%{_hhl_cross_info_dir}/dir --remove \
+	%{_hhl_cross_info_dir}/bfd.info
 
 %post -n %{_pkgname}-doc
 for info in as binutils ld ; do
-	install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir \
-		%{_stm_cross_info_dir}/${info}.info
+	install-info --quiet --dir-file=%{_hhl_cross_info_dir}/dir \
+		%{_hhl_cross_info_dir}/${info}.info
 done
 
 %preun -n %{_pkgname}-doc
 for info in as binutils ld ; do
-	install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir --remove \
-		%{_stm_cross_info_dir}/${info}.info
+	install-info --quiet --dir-file=%{_hhl_cross_info_dir}/dir --remove \
+		%{_hhl_cross_info_dir}/${info}.info
 done
 
 %changelog
@@ -228,7 +228,7 @@
 * Thu Dec  9 2004 Stuart Menefy <stuart.menefy@st.com>
 - Updated for STtoolbuilder
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Tue Sep  7 2004 Stuart Menefy <stuart.menefy@st.com>
 - Updated to 2.15.91.0.1.
 * Fri Nov 14 2003 Stuart Menefy <stuart.menefy@st.com>
