--- SPECS/stm-target-gcc.spec	2011-07-04 18:37:44.665058535 +0200
+++ SPECS/stm-target-gcc.spec.new	2011-07-04 18:24:04.551789410 +0200
@@ -236,7 +236,6 @@
 	--enable-nls \
 	--enable-c99 \
 	--enable-long-long \
-	--with-system-zlib \
 	--enable-shared \
 %if "%{_stm_target_arch}" != "armv7"
 	--disable-libgomp \
@@ -378,6 +377,7 @@
 %dir %{_stm_cross_target_dir}%{_stm_target_lib_dir}/gcc/%{_stm_target_config}
 %dir %{_stm_cross_target_dir}%{_stm_target_lib_dir}/gcc/%{_stm_target_config}/%{version}
 %dir %{_stm_cross_target_dir}%{_stm_target_lib_dir}/gcc/%{_stm_target_config}/%{version}/include
+%dir %{_stm_cross_target_dir}%{_stm_target_lib_dir}/gcc/%{_stm_target_config}/%{version}/include/ssl
 %dir %{_gccdir}/include/ssp
 %{_gccdir}/include/ssp/ssp.h
 %{_gccdir}/include/ssp/stdio.h
