--- SPECS/stm-host-autotools.spec.orig	2010-05-14 18:19:40.000000000 +0200
+++ SPECS/stm-host-autotools.spec	2010-05-14 18:20:41.000000000 +0200
@@ -6,11 +6,9 @@
 Group: Development/Base
 Source0: http://ftp.gnu.org/gnu/autoconf/autoconf-2.59.tar.bz2
 Source1: http://ftp.gnu.org/gnu/automake/automake-1.8.5.tar.bz2
-Source2: http://ftp.gnu.org/gnu/libtool/libtool-1.5.10.tar.gz
 Source3: http://www.freedesktop.org/software/pkgconfig/releases/pkgconfig-0.21.tar.gz
 Patch0: automake-1.8.5-stm.patch
 Patch1: autoconf-2.59-stm.patch
-Patch2: libtool-1.5.10-stm.patch
 Patch3: pkgconfig-0.21-cross.patch
 Requires	: %{_stm_pkg_prefix}-host-filesystem
 Buildroot: %{_tmppath}/%{name}-%{version}-root
@@ -30,12 +28,6 @@
  - Automake is a tool for automatically generating
    `Makefile.in' files compliant with the GNU Coding Standards.
 
- - The libtool package contains the GNU libtool, a set of shell scripts
-   which automatically configure UNIX and UNIX-like architectures to
-   generically build shared libraries.  Libtool provides a consistent,
-   portable interface which simplifies the process of using shared
-   libraries.
-
  - The pkgconfig package makes putting together all the build
    flags when compiling/linking a lot easier.
 
@@ -49,10 +41,9 @@
 to select the corect version of the tools as appropriate.
 
 %prep
-%setup -n autotools-%{version} -c -a0 -a 1 -a 2 -a 3
+%setup -n autotools-%{version} -c -a0 -a 1 -a 3
 patch -d automake-1.8.5 -p1 < %PATCH0
 patch -d autoconf-2.59 -p1 < %PATCH1
-patch -d libtool-1.5.10 -p1 < %PATCH2
 patch -d pkg-config-0.21 -p1 < %PATCH3
 
 %build
@@ -86,7 +77,7 @@
 make install
 popd
 
-for d in autoconf-2.59 automake-1.8.5 libtool-1.5.10 ; do
+for d in autoconf-2.59 automake-1.8.5 ; do
   cd $d
   ./configure --prefix=%{_stm_host_dir}
   %make
@@ -104,7 +95,7 @@
 %install
 %host_setup
 
-for d in autoconf-2.59 automake-1.8.5 libtool-1.5.10 pkg-config-0.21; do
+for d in autoconf-2.59 automake-1.8.5 pkg-config-0.21; do
   cd $d
   %make prefix=%{buildroot}%{_stm_host_dir} install
   cd ..
@@ -120,10 +111,7 @@
 %compress_man %{buildroot}%{_stm_host_dir}/man
 
 # Remove unnecessary files
-rm -f %{buildroot}%{_stm_host_dir}/include/ltdl.h
 rm -f %{buildroot}%{_stm_host_dir}/info/*
-rm -f %{buildroot}%{_stm_host_dir}/lib/libltdl.a
-rm -f %{buildroot}%{_stm_host_dir}/lib/libltdl.la
 
 %clean
 rm -rf %{buildroot}
@@ -137,7 +125,6 @@
 %{_stm_host_dir}/man/man1/*.gz
 %{_stm_host_dir}/share/aclocal/pkg.m4
 %{_stm_host_dir}/lib/pkgconfig
-%{_stm_host_dir}/lib/libltdl.so*
 
 %changelog
 * Fri Jan 26 2007 Carl Shaw <carl.shaw@st.com> 8
