--- SPECS/stm-target-bc.spec	2006-10-12 22:14:34.000000000 +0200
+++ SPECS/stm-target-bc.spec-new	2007-07-21 11:28:56.000000000 +0200
@@ -1,15 +1,15 @@
 Summary: GNU bc arbitrary precision calculator 
-Name: %{_stm_pkg_prefix}-target-bc
+Name: %{_hhl_pkg_prefix}-target-bc
 Version: 1.06
 Release: 3
 Source: bc-%{version}.tar.gz
 Patch0: bc-bc-info.patch
 License: GPL
 Group: Applications/Engineering
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir	%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-bc
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir	%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-bc
 
 %description
 bc is an arbitrary precision numeric processing language.  Syntax is
@@ -39,38 +39,38 @@
 %install
 rm -rf %{buildroot}
 %target_setup
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_prefix}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_exec_prefix}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_prefix}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_exec_prefix}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}
 %target_makeinstall
-rm -f %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir*
-%compress_man %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}
-%compress_man %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
+rm -f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir*
+%compress_man %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}
+%compress_man %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}
 
 %clean
 rm -rf %{buildroot}
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/*
-%doc %{_stm_cross_target_dir}%{_stm_target_info_dir}/*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man?/*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man?/*
 %doc AUTHORS README COPYING COPYING.LIB FAQ INSTALL NEWS
 
 %post -n %{_pkgname}
 %target_script_prefix_setup
 for i in bc dc ; do
-  install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir \
-        %{_stm_cross_target_dir}%{_stm_target_info_dir}/${i}.info
+  install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir \
+        %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/${i}.info
 done
 
 %preun -n %{_pkgname}
 %target_script_prefix_setup
 for i in bc dc ; do
-  install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir --remove \
-        %{_stm_cross_target_dir}%{_stm_target_info_dir}/${i}.info
+  install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir --remove \
+        %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/${i}.info
 done
 
 
