--- SPECS/stm-target-glib2.spec	2006-12-15 16:10:08.000000000 +0100
+++ SPECS/stm-target-glib2.spec-new	2007-07-30 13:39:57.000000000 +0200
@@ -1,19 +1,19 @@
-Name:		%{_stm_pkg_prefix}-target-glib2
+Name:		%{_hhl_pkg_prefix}-target-glib2
 Version:	2.12.4
 Release:	9
 License:	LGPL
 Source:		ftp://ftp.gimp.org/pub/gtk/v2.6/glib-%{version}.tar.bz2
 URL:		http://www.gtk.org/
-Buildroot:	%{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix:		%{_stm_cross_target_dir}
-BuildRequires:	%{_stm_pkg_prefix}-host-rpmconfig
-
-%define _pkgname 			%{_stm_pkg_prefix}-%{_stm_target_arch}-glib2
-%define _docdir				%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-
-%define _stm_target_locale_dir		%{_stm_target_data_dir}/locale
-%define _stm_target_pkgconfig_dir	%{_stm_target_lib_dir}/pkgconfig
-%define _stm_host_toolprefix            target-
+Buildroot:	%{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix:		%{_hhl_cross_target_dir}
+BuildRequires:	%{_hhl_pkg_prefix}-host-rpmconfig
+
+%define _pkgname 			%{_hhl_pkg_prefix}-%{_hhl_target_arch}-glib2
+%define _docdir				%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+
+%define _hhl_target_locale_dir		%{_hhl_target_data_dir}/locale
+%define _hhl_target_pkgconfig_dir	%{_hhl_target_lib_dir}/pkgconfig
+%define _hhl_host_toolprefix            target-
 
 #
 #  Source Package
@@ -92,7 +92,7 @@
 
 %build
 %target_setup
-export LIBTOOL_PREFIX_BASE="%{_stm_cross_target_dir}"
+export LIBTOOL_PREFIX_BASE="%{_hhl_cross_target_dir}"
 %replaceautofiles
 
 mkdir -p %_objdir
@@ -118,7 +118,7 @@
 ac_cv_func_posix_getpwuid_r=\${ac_cv_func_posix_getpwuid_r=yes}
 EOF
 
-if [ "%{_stm_target_arch}" == "sh4" ]; then
+if [ "%{_hhl_target_arch}" == "sh4" ]; then
 	echo "glib_cv_va_copy=\${glib_cv_va_copy=yes}" >> config.cache
 	echo "glib_cv___va_copy=\${glib_cv___va_copy=yes}" >> config.cache
 	echo "glib_cv_va_val_copy=\${glib_cv_va_val_copy=yes}" >> config.cache
@@ -128,20 +128,20 @@
 	echo "glib_cv_va_val_copy=\${glib_cv_va_val_copy=yes}" >> config.cache
 fi
 
-export GLIB_MKENUMS="%{_stm_host_bin_dir}/%{_stm_host_toolprefix}glib-mkenums"
-export GLIB_GENMARSHAL="%{_stm_host_bin_dir}/%{_stm_host_toolprefix}glib-genmarshal"
-export GOBJECT_QUERY="%{_stm_host_bin_dir}/%{_stm_host_toolprefix}gobject-query"
+export GLIB_MKENUMS="%{_hhl_host_bin_dir}/%{_hhl_host_toolprefix}glib-mkenums"
+export GLIB_GENMARSHAL="%{_hhl_host_bin_dir}/%{_hhl_host_toolprefix}glib-genmarshal"
+export GOBJECT_QUERY="%{_hhl_host_bin_dir}/%{_hhl_host_toolprefix}gobject-query"
 
 %target_configure --cache-file="config.cache" \
         --disable-gtk-doc --with-threads="posix"
 
-%pmake
+%make
 
 %install
 %target_setup
 
 rm -rf %{buildroot}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_pkgconfig_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_pkgconfig_dir}
 
 %target_makeinstall
 
@@ -149,36 +149,36 @@
 sed     -e 's/^exec_prefix=.*/exec_prefix=\${prefix}/g' \
 	-e 's/^libdir=.*/libdir=\${prefix}\/lib/g' \
 	-e 's/^includedir=.*/includedir=\${prefix}\/include/g' \
-	-i %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/pkgconfig/glib-2.0.pc
+	-i %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/pkgconfig/glib-2.0.pc
 
 #  Remove perl files
-rm %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}/glib-mkenums
+rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/glib-mkenums
 
 %clean
 rm -rf %{buildroot}
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.so.*
-%{_stm_cross_target_dir}%{_stm_target_locale_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.so.*
+%{_hhl_cross_target_dir}%{_hhl_target_locale_dir}/*
 %doc COPYING AUTHORS
 
 %files -n %{_pkgname}-dev
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/*
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.la
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/glib-2.0/*
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/glib-2.0/*
-%{_stm_cross_target_dir}%{_stm_target_pkgconfig_dir}/*
-%{_stm_cross_target_dir}%{_stm_target_data_dir}/aclocal/*
-%{_stm_cross_target_dir}%{_stm_target_data_dir}/glib-2.0/*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.la
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/glib-2.0/*
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/glib-2.0/*
+%{_hhl_cross_target_dir}%{_hhl_target_pkgconfig_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/aclocal/*
+%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/glib-2.0/*
 %doc COPYING AUTHORS
 
 %files -n %{_pkgname}-doc
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_data_dir}/gtk-doc/*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man?/*
+%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/gtk-doc/*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man?/*
 %doc COPYING AUTHORS
 %doc README NEWS
 
@@ -208,7 +208,7 @@
 - Updated to version 2.6.4 and added ST200 support
 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Mon Jul 15 2002 Andrea Cisternino <andrea.cisternino@st.com>
 - Upgraded to 2.0.4. Major changes in the creation of support files
 
