--- stm-cross-gcc.spec.org	2011-03-22 02:05:02.000000000 +0100
+++ stm-cross-gcc.spec	2011-10-01 15:06:08.000000000 +0200
@@ -183,6 +183,7 @@
 
 %build
 %cross_setup
+export CC="gcc -m32"
 
 %cross_configure \
         --disable-checking \
@@ -221,7 +222,7 @@
         --with-mpfr=%{_stm_cross_dir} \
         --with-libelf=%{_stm_cross_dir} \
         --without-ppl \
-        --with-gxx-include-dir="\\\${prefix}/target%{_stm_target_include_dir}/c++/%{version}" \
+        --with-gxx-include-dir="\\\${prefix}/../../cdkroot%{_stm_target_include_dir}/c++/%{version}" \
         --enable-__cxa_atexit \
 %if "%{_stm_target_fpu}" == "no"
 %if_target_arch sh4_uclibc
@@ -257,13 +258,6 @@
 %cross_setup
 %cross_makeinstall
 
-%if_target_arch sh4 sh4_uclibc
-%make pdf html
-
-install -d %{buildroot}/%{_stm_cross_doc_dir}/html
-cp gcc/doc/*.pdf %{buildroot}/%{_stm_cross_doc_dir}
-cp -r gcc/HTML/gcc-%{version} %{buildroot}/%{_stm_cross_doc_dir}/html
-%endif
 
 cd ..
 
@@ -334,14 +328,16 @@
 	xargs --no-run-if-empty --verbose rm
 # Remove some installed files we don't ship, to keep rpm happy
 rm %{buildroot}%{_stm_cross_bin_dir}/%{_stm_target_toolprefix}gccbug
-rm %{buildroot}%{_stm_cross_info_dir}/cppinternals.info.gz
-rm %{buildroot}%{_stm_cross_info_dir}/gccinstall.info.gz
-rm %{buildroot}%{_stm_cross_info_dir}/gccint.info.gz
-rm -f %{buildroot}%{_stm_cross_info_dir}/dir.*
+rm %{buildroot}%{_stm_cross_info_dir}/cppinternals.info.gz || true
+rm %{buildroot}%{_stm_cross_info_dir}/gccinstall.info.gz || true
+rm %{buildroot}%{_stm_cross_info_dir}/gccint.info.gz || true
+rm -f %{buildroot}%{_stm_cross_info_dir}/dir.* || true
 rm -r %{buildroot}%{_stm_cross_lib_dir}/gcc/%{_stm_target_config}/%{version}/install-tools
 rm -f %{buildroot}%{_stm_cross_lib_dir}/libiberty.a
-rm -f  %{buildroot}%{_stm_cross_dir}/lib64/libiberty.a
+rm -f  %{buildroot}%{_stm_devkit_dir}/lib{32,64}/libiberty.a || true
 rm -r %{buildroot}%{_stm_cross_libexec_dir}/gcc/%{_stm_target_config}/%{version}/install-tools
+rm %{buildroot}%{_stm_cross_info_dir}/cpp.info* || true
+rm %{buildroot}%{_stm_cross_info_dir}/gcc.info* || true
 
 %unfixincludes  %{buildroot}%{_gccdir}
 
@@ -586,10 +582,6 @@
 %{_stm_cross_targetconf_dir}/bin/cpp
 %{_stm_cross_sharedstate_dir}/locale/*/LC_MESSAGES/cpplib.mo
 
-%files -n %{_cpp_pkgname}-doc
-%defattr(-,root,root)
-%{_stm_cross_info_dir}/cpp.info.gz
-
 %files -n %{_gpp_pkgname}
 %defattr(-,root,root)
 %{_stm_cross_bin_dir}/%{_stm_target_toolprefix}g++
@@ -604,7 +596,6 @@
 
 %files -n %{_gcc_pkgname}-doc
 %defattr(-,root,root)
-%{_stm_cross_info_dir}/gcc.info.gz
 %{_stm_cross_man_dir}/man7/*
 
 %if_target_arch sh4 sh4_uclibc
@@ -621,22 +612,6 @@
 %{_stm_cross_target_dir}/lib/m4-nofpu/libgcc_s-%{version}.so.1
 %endif
 
-%post -n %{_cpp_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir \
-	%{_stm_cross_info_dir}/cpp.info
-
-%preun -n %{_cpp_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir --remove \
-	%{_stm_cross_info_dir}/cpp.info
-
-%post -n %{_gcc_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir \
-	%{_stm_cross_info_dir}/gcc.info
-
-%preun -n %{_gcc_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_info_dir}/dir --remove \
-	%{_stm_cross_info_dir}/gcc.info
-
 %changelog
 * Wed Mar 16 2011 Christian Bruel <christian.bruel@st.com> 78
 - [Add patch: gcc-4.5.2-float_const_r0.patch; Bugzilla: 11193] Fix bad reload constraints on r0 register.
