--- SPECS/stm-host-mtd-utils.spec	2005-09-23 22:30:08.000000000 +0200
+++ SPECS/stm-host-mtd-utils.spec-new	2007-06-16 12:44:45.000000000 +0200
@@ -3,8 +3,9 @@
 Name: %{_hhl_pkg_prefix}-host-mtd-utils
 Version: 20050428
 Release: 4
-Copyright: GPL
+License: GPL
 Source: ftp://ftp.uk.linux.org/pub/people/dwmw2/mtd/CVS/mtd-snapshot-%{version}.tar.bz2
+Patch0: mtd-snapshot-20050428-srcdir.diff
 
 Buildroot: %{_tmppath}/%{name}-root
 Prefix: %{_hhl_install_prefix}
@@ -25,6 +26,7 @@
 
 %prep
 %setup -q -n mtd
+%patch -p1
 
 %build
 %host_setup
