--- SPECS/stm-target-glibc-sh4processed.spec	2006-10-20 20:39:33.000000000 +0200
+++ SPECS/stm-target-glibc-sh4processed.spec-new	2007-12-10 22:50:36.000000000 +0100
@@ -220,7 +220,7 @@
 fi
 
 # Remove files we don't package to keep RPM happy
-rm -rf %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir
+rm -rf %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir* | true
 
 %compress_man %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}
 %compress_man %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
