--- SPECS/stm-cross-distributionutils.spec	2005-09-23 22:30:07.000000000 +0200
+++ SPECS/stm-cross-distributionutils.spec-new	2007-06-16 13:27:42.000000000 +0200
@@ -4,6 +4,7 @@
 Release: 2
 Source: cross-hardhatutils.tar.gz
 Patch: cross-hardhatutils-stm.patch
+Patch1: hardhatutils-srcdir.diff
 License: GPL
 Group: System Environment/Base
 Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
@@ -31,6 +32,7 @@
 %target_check
 %setup -n hardhatutils -q
 %patch -p1
+%patch1 -p1
 
 %build
 make
