-- mpc-0.8.2/configure.ac.orig	2010-04-28 13:53:55.000000000 +0200
+++ mpc-0.8.2/configure.ac	2010-04-28 13:55:22.000000000 +0200
@@ -106,18 +106,6 @@
 AC_CHECK_LIB([gmp], [__gmpz_init],
              [LIBS="-lgmp $LIBS"],
              [AC_MSG_ERROR([libgmp not found or uses a different ABI.])])
-AC_MSG_CHECKING(for MPFR)
-LIBS="-lmpfr $LIBS"
-AC_LINK_IFELSE(
-        [AC_LANG_PROGRAM(
-                [[#include "mpfr.h"]],
-                [[mpfr_t x;  mpfr_init(x) ; mpfr_clear(x);]]
-        )],
-        [AC_MSG_RESULT(yes)],
-        [
-        AC_MSG_RESULT(no)
-        AC_MSG_ERROR([libmpfr not found or uses a different ABI.])
-        ])
 
 # Check for a recent GMP
 AC_MSG_CHECKING(for recent GMP)
