--- stslave/Makefile.orig	2010-04-17 14:05:23.000000000 +0200
+++ stslave/Makefile	2010-04-17 14:04:43.000000000 +0200
@@ -23,7 +23,7 @@
 install: cmdlib.a $(TARGET)
 
 $(TARGET): $(OBJ) stlog.h commands/cmdlib.a
-	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $(OBJ) commands/cmdlib.a -lbfd -liberty
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $(OBJ) commands/cmdlib.a -lbfd -lz -liberty
 
 cmdlib.a:
 	make -C commands AR=$(AR) CC=$(CC)
