--- SPECS/stm-target-e2fsprogs.spec	2006-10-12 22:14:34.000000000 +0200
+++ SPECS/stm-target-e2fsprogs.spec-new	2007-06-16 12:59:40.000000000 +0200
@@ -1,21 +1,21 @@
 Summary: The EXT2 file system utilities and libraries.
-Name: %{_stm_pkg_prefix}-target-e2fsprogs
+Name: %{_hhl_pkg_prefix}-target-e2fsprogs
 Version: 1.39
 Release: 8
 Source: http://prdownloads.sourceforge.net/e2fsprogs/e2fsprogs-%{version}.tar.gz
 Patch0: e2fsprogs-1.39-cross.patch
 License: GPL
 Group: Applications/File
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir	%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-
-%define _pkgname	%{_stm_pkg_prefix}-%{_stm_target_arch}-e2fsprogs
-%define _comerr_pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-comerr
-%define _ss_pkgname	%{_stm_pkg_prefix}-%{_stm_target_arch}-ss
-%define _uuid_pkgname	%{_stm_pkg_prefix}-%{_stm_target_arch}-uuid
-%define _e2fslibs_pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-e2fslibs
-%define _libblkid_pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-libblkid
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir	%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+
+%define _pkgname	%{_hhl_pkg_prefix}-%{_hhl_target_arch}-e2fsprogs
+%define _comerr_pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-comerr
+%define _ss_pkgname	%{_hhl_pkg_prefix}-%{_hhl_target_arch}-ss
+%define _uuid_pkgname	%{_hhl_pkg_prefix}-%{_hhl_target_arch}-uuid
+%define _e2fslibs_pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-e2fslibs
+%define _libblkid_pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libblkid
 
 %description
 EXT2 stands for "Extended Filesystem", version 2. It's the main
@@ -27,7 +27,7 @@
 %package -n %{_pkgname}
 Summary: The EXT2 file system utilities and libraries.
 Group: Applications/File
-Provides: %{_stm_pkg_prefix}-%{_stm_target_arch}-libcomerr %{_stm_pkg_prefix}-%{_stm_target_arch}-libss %{_stm_pkg_prefix}-%{_stm_target_arch}-libext2fs %{_stm_pkg_prefix}-%{_stm_target_arch}-libe2p2 %{_stm_pkg_prefix}-%{_stm_target_arch}-libuuid %{_stm_pkg_prefix}-%{_stm_target_arch}-e2fslibs %{_stm_pkg_prefix}-%{_stm_target_arch}-libblkid
+Provides: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libcomerr %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libss %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libext2fs %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libe2p2 %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libuuid %{_hhl_pkg_prefix}-%{_hhl_target_arch}-e2fslibs %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libblkid
 %description -n %{_pkgname}
 EXT2 stands for "Extended Filesystem", version 2. It's the main
 filesystem type used for hard disks on Debian and other Linux systems.
@@ -61,7 +61,7 @@
 %package -n %{_comerr_pkgname}-dev
 Summary: The Common Error Description library - headers and static libraries.
 Group: Development/Libraries
-Requires: %{_stm_pkg_prefix}-%{_stm_target_arch}-libcomerr
+Requires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libcomerr
 %description -n %{_comerr_pkgname}-dev
 libcomerr is an attempt to present a common error-handling mechanism to
 manipulate the most common form of error code in a fashion that does not
@@ -70,7 +70,7 @@
 %package -n %{_ss_pkgname}-dev
 Summary: Command-line interface parsing library - headers and static libraries.
 Group: Development/Tools
-Requires: %{_stm_pkg_prefix}-%{_stm_target_arch}-libss
+Requires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libss
 %description -n %{_ss_pkgname}-dev
 This package includes a tool that parses a command table to generate
 a simple command-line interface parser, the include files needed to
@@ -81,7 +81,7 @@
 %package -n %{_uuid_pkgname}-dev
 Summary: Universally unique id library - headers and static libraries.
 Group: Development/Libraries
-Requires: %{_stm_pkg_prefix}-%{_stm_target_arch}-libuuid
+Requires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libuuid
 %description -n %{_uuid_pkgname}-dev
 libuuid generates and parses 128-bit universally unique id's (UUID's),
 using a standard which is blessed by both Microsoft and DCE, and is
@@ -94,8 +94,8 @@
 %package -n %{_e2fslibs_pkgname}-dev
 Summary: The headers and static libraries for ext2fs-aware tools-development.
 Group: Development/System
-Requires: %{_stm_pkg_prefix}-%{_stm_target_arch}-comerr-dev %{_stm_pkg_prefix}-%{_stm_target_arch}-e2fsprogs
-Provides: %{_stm_pkg_prefix}-%{_stm_target_arch}-e2fs-dev %{_stm_pkg_prefix}-%{_stm_target_arch}-e2p-dev
+Requires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-comerr-dev %{_hhl_pkg_prefix}-%{_hhl_target_arch}-e2fsprogs
+Provides: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-e2fs-dev %{_hhl_pkg_prefix}-%{_hhl_target_arch}-e2p-dev
 %description -n %{_e2fslibs_pkgname}-dev
 EXT2FS stands for "Extended Filesystem", version 2. It's the filesystem
 type used for hard disks on Debian and other Linux systems.
@@ -121,7 +121,7 @@
 %package -n %{_libblkid_pkgname}-dev
 Summary: Block device id library
 Group: Development/System
-Requires: %{_stm_pkg_prefix}-%{_stm_target_arch}-libblkid
+Requires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libblkid
 %description -n %{_libblkid_pkgname}-dev
 The blkid library which allows system programs like fsck and 
 mount to quickly and easily find block devices by filesystem UUID and 
@@ -152,22 +152,25 @@
 
 export LDCONFIG=/bin/true
 %target_makeinstall install-libs \
-	root_bindir=%{buildroot}%{_stm_cross_target_dir}/bin \
-	root_sbindir=%{buildroot}%{_stm_cross_target_dir}/sbin \
-	root_libdir=%{buildroot}%{_stm_cross_target_dir}/lib
+	root_bindir=%{buildroot}%{_hhl_cross_target_dir}/bin \
+	root_sbindir=%{buildroot}%{_hhl_cross_target_dir}/sbin \
+	root_libdir=%{buildroot}%{_hhl_cross_target_dir}/lib
 
 # Install static version of e2fsck
-install e2fsck/e2fsck.static %{buildroot}%{_stm_cross_target_dir}/sbin
-ln -s e2fsck.8 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/e2fsck.static.8
+install e2fsck/e2fsck.static %{buildroot}%{_hhl_cross_target_dir}/sbin
+ln -s e2fsck.8 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/e2fsck.static.8
+
+# Install config
+install ../misc/mke2fs.conf %{buildroot}%{_hhl_cross_target_dir}/etc
 
 cd ..
 
-%compress_man %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
-%compress_man %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}
+%compress_man %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}
+%compress_man %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}
 
 # Make shared library symbolic links relative
 for lib in libcom_err libe2p libext2fs libss libuuid libblkid ; do
-  p=%{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/$lib.so
+  p=%{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/$lib.so
   l=`readlink $p`
   b=`basename $l`
   ln -s -f ../../lib/$b $p
@@ -183,128 +186,128 @@
 #   ...-libuuid ...-e2fslibs ...-libblkid
 %files -n %{_pkgname} -f e2fsprogs.lang
 %defattr(-,root,root)
-%{_stm_cross_target_dir}/sbin/badblocks
-%{_stm_cross_target_dir}/sbin/blkid
-%{_stm_cross_target_dir}/sbin/debugfs
-%{_stm_cross_target_dir}/sbin/dumpe2fs
-%{_stm_cross_target_dir}/sbin/e2fsck
-%{_stm_cross_target_dir}/sbin/e2image
-%{_stm_cross_target_dir}/sbin/e2label
-%{_stm_cross_target_dir}/sbin/findfs
-%{_stm_cross_target_dir}/sbin/fsck
-%{_stm_cross_target_dir}/sbin/fsck.ext2
-%{_stm_cross_target_dir}/sbin/fsck.ext3
-%{_stm_cross_target_dir}/sbin/logsave
-%{_stm_cross_target_dir}/sbin/mke2fs
-%{_stm_cross_target_dir}/sbin/mkfs.ext2
-%{_stm_cross_target_dir}/sbin/mkfs.ext3
-%{_stm_cross_target_dir}/sbin/resize2fs
-%{_stm_cross_target_dir}/sbin/tune2fs
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/chattr
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/lsattr
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/filefrag
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/mklost+found
-%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/mke2fs.conf
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/e2initrd_helper
-%{_stm_cross_target_dir}/lib/libblkid.so.*
-%{_stm_cross_target_dir}/lib/libcom_err.so.*
-%{_stm_cross_target_dir}/lib/libe2p.so.*
-%{_stm_cross_target_dir}/lib/libext2fs.so.*
-%{_stm_cross_target_dir}/lib/libss.so.*
-%{_stm_cross_target_dir}/lib/libuuid.so.*
+%{_hhl_cross_target_dir}/sbin/badblocks
+%{_hhl_cross_target_dir}/sbin/blkid
+%{_hhl_cross_target_dir}/sbin/debugfs
+%{_hhl_cross_target_dir}/sbin/dumpe2fs
+%{_hhl_cross_target_dir}/sbin/e2fsck
+%{_hhl_cross_target_dir}/sbin/e2image
+%{_hhl_cross_target_dir}/sbin/e2label
+%{_hhl_cross_target_dir}/sbin/findfs
+%{_hhl_cross_target_dir}/sbin/fsck
+%{_hhl_cross_target_dir}/sbin/fsck.ext2
+%{_hhl_cross_target_dir}/sbin/fsck.ext3
+%{_hhl_cross_target_dir}/sbin/logsave
+%{_hhl_cross_target_dir}/sbin/mke2fs
+%{_hhl_cross_target_dir}/sbin/mkfs.ext2
+%{_hhl_cross_target_dir}/sbin/mkfs.ext3
+%{_hhl_cross_target_dir}/sbin/resize2fs
+%{_hhl_cross_target_dir}/sbin/tune2fs
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/chattr
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/lsattr
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/filefrag
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/mklost+found
+%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/mke2fs.conf
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/e2initrd_helper
+%{_hhl_cross_target_dir}/lib/libblkid.so.*
+%{_hhl_cross_target_dir}/lib/libcom_err.so.*
+%{_hhl_cross_target_dir}/lib/libe2p.so.*
+%{_hhl_cross_target_dir}/lib/libext2fs.so.*
+%{_hhl_cross_target_dir}/lib/libss.so.*
+%{_hhl_cross_target_dir}/lib/libuuid.so.*
 %doc LICENSE
 
 %files -n %{_pkgname}-doc
 %defattr(-,root,root)
 %doc README RELEASE-NOTES SHLIBS
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/badblocks.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/blkid.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/debugfs.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/dumpe2fs.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/e2fsck.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/e2image.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/e2label.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/filefrag.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/findfs.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/fsck.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/fsck.ext2.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/fsck.ext3.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/logsave.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/mke2fs.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/mkfs.ext2.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/mkfs.ext3.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/mklost+found.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/resize2fs.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/tune2fs.8*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/chattr.1*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/lsattr.1*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man5/e2fsck.conf.5*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man5/mke2fs.conf.5*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/badblocks.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/blkid.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/debugfs.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/dumpe2fs.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/e2fsck.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/e2image.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/e2label.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/filefrag.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/findfs.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/fsck.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/fsck.ext2.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/fsck.ext3.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/logsave.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/mke2fs.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/mkfs.ext2.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/mkfs.ext3.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/mklost+found.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/resize2fs.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/tune2fs.8*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/chattr.1*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/lsattr.1*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/e2fsck.conf.5*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/mke2fs.conf.5*
 %doc LICENSE
 
 %files -n %{_pkgname}-static
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/e2fsck.static*
-%{_stm_cross_target_dir}/sbin/e2fsck.static
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/e2fsck.static*
+%{_hhl_cross_target_dir}/sbin/e2fsck.static
 %doc LICENSE
 
 %files -n %{_e2fslibs_pkgname}-dev
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_info_dir}/libext2fs.info*
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libext2fs.*
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libe2p.*
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/ext2fs
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/e2p
-%{_stm_cross_target_dir}%{_stm_target_pkgconfig_dir}/ext2fs.pc
-%{_stm_cross_target_dir}%{_stm_target_pkgconfig_dir}/e2p.pc
+%doc %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/libext2fs.info*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libext2fs.*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libe2p.*
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/ext2fs
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/e2p
+%{_hhl_cross_target_dir}%{_hhl_target_pkgconfig_dir}/ext2fs.pc
+%{_hhl_cross_target_dir}%{_hhl_target_pkgconfig_dir}/e2p.pc
 %doc LICENSE
 
 %files -n %{_uuid_pkgname}-dev
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/uuidgen*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man3/*uuid*
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/uuidgen
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libuuid.*
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/uuid
-%{_stm_cross_target_dir}%{_stm_target_pkgconfig_dir}/uuid.pc
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/uuidgen*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3/*uuid*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/uuidgen
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libuuid.*
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/uuid
+%{_hhl_cross_target_dir}%{_hhl_target_pkgconfig_dir}/uuid.pc
 %doc LICENSE
 
 %files -n %{_comerr_pkgname}-dev
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/compile_et*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man3/com_err.3*
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/compile_et
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libcom_err.*
-%{_stm_cross_target_dir}%{_stm_target_data_dir}/et
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/et
-%{_stm_cross_target_dir}%{_stm_target_pkgconfig_dir}/com_err.pc
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/compile_et*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3/com_err.3*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/compile_et
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libcom_err.*
+%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/et
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/et
+%{_hhl_cross_target_dir}%{_hhl_target_pkgconfig_dir}/com_err.pc
 %doc LICENSE
 
 %files -n %{_ss_pkgname}-dev
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/mk_cmds.*
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/mk_cmds
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/ss
-%{_stm_cross_target_dir}%{_stm_target_data_dir}/ss
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libss.*
-%{_stm_cross_target_dir}%{_stm_target_pkgconfig_dir}/ss.pc
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/mk_cmds.*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/mk_cmds
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/ss
+%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/ss
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libss.*
+%{_hhl_cross_target_dir}%{_hhl_target_pkgconfig_dir}/ss.pc
 %doc LICENSE
 
 %files -n %{_libblkid_pkgname}-dev
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man3/libblkid.3*
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/blkid
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libblkid.*
-%{_stm_cross_target_dir}%{_stm_target_pkgconfig_dir}/blkid.pc
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3/libblkid.3*
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/blkid
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libblkid.*
+%{_hhl_cross_target_dir}%{_hhl_target_pkgconfig_dir}/blkid.pc
 %doc LICENSE
 
 %post -n %{_e2fslibs_pkgname}-dev
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir \
-        %{_stm_cross_target_dir}%{_stm_target_info_dir}/libext2fs.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir \
+        %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/libext2fs.info
 
 %preun -n %{_e2fslibs_pkgname}-dev
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir --remove \
-        %{_stm_cross_target_dir}%{_stm_target_info_dir}/libext2fs.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir --remove \
+        %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/libext2fs.info
 
 %changelog
 * Thu Aug 24 2006 Stuart Menefy <stuart.menefy@st.com>
@@ -315,7 +318,7 @@
 - Automatically removed dependency opt-out line
 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Mon Sep 06 2004 Stuart Menefy <stuart.menefy@st.com>
 - Use %compress_man macros. Fixed install-info paths.
 
