--- rpmconfig-2.3.orig/common	2010-05-21 23:40:10.000000000 +0200
+++ rpmconfig-2.3/common	2010-05-21 23:40:02.000000000 +0200
@@ -367,7 +367,11 @@
   done ; 
 
 %replaceautofiles \
-  atbase=%{_stm_host_dir}/share/automake-1.8 ;\
+  if [ -d %{_stm_host_dir}/share/automake-1.8 ]; then \
+    atbase=%{_stm_host_dir}/share/automake-1.8 ; \
+  else \
+    atbase=%{_stm_host_dir}/share/misc ; \
+  fi \
   subfiles="`find .  -name config\.sub`" ; \
   guessfiles="`find . -name config\.guess`" ; \
   if [ -n "$subfiles" ]; then\
