--- SPECS/stm-target-libjpeg.spec	2006-10-12 22:14:35.000000000 +0200
+++ SPECS/stm-target-libjpeg.spec-new	2007-07-14 10:38:45.000000000 +0200
@@ -1,6 +1,6 @@
 # $Id: stm-target-libjpeg.spec,v 1.13 2006/10/12 19:57:31 shawc Exp $
 
-Name:		%{_stm_pkg_prefix}-target-libjpeg
+Name:		%{_hhl_pkg_prefix}-target-libjpeg
 Version:	6b
 Release: 	8
 License:	Freely distributable
@@ -10,12 +10,12 @@
 Patch2:		libjpeg-6b-ltconfig.patch
 Patch3:		libjpeg-6b-man.patch
 URL:		http://www.ijg.org/
-Buildroot:	%{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix:		%{_stm_cross_target_dir}
-#BuildRequires:	%{_stm_pkg_prefix}-host-rpmconfig
+Buildroot:	%{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix:		%{_hhl_cross_target_dir}
+#BuildRequires:	%{_hhl_pkg_prefix}-host-rpmconfig
 
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-libjpeg
-%define _docdir  %{_stm_cross_target_dir}%{_stm_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-libjpeg
+%define _docdir  %{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
 
 #
 #  Source Package
@@ -45,7 +45,7 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.so.*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.so.*
 %doc README
 
 #
@@ -65,11 +65,11 @@
 
 %files -n %{_pkgname}-dev
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/*
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.a
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.la
-%{_stm_cross_libtool_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.a
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.la
+%{_hhl_cross_libtool_dir}/*
 %doc README
 
 #
@@ -89,8 +89,8 @@
 
 %files -n %{_pkgname}-progs
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/*
-%{_stm_cross_target_dir}%{_stm_target_man_dir}/man1/*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/*
 %doc README
 
 #
@@ -141,19 +141,19 @@
 %target_setup
 
 rm -rf %{buildroot}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_include_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man1
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_include_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1
 
 %define _objdir .
 %target_makeinstall install-lib install-headers
 
 # compress documentation
-gzip -9r %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
+gzip -9r %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}
 
 # fix some flags
-cd %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
+cd %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
 chmod -x *.la
 chmod +x *.so.*
 
@@ -172,7 +172,7 @@
 * Tue May 10 2005 Carl Shaw <carl.shaw@st.com>
 - Added support for ST200 config 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Mon May 17 2004 Stuart Menefy <stuart.menefy@st.com>
 - Replaced jpegsrc.v6b-autoconf.patch with Debian patch libjpeg6b_6b-9.diff.gz
   plus config.sub-sh.patch, libjpeg-6b-ltconfig.patch and libjpeg-6b-man.patch
