--- rpmconfig-2.2/common	2008-04-28 09:40:23.000000000 +0200
+++ rpmconfig-2.2/common.new	2010-02-05 23:50:27.000000000 +0100
@@ -533,6 +533,7 @@
 find %* "(" "(" -type f -o -type l ")" -a '!' -name "*.gz" ")" -print | while read AFile ; do \
   test -f $AFile && gzip -9 -f $AFile; \
   test -L $AFile && ln -s `readlink $AFile`.gz $AFile.gz && rm $AFile; \
+  true; \
 done ; 
 
 # Update the password file
