--- SPECS/stm-target-autofs.spec	2007-03-22 16:43:10.000000000 +0100
+++ SPECS/stm-target-autofs.spec-new	2007-07-21 13:25:27.000000000 +0200
@@ -3,7 +3,7 @@
 Summary(fr): démon autofs
 Summary(tr): autofs sunucu süreci
 Summary(sv): autofs-daemon
-Name: %{_stm_pkg_prefix}-target-autofs
+Name: %{_hhl_pkg_prefix}-target-autofs
 Version: 3.1.7
 Release: 9
 Source: ftp://ftp.kernel.org/pub/linux/daemons/autofs/autofs-%{version}.tar.bz2
@@ -12,10 +12,10 @@
 Patch2: autofs-3.1.7-labelfix.patch
 License: GPL
 Group: System Environment/Daemons
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir	%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-autofs
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir	%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-autofs
 
 %description
 autofs is a daemon which automatically mounts filesystems when you use
@@ -45,8 +45,8 @@
 %package -n %{_pkgname}
 Summary: autofs daemon
 Group: System Environment/Daemons
-PreReq: %{_stm_pkg_prefix}-%{_stm_target_arch}-initdconfig
-PreReq: %{_stm_pkg_prefix}-%{_stm_target_arch}-initscripts
+PreReq: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-initdconfig
+PreReq: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-initscripts
 %description -n %{_pkgname}
 autofs is a daemon which automatically mounts filesystems when you use
 them, and unmounts them later when you are not using them.  This can
@@ -60,38 +60,42 @@
 
 %build
 %target_setup
+
+rm -rf %{buildroot}
+
 initdir=/etc/rc.d/init.d			\
 	./configure				\
-	--host=%{_stm_target_config}		\
-	--mandir=%{_stm_target_man_dir}
+	--host=%{_hhl_target_config}		\
+	--mandir=%{_hhl_target_man_dir}
 %make						\
-	 CC=%{_stm_target_toolprefix}gcc	\
-	 STRIP=%{_stm_target_toolprefix}strip
+	 CC=%{_hhl_target_toolprefix}gcc	\
+	 STRIP=%{_hhl_target_toolprefix}strip
 
 %install
 %target_setup
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/rc.d/init.d
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man5
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/rc.d/init.d
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
 
 make install install_samples			\
-	CC=%{_stm_target_toolprefix}gcc		\
-	STRIP=%{_stm_target_toolprefix}strip	\
-	INSTALLROOT=%{buildroot}%{_stm_cross_target_dir}
-
-install -m 755 -d %{buildroot}%{_stm_cross_target_dir}/misc
-
-%target_install_fixup
+	CC=%{_hhl_target_toolprefix}gcc		\
+	STRIP=%{_hhl_target_toolprefix}strip	\
+	INSTALLROOT=%{buildroot}%{_hhl_cross_target_dir}
+
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/autofs.5
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/auto.master.5
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/autofs.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/automount.8
 
 %post -n %{_pkgname}
 %target_script_prefix_setup
 # Register the service
-(%{_stm_target_toolprefix}initdconfig --add autofs) &> /dev/null \
+(%{_hhl_target_toolprefix}initdconfig --add autofs) &> /dev/null \
 	|| echo "Unable to enable initd service: autofs"
 
-(%{_stm_target_toolprefix}initdconfig autofs off) &> /dev/null \
+(%{_hhl_target_toolprefix}initdconfig autofs off) &> /dev/null \
 	|| echo "Unable to switch off initd service: autofs"
 
 %clean
@@ -99,27 +103,26 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/rc.d/init.d/autofs
-%config(missingok) %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/auto.master
-%config(missingok) %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/auto.misc
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/automount
-%dir %{_stm_cross_target_dir}/misc
-%dir %{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/lookup_file.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/lookup_multi.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/lookup_nisplus.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/lookup_program.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/lookup_userhome.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/lookup_yp.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/mount_afs.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/mount_autofs.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/mount_bind.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/mount_changer.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/mount_ext2.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/mount_generic.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/mount_nfs.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/autofs/parse_sun.so
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man?/*
+%config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/rc.d/init.d/autofs
+%config(missingok) %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/auto.master
+%config(missingok) %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/auto.misc
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/automount
+%dir %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/lookup_file.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/lookup_multi.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/lookup_nisplus.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/lookup_program.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/lookup_userhome.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/lookup_yp.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/mount_afs.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/mount_autofs.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/mount_bind.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/mount_changer.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/mount_ext2.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/mount_generic.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/mount_nfs.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/autofs/parse_sun.so
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man?/*
 %doc COPYRIGHT NEWS README TODO
 
 %changelog
@@ -147,7 +150,7 @@
 - Automatically removed dependency opt-out line
 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Fri Jul 25 2003 Stuart Menefy <stuart.menefy@st.com>
 - Force the locaton of the initdir, otherwise it is derived from the host,
   which may or may not be correct.
