--- SPECS/stm-target-hotplug.spec	2006-01-31 02:00:02.000000000 +0100
+++ SPECS/stm-target-hotplug.spec-new	2007-08-21 17:06:04.000000000 +0200
@@ -4,7 +4,7 @@
 %define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-hotplug
 Version:	2004_09_23
 Release: 	4
-Copyright:	GPL
+License:	GPL
 Source:		hotplug-%{version}.tar.gz
 Patch1:		hotplug-2004_09_23_usb_agent.patch
 Buildroot:	%{_tmppath}/%{name}-%{_hhl_target_arch}-root
