--- SPECS/stm-target-gcc-sh4processed.spec	2006-02-07 18:25:20.000000000 +0100
+++ SPECS/stm-target-gcc-sh4processed.spec-new	2007-06-16 11:04:55.000000000 +0200
@@ -313,7 +313,7 @@
 rm  %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/cppinternals.info.gz
 rm  %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/gccint.info.gz
 rm  %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/gccinstall.info.gz
-rm  %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir.gz
+rm  %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir* || true
 rm -r %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man7
 rm -r %{buildroot}%{_gccdir}/install-tools
 rm -r %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_libexec_dir}/gcc/%{_hhl_target_config}/%{gcc_version}/install-tools
