--- SPECS/stm-target-directfb.spec	2007-12-07 14:02:40.000000000 +0100
+++ SPECS/stm-target-directfb.spec-new	2008-05-05 21:34:55.000000000 +0200
@@ -87,7 +87,7 @@
 #
 %build
 %target_setup
-export FREETYPE_CONFIG="%{_stm_cross_bin_dir}/%{_stm_target_toolprefix}freetype-config"
+export FREETYPE_CONFIG="%{_stm_cross_bin_dir}/freetype-config"
 %target_configure --enable-static --disable-sdl --with-gfxdrivers=stgfx
 
 %make
