--- SPECS/stm-target-portmap.spec	2007-03-22 16:43:10.000000000 +0100
+++ SPECS/stm-target-portmap.spec-new	2007-06-14 13:28:14.000000000 +0200
@@ -1,5 +1,5 @@
 Summary: The RPC portmapper
-Name: %{_stm_pkg_prefix}-target-portmap
+Name: %{_hhl_pkg_prefix}-target-portmap
 Version: 5beta
 Release: 5
 Source: ftp://coast.cs.purdue.edu/pub/tools/unix/netutils/portmap/portmap_%{version}.tar.gz
@@ -10,11 +10,11 @@
 Patch13: portmap-errno.patch
 License: BSD
 Group: System Environment/Daemons
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir		%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-BuildRequires: %{_stm_pkg_prefix}-%{_stm_target_arch}-tcp-wrappers-dev
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-portmap
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir		%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+BuildRequires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-tcp-wrappers-dev
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-portmap
 
 %description
 Portmap is a server that converts RPC (Remote Procedure Call) program
@@ -27,9 +27,9 @@
 %package -n %{_pkgname}
 Summary: The RPC portmapper
 Group: System Environment/Daemons
-PreReq		: %{_stm_pkg_prefix}-%{_stm_target_arch}-initdconfig
-PreReq		: %{_stm_pkg_prefix}-%{_stm_target_arch}-initscripts
-Requires	: %{_stm_pkg_prefix}-%{_stm_target_arch}-tcp-wrappers
+PreReq		: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-initdconfig
+PreReq		: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-initscripts
+Requires	: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-tcp-wrappers
 %description -n %{_pkgname}
 Portmap is a server that converts RPC (Remote Procedure Call) program
 numbers into DARPA protocol port numbers.  It must be running in order
@@ -51,21 +51,29 @@
 
 %install
 %target_setup
-mkdir -p %{buildroot}%{_stm_cross_target_dir}/sbin
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/rc.d/init.d
-
-%make BASEDIR=%{buildroot}%{_stm_cross_target_dir} install
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d
-install -m 0755 debian/init.d %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d/portmap
 
-%target_install_fixup
+rm -rf %{buildroot}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}/sbin
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/rc.d/init.d
+
+%make BASEDIR=%{buildroot}%{_hhl_cross_target_dir} install
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d
+install -m 0755 debian/init.d %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d/portmap
+
+#install -m 644 rdate.1 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/pmap_dump.8
+#install -m 644 rdate.1 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/pmap_set.8
+#install -m 644 rdate.1 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/portmap.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/pmap_dump.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/pmap_set.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/portmap.8
+
 cp debian/copyright LICENSE
 
 %post -n %{_pkgname}
 %target_script_prefix_setup
 # Register the service
-(%{_stm_target_toolprefix}initdconfig --add portmap) &> /dev/null \
+(%{_hhl_target_toolprefix}initdconfig --add portmap) &> /dev/null \
 	|| echo "Unable to enable initd service: portmap"
 
 %clean
@@ -74,13 +82,13 @@
 %files -n %{_pkgname}
 %defattr(-,root,root)
 
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/pmap_dump.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/pmap_set.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/portmap.8.gz
-%config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d/portmap
-%{_stm_cross_target_dir}/sbin/pmap_dump
-%{_stm_cross_target_dir}/sbin/pmap_set
-%{_stm_cross_target_dir}/sbin/portmap
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/pmap_dump.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/pmap_set.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/portmap.8.gz
+%config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d/portmap
+%{_hhl_cross_target_dir}/sbin/pmap_dump
+%{_hhl_cross_target_dir}/sbin/pmap_set
+%{_hhl_cross_target_dir}/sbin/portmap
 %doc BLURB CHANGES README debian/copyright
 %doc LICENSE
 
@@ -97,6 +105,6 @@
 * Wed Dec 22 2004 Stuart Menefy <stuart.menefy@st.com>
 - Removed logging of build. Added portmap-errno.patch.
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Sat Jan 26 2002 Stuart Menefy <stuart.menefy@st.com>
 - Copied from MontaVista version 5beta-hhl2.0.1
