--- SPECS/stm-target-gcc.spec	2011-07-04 18:37:44.665058535 +0200
+++ SPECS/stm-target-gcc.spec.new	2011-07-04 18:24:04.551789410 +0200
@@ -236,7 +236,6 @@
 	--enable-nls \
 	--enable-c99 \
 	--enable-long-long \
-	--with-system-zlib \
 	--enable-shared \
 %if "%{_stm_target_arch}" != "armv7"
 	--disable-libgomp \
@@ -353,6 +353,9 @@
 rm -r %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man7
 rm -r %{buildroot}%{_gccdir}/install-tools
 rm -r %{buildroot}%{_stm_cross_target_dir}%{_stm_target_libexec_dir}/gcc/%{_stm_target_config}/%{version}/install-tools
+rm -rf %{buildroot}%{_gccdir}/include/ssl
+rm -rf %{buildroot}%{_gccdir}/include/xulrunner-1.9.2.17/nss
+rm -rf %{buildroot}%{_gccdir}/include/seamonkey
 
 %unfixincludes %{buildroot}%{_gccdir}
 
