diff -Nhur rpmconfig-2.3/common rpmconfig-2.3.orig//common
--- rpmconfig-2.3/common	2010-05-13 10:03:58.000000000 +0200
+++ rpmconfig-2.3.orig//common	2010-05-13 10:00:47.000000000 +0200
@@ -316,7 +316,7 @@
 %autoremake \
 install="--install" ; \
 [ -n "`autoreconf --version | grep 2.13`" ] && install="" ; \
-autoreconf --verbose --force $install -I%{_stm_host_sharedstate_dir}/aclocal
+autoreconf --verbose --force $install
 
 
 %target_autoreconf \
@@ -332,7 +332,7 @@
     cn=`basename $conf`; \
     echo "cd $dn"; \
     pushd $dn ; \
-    autoreconf --force --install -I%{_stm_host_sharedstate_dir}/aclocal ; \
+    autoreconf --force --install ; \
     %replacelibtool \
     popd; \
   done ;
