--- SPECS/stm-target-rdate.spec	2007-02-08 18:20:49.000000000 +0100
+++ SPECS/stm-target-rdate.spec-new	2007-06-14 13:15:58.000000000 +0200
@@ -1,14 +1,14 @@
 Summary: Set the system's date from a remote host.
-Name: %{_stm_pkg_prefix}-target-rdate
+Name: %{_hhl_pkg_prefix}-target-rdate
 Version: 1.4
 Release: 5
 License: BSD
 Group: Applications/Internet
 Source: rdate-%{version}.tar.gz
 Buildroot: %{_tmppath}/%{_pkgname}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir %{_stm_cross_target_dir}%{_stm_target_doc_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-rdate
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir %{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-rdate
 
 %description
 Rdate displays and sets the local date and time from the host name or address
@@ -28,16 +28,19 @@
 
 %build
 %target_setup
-%{_stm_target_toolprefix}gcc -O2 "-D__COPYRIGHT(x)=" "-D__RCSID(x)=" \
+%{_hhl_target_toolprefix}gcc -O2 "-D__COPYRIGHT(x)=" "-D__RCSID(x)=" \
 	rdate.c -o rdate -lutil
 
 %install
 %target_setup
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-install -m 755 rdate %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}
-install -m 644 rdate.1 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/rdate.1
-%target_install_fixup
+
+rm -rf %{buildroot}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1
+install -m 755 rdate %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}
+install -m 644 rdate.1 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/rdate.1
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/rdate.1
+
 cp COPYING LICENSE
 
 %clean
@@ -45,8 +48,8 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/*
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man*/*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man*/*
 %doc LICENSE
 
 %changelog
@@ -60,6 +63,6 @@
 - Automatically removed dependency opt-out line
 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Wed Feb 06 2002 Stuart Menefy <stuart.menefy@st.com>
 - Written.
