--- SPECS/stm-target-glibc-sh4processed.spec	2006-10-20 20:39:33.000000000 +0200
+++ SPECS/stm-target-glibc-sh4processed.spec-new	2007-06-16 15:41:17.000000000 +0200
@@ -1,5 +1,5 @@
 Summary: The GNU C Library with nptl
-Name: %{_stm_pkg_prefix}-target-glibc
+Name: %{_hhl_pkg_prefix}-target-glibc
 %define glibc_version 2.5
 Version: %{glibc_version}
 Release: 27
@@ -19,14 +19,14 @@
 
 
 
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-glibc
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-glibc
 
 License: LGPL/BSD
 Group:	System Environment/Libraries
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir          %{_stm_cross_target_dir}/%{_stm_target_doc_dir}
-BuildRequires: %{_stm_pkg_prefix}-%{_stm_target_arch}-kernel-headers
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir          %{_hhl_cross_target_dir}/%{_hhl_target_doc_dir}
+BuildRequires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-kernel-headers
 
 %description
 The GNU C Library including standard libraries used by nearly all
@@ -38,7 +38,7 @@
 %package -n %{_pkgname}
 Summary: GNU C Library: Shared libraries and Timezone data
 Group:	System Environment/Base
-Requires: %{_stm_pkg_prefix}-%{_stm_target_arch}-filesystem
+Requires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-filesystem
 %description -n %{_pkgname}
 Contains the standard libraries that are used by nearly all programs on
 the system. This package includes shared versions of the standard C library
@@ -50,7 +50,7 @@
 %package -n %{_pkgname}-dev
 Summary: GNU C Library: Development Libraries and Header Files.
 Group:	Development/Languages
-Requires: %{_stm_pkg_prefix}-%{_stm_target_arch}-kernel-headers
+Requires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-kernel-headers
 %description -n %{_pkgname}-dev
 Contains the symlinks, headers, and object files needed to compile
 and link programs which use the standard C library. 
@@ -144,13 +144,13 @@
 	--with-__thread \
    --enable-add-ons=nptl \
    --enable-shared \
-        --build=%{_stm_host_config} \
+        --build=%{_hhl_host_config} \
 	--enable-kernel=2.6.11 \
         --without-cvs \
 	--disable-profile \
         --disable-debug \
         --without-gd \
-        --with-headers=%{_stm_cross_target_dir}/usr/include
+        --with-headers=%{_hhl_cross_target_dir}/usr/include
 
 %make
 
@@ -163,12 +163,12 @@
 %target_setup
 rm -rf %{buildroot}
 cd %{_objdir} 
-make install install_root=%{buildroot}%{_stm_cross_target_dir}
+make install install_root=%{buildroot}%{_hhl_cross_target_dir}
 
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man1
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/default
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/default
 
 #  For the st231, we need to remove __restrict_arr from a few files or open64 will not be happy
 #  hopefully this can be removed in future when the front end version for C moves to gcc 3.4
@@ -178,146 +178,146 @@
 cd host_timezone
 # CRS - TODO Deal with systemv and backward which are links...
 TZBASE="africa antarctica asia australasia europe northamerica southamerica etcetera factory solar87 solar88 solar89"
-./zic -d %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/zoneinfo/right/ -L leapseconds -y yearistype $TZBASE 
-./zic -d %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/zoneinfo/posix/ -L /dev/null -y yearistype $TZBASE 
-./zic -d %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/zoneinfo/ -L /dev/null -y yearistype $TZBASE 
-cp %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/zoneinfo/America/New_York %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/zoneinfo/posixrules
+./zic -d %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/zoneinfo/right/ -L leapseconds -y yearistype $TZBASE 
+./zic -d %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/zoneinfo/posix/ -L /dev/null -y yearistype $TZBASE 
+./zic -d %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/zoneinfo/ -L /dev/null -y yearistype $TZBASE 
+cp %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/zoneinfo/America/New_York %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/zoneinfo/posixrules
 cd ..
 
 # Fix up libc.so and libpthread.so to use relative paths.
 # libgcc_s.so doesn't appear to need this treatment.
 for lib in libc libpthread ; do
-  cat %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/${lib}.so \
+  cat %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/${lib}.so \
 	| sed 's# /[^ ]*/lib# lib#g' \
-	> %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/${lib}.so.new
-  mv -f %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/${lib}.so.new \
-	%{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/${lib}.so
+	> %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/${lib}.so.new
+  mv -f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/${lib}.so.new \
+	%{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/${lib}.so
 done
 
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}
-mv %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/locale/locale.alias %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/locale.alias
-ln -s ../../..%{_stm_target_sysconf_dir}/locale.alias %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/locale/locale.alias
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}
+mv %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/locale/locale.alias %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/locale.alias
+ln -s ../../..%{_hhl_target_sysconf_dir}/locale.alias %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/locale/locale.alias
 
 cd ..
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man3
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3
 
-install -m 644 nscd/nscd.conf %{buildroot}%{_stm_cross_target_dir}/etc/nscd.conf
-install -m 644 nss/nsswitch.conf %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/
-echo "order hosts,bind" > %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/host.conf
+install -m 644 nscd/nscd.conf %{buildroot}%{_hhl_cross_target_dir}/etc/nscd.conf
+install -m 644 nss/nsswitch.conf %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/
+echo "order hosts,bind" > %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/host.conf
 
-install -m 755 debian/local/usr_sbin/tzconfig   %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-install -m 644 debian/local/manpages/tzconfig.8 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
+install -m 755 debian/local/usr_sbin/tzconfig   %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+install -m 644 debian/local/manpages/tzconfig.8 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
 
-install -m 755 debian/debhelper.in/nscd.init    %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d/nscd
+install -m 755 debian/debhelper.in/nscd.init    %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d/nscd
 
 # Generate the ld.so.conf and ld.so.cache files
-touch %{buildroot}%{_stm_cross_target_dir}/etc/ld.so.conf
-touch %{buildroot}%{_stm_cross_target_dir}/etc/ld.so.cache
+touch %{buildroot}%{_hhl_cross_target_dir}/etc/ld.so.conf
+touch %{buildroot}%{_hhl_cross_target_dir}/etc/ld.so.cache
 
 # ST231 build puts info info target/usr/info instead of target/usr/share/info for some as yet unknown reason...
-if [ -d %{buildroot}%{_stm_cross_target_dir}/usr/info ] ; then
-	mv %{buildroot}%{_stm_cross_target_dir}/usr/info %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}
+if [ -d %{buildroot}%{_hhl_cross_target_dir}/usr/info ] ; then
+	mv %{buildroot}%{_hhl_cross_target_dir}/usr/info %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}
 fi
 
 # Remove files we don't package to keep RPM happy
-rm -rf %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir
+rm -rf %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir*
 
-%compress_man %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}
-%compress_man %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
+%compress_man %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}
+%compress_man %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}
 
-mv %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/{zdump,rpcinfo} %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}/
+mv %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/{zdump,rpcinfo} %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/
 
 # This is complicated enough we need to build file lists....
 # glibc file list generation follows
 echo "%defattr(-,root,root)" > glibc.filelist
 
-for i in `ls %{buildroot}%{_stm_cross_target_dir}/lib/ld-%{glibc_version}.so | sed -e s,%{buildroot},,`; do
+for i in `ls %{buildroot}%{_hhl_cross_target_dir}/lib/ld-%{glibc_version}.so | sed -e s,%{buildroot},,`; do
  echo "%attr(0755,root,root) $i" >> glibc.filelist
 done
 
-for i in `ls %{buildroot}%{_stm_cross_target_dir}/lib/lib*-%{glibc_version}.so | sed -e s,%{buildroot},, | grep -v libc-%{glibc_version}.so`; do
+for i in `ls %{buildroot}%{_hhl_cross_target_dir}/lib/lib*-%{glibc_version}.so | sed -e s,%{buildroot},, | grep -v libc-%{glibc_version}.so`; do
  echo "%attr(0644,root,root) $i" >> glibc.filelist
 done
 
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}/lib/libc-%{glibc_version}.so" >> glibc.filelist
-echo "%attr(0644,root,root) %{_stm_cross_target_dir}/lib/libSegFault.so" >> glibc.filelist
-echo "%attr(0644,root,root) %{_stm_cross_target_dir}/lib/libthread_db-1.0.so" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}/lib/libc-%{glibc_version}.so" >> glibc.filelist
+echo "%attr(0644,root,root) %{_hhl_cross_target_dir}/lib/libSegFault.so" >> glibc.filelist
+echo "%attr(0644,root,root) %{_hhl_cross_target_dir}/lib/libthread_db-1.0.so" >> glibc.filelist
 
-if [ -f %{buildroot}%{_stm_cross_target_dir}/lib/libpthread-0.10.so ]; then
-echo "%attr(0644,root,root) %{_stm_cross_target_dir}/lib/libpthread-0.10.so" >> glibc.filelist
+if [ -f %{buildroot}%{_hhl_cross_target_dir}/lib/libpthread-0.10.so ]; then
+echo "%attr(0644,root,root) %{_hhl_cross_target_dir}/lib/libpthread-0.10.so" >> glibc.filelist
 fi
 
-for i in `find %{buildroot}%{_stm_cross_target_dir}/lib/ -type l -name '*.so.*' ! -name 'libnss*_*.so.1*' | sed -e s,%{buildroot},,`; do
+for i in `find %{buildroot}%{_hhl_cross_target_dir}/lib/ -type l -name '*.so.*' ! -name 'libnss*_*.so.1*' | sed -e s,%{buildroot},,`; do
  echo "$i" >> glibc.filelist
 done
 
-echo "%dir %{_stm_cross_target_dir}%{_stm_target_lib_dir}/gconv" >> glibc.filelist
-for i in `ls %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/gconv/* | sed -e s,%{buildroot},,`; do
+echo "%dir %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/gconv" >> glibc.filelist
+for i in `ls %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/gconv/* | sed -e s,%{buildroot},,`; do
  echo "%attr(0644,root,root) $i" >> glibc.filelist
 done
 
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/getent" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/iconv" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/locale" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/localedef" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/catchsegv" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/tzselect" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/ldd" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/rpcgen" >> glibc.filelist
-
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}/sbin/ldconfig" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}/sbin/sln" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/zdump" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/rpcinfo" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_sbin_dir}/iconvconfig" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_sbin_dir}/zic" >> glibc.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_sbin_dir}/tzconfig" >> glibc.filelist
-echo "%attr(4755,root,root) %{_stm_cross_target_dir}%{_stm_target_libexec_dir}/pt_chown" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/getent" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/iconv" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/locale" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/localedef" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/catchsegv" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/tzselect" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/ldd" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/rpcgen" >> glibc.filelist
+
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}/sbin/ldconfig" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}/sbin/sln" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/zdump" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/rpcinfo" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/iconvconfig" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/zic" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/tzconfig" >> glibc.filelist
+echo "%attr(4755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_libexec_dir}/pt_chown" >> glibc.filelist
 
-echo "%attr(0644,root,root) %doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/tzconfig.8.gz" >> glibc.filelist
+echo "%attr(0644,root,root) %doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/tzconfig.8.gz" >> glibc.filelist
 
-for i in `find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/zoneinfo -type d | sed -e s,%{buildroot},,`; do
+for i in `find %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/zoneinfo -type d | sed -e s,%{buildroot},,`; do
  echo "%dir $i" >> glibc.filelist
 done
-for i in `find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/zoneinfo/ ! -type d | sed -e s,%{buildroot},,`; do
+for i in `find %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/zoneinfo/ ! -type d | sed -e s,%{buildroot},,`; do
  echo "%attr(0644,root,root) $i" >> glibc.filelist
 done
 
-echo "%attr(0644,root,root) %config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/nsswitch.conf" >> glibc.filelist
-echo "%attr(0644,root,root) %config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/host.conf" >> glibc.filelist
+echo "%attr(0644,root,root) %config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/nsswitch.conf" >> glibc.filelist
+echo "%attr(0644,root,root) %config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/host.conf" >> glibc.filelist
 
-echo "%attr(0644,root,root) %config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/rpc" >> glibc.filelist
-echo "%attr(0644,root,root) %config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/ld.so.cache" >> glibc.filelist
-echo "%attr(0644,root,root) %config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/ld.so.conf" >> glibc.filelist
+echo "%attr(0644,root,root) %config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/rpc" >> glibc.filelist
+echo "%attr(0644,root,root) %config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/ld.so.cache" >> glibc.filelist
+echo "%attr(0644,root,root) %config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/ld.so.conf" >> glibc.filelist
 
 # glibc-dev file list follows
 echo "%defattr(-,root,root)" > glibc-dev.filelist
 
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/gencat" >> glibc-dev.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/getconf" >> glibc-dev.filelist
-# echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/mtrace" >> glibc-dev.filelist perl
-rm -f %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}/mtrace
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/gencat" >> glibc-dev.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/getconf" >> glibc-dev.filelist
+# echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/mtrace" >> glibc-dev.filelist perl
+rm -f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/mtrace
 
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/pcprofiledump" >> glibc-dev.filelist
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/xtrace" >> glibc-dev.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/pcprofiledump" >> glibc-dev.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/xtrace" >> glibc-dev.filelist
 
-echo "%attr(0644,root,root) %{_stm_cross_target_dir}/lib/libmemusage.so" >> glibc-dev.filelist
-echo "%attr(0644,root,root) %{_stm_cross_target_dir}/lib/libpcprofile.so" >> glibc-dev.filelist
+echo "%attr(0644,root,root) %{_hhl_cross_target_dir}/lib/libmemusage.so" >> glibc-dev.filelist
+echo "%attr(0644,root,root) %{_hhl_cross_target_dir}/lib/libpcprofile.so" >> glibc-dev.filelist
 
-for i in `ls %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.o | sed -e s,%{buildroot},,`; do
+for i in `ls %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.o | sed -e s,%{buildroot},,`; do
  echo "%attr(0644,root,root) $i" >> glibc-dev.filelist
 done
 
-for i in `ls %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.a | sed -e 's,%{buildroot},,;/^.*_.\.a$/d;/^.*_pic.a/d'`; do
+for i in `ls %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.a | sed -e 's,%{buildroot},,;/^.*_.\.a$/d;/^.*_pic.a/d'`; do
  echo "%attr(0644,root,root) $i" >> glibc-dev.filelist
 done
 
-echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libc.so" >> glibc-dev.filelist
+echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libc.so" >> glibc-dev.filelist
 # This should handle the links..
 # This group was removed..  at this point I'm assuming we DO want it.
 #  */libthread_db.so|*/libdb.so) ;;
 
-for i in `ls %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/lib*.so | sed -e s,%{buildroot},,`; do
+for i in `ls %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/lib*.so | sed -e s,%{buildroot},,`; do
  case "$i" in \
  *-%{glibc_version}.so | *-0.[789].so ) ;; \
  */libSegFault.so) ;; \
@@ -327,80 +327,80 @@
 done
 # Unlike debian we _WANT_ our links relative!
 
-for i in `find %{buildroot}%{_stm_cross_target_dir}%{_includedir} -type d | sed -e s,%{buildroot},,`; do
+for i in `find %{buildroot}%{_hhl_cross_target_dir}%{_includedir} -type d | sed -e s,%{buildroot},,`; do
  echo "%dir $i" >> glibc-dev.filelist
 done
 
-for i in `find %{buildroot}%{_stm_cross_target_dir}%{_includedir} ! -type d | sed -e s,%{buildroot},,`; do
+for i in `find %{buildroot}%{_hhl_cross_target_dir}%{_includedir} ! -type d | sed -e s,%{buildroot},,`; do
  echo "%attr(0644,root,root) $i" >> glibc-dev.filelist
 done
 
 # glibc-prof file list follows
 echo "%defattr(-,root,root)" > glibc-prof.filelist
 
-for i in `ls %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*_p.a | sed -e s,%{buildroot},,`; do
+for i in `ls %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*_p.a | sed -e s,%{buildroot},,`; do
  echo "%attr(0644,root,root) $i" >> glibc-prof.filelist
 done 
 
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_bin_dir}/sprof" >> glibc-prof.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/sprof" >> glibc-prof.filelist
 
 # glibc-pic file list follows
 echo "%defattr(-,root,root)" > glibc-pic.filelist
 
-# echo "%dir %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libc_pic" >> glibc-pic.filelist
-# echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libc_pic.a" >> glibc-pic.filelist
-# echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libc_pic.map" >> glibc-pic.filelist
+# echo "%dir %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libc_pic" >> glibc-pic.filelist
+# echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libc_pic.a" >> glibc-pic.filelist
+# echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libc_pic.map" >> glibc-pic.filelist
 # 
-# echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libc_pic/soinit.o" >> glibc-pic.filelist
-# echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libc_pic/sofini.o" >> glibc-pic.filelist
-# echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libc_pic/interp.o" >> glibc-pic.filelist
+# echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libc_pic/soinit.o" >> glibc-pic.filelist
+# echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libc_pic/sofini.o" >> glibc-pic.filelist
+# echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libc_pic/interp.o" >> glibc-pic.filelist
 # 
-# echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libm_pic.a" >> glibc-pic.filelist
-# echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libm_pic.map" >> glibc-pic.filelist
+# echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libm_pic.a" >> glibc-pic.filelist
+# echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libm_pic.map" >> glibc-pic.filelist
 # 
-# echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libresolv_pic.a" >> glibc-pic.filelist
-# echo "%attr(0644,root,root) %{_stm_cross_target_dir}%{_stm_target_lib_dir}/libresolv_pic.map" >> glibc-pic.filelist
+# echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libresolv_pic.a" >> glibc-pic.filelist
+# echo "%attr(0644,root,root) %{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libresolv_pic.map" >> glibc-pic.filelist
 
 # glibc-locales file list follows
 echo "%defattr(-,root,root)" > glibc-locales.filelist
 
-for i in `find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/locale -type d | sed -e s,%{buildroot},,`; do
+for i in `find %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/locale -type d | sed -e s,%{buildroot},,`; do
  echo "%dir $i" >> glibc-locales.filelist
 done
 
-for i in `find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/locale ! -type d | sed -e s,%{buildroot},,`; do
+for i in `find %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/locale ! -type d | sed -e s,%{buildroot},,`; do
  echo "%attr(0644,root,root) $i" >> glibc-locales.filelist
 done
 
-echo "%attr(0644,root,root) %config %{_stm_cross_target_dir}/etc/locale.alias" >> glibc-locales.filelist
+echo "%attr(0644,root,root) %config %{_hhl_cross_target_dir}/etc/locale.alias" >> glibc-locales.filelist
 
 # glibc-i18ndata file list follows
 echo "%defattr(-,root,root)" > glibc-i18ndata.filelist
 
-for i in `find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/i18n -type d | sed -e s,%{buildroot},,`; do
+for i in `find %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/i18n -type d | sed -e s,%{buildroot},,`; do
  echo "%dir $i" >> glibc-i18ndata.filelist
 done
 
-for i in `find %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/i18n ! -type d | sed -e s,%{buildroot},,`; do
+for i in `find %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/i18n ! -type d | sed -e s,%{buildroot},,`; do
  echo "%attr(0644,root,root) $i" >> glibc-i18ndata.filelist
 done
 
 # glibc-nscd file list follows
 echo "%defattr(-,root,root)" > glibc-nscd.filelist
 
-echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_sbin_dir}/nscd" >> glibc-nscd.filelist
-#echo "%attr(0755,root,root) %{_stm_cross_target_dir}%{_stm_target_sbin_dir}/nscd_nischeck" >> glibc-nscd.filelist
-echo "%attr(0644,root,root) %config %{_stm_cross_target_dir}/etc/nscd.conf" >> glibc-nscd.filelist
-echo "%attr(0755,root,root) %config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d/nscd" >> glibc.filelist
+echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/nscd" >> glibc-nscd.filelist
+#echo "%attr(0755,root,root) %{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/nscd_nischeck" >> glibc-nscd.filelist
+echo "%attr(0644,root,root) %config %{_hhl_cross_target_dir}/etc/nscd.conf" >> glibc-nscd.filelist
+echo "%attr(0755,root,root) %config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d/nscd" >> glibc.filelist
 
 # glibc-doc file list follows
 echo "%defattr(-,root,root)" > glibc-doc.filelist
 
-for i in `ls %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}/*.info* | sed -e s,%{buildroot},,`; do
+for i in `ls %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/*.info* | sed -e s,%{buildroot},,`; do
  echo "%attr(0644,root,root) %doc $i" >> glibc-doc.filelist
 done 
 
-for i in `ls %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man3/* | sed -e s,%{buildroot},,`; do
+for i in `ls %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3/* | sed -e s,%{buildroot},,`; do
  echo "%attr(0644,root,root) %doc $i" >> glibc-doc.filelist
 done
 
@@ -451,12 +451,12 @@
 %doc COPYING
 
 %post -n %{_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir \
-	%{_stm_cross_target_dir}%{_stm_target_info_dir}/libc.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir \
+	%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/libc.info
 
 %preun -n %{_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir --remove \
-	%{_stm_cross_target_dir}%{_stm_target_info_dir}/libc.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir --remove \
+	%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/libc.info
 
 %changelog
 * Fri Oct 20 2006 Carl Shaw <carl.shaw@st.com> 27
@@ -524,7 +524,7 @@
 * Thu Dec  9 2004 Stuart Menefy <stuart.menefy@st.com>
 - Updated for STtoolbuilder
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Tue Sep  7 2004 Stuart Menefy <stuart.menefy@st.com>
 - Updated Debian patch to glibc_2.3.2.ds1-13.
 
