--- old/stm-target-alsa-lib-sh4processed.spec	2010-02-21 10:32:48.079836417 +0100
+++ new/stm-target-alsa-lib-sh4processed.spec	2010-02-21 11:38:02.701132666 +0100
@@ -39,12 +39,13 @@
 
 %build
 %target_setup
+libtoolize -f -c
+autoreconf -f -i
 ./configure \
 	--host=%{_stm_target_config} \
         --target=%{_stm_target_config} \
         --prefix=%{_stm_target_prefix} \
         --exec-prefix=%{_stm_target_exec_prefix} \
-	--program-transform-name=s,x,x \
 	--with-debug=no
 %make
 
