--- SPECS/stm-target-makedev.spec	2005-09-30 17:07:46.000000000 +0200
+++ SPECS/stm-target-makedev.spec-new	2007-06-16 12:23:10.000000000 +0200
@@ -90,8 +90,7 @@
 
 if [ `whoami` != "root" ] ; then
 	echo "You are not root."
-	echo "Not populating ${RPM_INSTALL_PREFIX}/dev."
-	exit 0
+	echo "Populating ${RPM_INSTALL_PREFIX}/dev with sudo rights."
 fi
 
 # Used to not make the system go nuts if something wrong happens
@@ -99,37 +98,38 @@
 if [ -d ${RPM_INSTALL_PREFIX}/dev ] ; then
 	echo Making devices in ${RPM_INSTALL_PREFIX}/dev. This may take a while!
 	cd ${RPM_INSTALL_PREFIX}/dev
+	echo Is populating for ${RPM_INSTALL_PREFIX}/dev ok?
 
 	MAKEDEV="./MAKEDEV -p ${RPM_INSTALL_PREFIX}/etc/passwd -g ${RPM_INSTALL_PREFIX}/etc/group"
-	${MAKEDEV} generic-sh
-	rm -f tty[2-9][0-9]
-	rm -f vcs[2-9][0-9]
-	rm -f vcsa[2-9][0-9]
-	${MAKEDEV} ttyAS0 ttyAS1 ttyAS2 ttyAS3
-	${MAKEDEV} ttyFW0 ttyFW1 ttyFW2 ttyFW3
-	${MAKEDEV} ide0 ide1
-	${MAKEDEV} random
-	${MAKEDEV} urandom
-	${MAKEDEV} sound
-	rm -f tty[r-za-e][0-9a-f]
-	rm -f pty[r-za-e][0-9a-f]
-	rm -rf pts
-	mkdir pts
-	chown root.root pts
-	chmod 0755 pts
-	rm -rf shm
-	mkdir shm
-	chown root.root shm
-	chmod 0755 shm
-	${MAKEDEV} ppp
-	${MAKEDEV} busmice
-	ln -s -f /dev/input/mouse0 mouse
-	${MAKEDEV} input
-	${MAKEDEV} mtd
-	${MAKEDEV} i2c
-	${MAKEDEV} lirc
-	${MAKEDEV} st200
-	${MAKEDEV} alsasnd
+	sudo ${MAKEDEV} generic-sh
+	sudo rm -f tty[2-9][0-9]
+	sudo rm -f vcs[2-9][0-9]
+	sudo rm -f vcsa[2-9][0-9]
+	sudo ${MAKEDEV} ttyAS0 ttyAS1 ttyAS2 ttyAS3
+	sudo ${MAKEDEV} ttyFW0 ttyFW1 ttyFW2 ttyFW3
+	sudo ${MAKEDEV} ide0 ide1
+	sudo ${MAKEDEV} random
+	sudo ${MAKEDEV} urandom
+	sudo ${MAKEDEV} sound
+	sudo rm -f tty[r-za-e][0-9a-f]
+	sudo rm -f pty[r-za-e][0-9a-f]
+	sudo rm -rf pts
+	sudo mkdir pts
+	sudo chown root.root pts
+	sudo chmod 0755 pts
+	sudo rm -rf shm
+	sudo mkdir shm
+	sudo chown root.root shm
+	sudo chmod 0755 shm
+	sudo ${MAKEDEV} ppp
+	sudo ${MAKEDEV} busmice
+	sudo ln -s -f /dev/input/mouse0 mouse
+	sudo ${MAKEDEV} input
+	sudo ${MAKEDEV} mtd
+	sudo ${MAKEDEV} i2c
+	sudo ${MAKEDEV} lirc
+	sudo ${MAKEDEV} st200
+	sudo ${MAKEDEV} alsasnd
 fi
 
 %files -n %{_pkgname}
