--- stm-host-glib2.spec.org	2012-06-27 15:47:49.998316147 +0200
+++ stm-host-glib2.spec	2012-06-27 15:48:17.606317292 +0200
@@ -10,7 +10,6 @@
 Buildroot:	%(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
 Prefix:		%{_stm_install_prefix}
 BuildRequires:	%{_stm_pkg_prefix}-host-rpmconfig
-BuildRequires:	%{_stm_pkg_prefix}-host-zlib-dev >= 1.2.3
 BuildRequires:	%{_stm_pkg_prefix}-host-libffi-dev
 
 AutoReq:	no
 

