--- SPECS/stm-cross-gcc-sh4processed.spec	2007-06-23 12:41:29.000000000 +0200
+++ SPECS/stm-cross-gcc-sh4processed.spec-new	2007-06-23 12:44:24.000000000 +0200
@@ -240,7 +240,7 @@
    --disable-multilib \
         --enable-symvers=gnu \
         --enable-__cxa_atexit \
-        --with-gxx-include-dir="\\\${prefix}/target%{_hhl_target_include_dir}/c++/%{version}" \
+        --with-gxx-include-dir="\\\${prefix}/../../cdkroot%{_hhl_target_include_dir}/c++/%{version}" \
 %if "%{_hhl_target_fpu}" == "no"
         --without-fp \
 %endif
@@ -332,7 +332,7 @@
 rm %{buildroot}%{_hhl_cross_info_dir}/cppinternals.info.gz
 rm %{buildroot}%{_hhl_cross_info_dir}/gccinstall.info.gz
 rm %{buildroot}%{_hhl_cross_info_dir}/gccint.info.gz
-rm -f %{buildroot}%{_hhl_cross_info_dir}/dir.gz
+rm -f %{buildroot}%{_hhl_cross_info_dir}/dir*
 rm -r %{buildroot}%{_hhl_cross_lib_dir}/gcc/%{_hhl_target_config}/%{gcc_version}/install-tools
 rm    %{buildroot}%{_hhl_cross_lib_dir}/libiberty.a
 rm -r %{buildroot}%{_hhl_cross_libexec_dir}/gcc/%{_hhl_target_config}/%{gcc_version}/install-tools
