--- SPECS/stm-target-distributionutils.spec	2005-09-23 22:30:09.000000000 +0200
+++ SPECS/stm-target-distributionutils.spec-new	2007-07-21 08:49:46.000000000 +0200
@@ -62,9 +62,8 @@
 
 %install
 rm -rf %{buildroot}
-%target_setup
-%target_makeinstall
-cd ..
+
+install -d %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
 
 install -m 755 shellconfig/shellconfig %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/.
 ln -s shellconfig %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/%{_hhl_target_toolprefix}shellconfig
@@ -72,28 +71,7 @@
 install -m 755 initdconfig/initdconfig %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/.
 ln -s initdconfig %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/%{_hhl_target_toolprefix}initdconfig
 
-mkdir -p %{buildroot}%{_hhl_cross_target_dir}/sbin
-install -m 755 start-stop-daemon/start-stop-daemon %{buildroot}%{_hhl_cross_target_dir}/sbin/.
-install -m 644 start-stop-daemon/start-stop-daemon.8 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
-
-mkdir -p %{buildroot}%{_hhl_cross_target_dir}/bin
-install -m 755 usleep/usleep %{buildroot}%{_hhl_cross_target_dir}/bin/.
-
 # Remove files we install but don't distribute
-rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/which
-rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/which.1
-rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/fr/man1/which.1
-
-rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/installkernel
-rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/mkboot
-rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/installkernel.8
-rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/mkboot.8
-
-rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/sensible-*
-rm %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/sensible-*
-
-gzip -9 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man*/*
-gzip -9 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/fr/man*/*
 
 cp debian/copyright LICENSE
 
@@ -102,12 +80,6 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_hhl_cross_target_dir}/bin/usleep
-%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/run-parts
-%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/tempfile
-%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/mktemp
-%{_hhl_cross_target_dir}/sbin/start-stop-daemon
-%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/savelog
 
 %{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/shellconfig
 %{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/%{_hhl_target_toolprefix}shellconfig
@@ -118,14 +90,7 @@
 
 %files -n %{_pkgname}-doc
 %defattr(-,root,root)
-%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/mktemp.1.gz
-%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1/tempfile.1.gz
-# %doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/installkernel.8.gz
-# %doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/mkboot.8.gz
-%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/run-parts.8.gz
-%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/savelog.8.gz
-%doc %lang(fr) %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/fr/man8/savelog.8.gz
-%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/start-stop-daemon.8.gz
+
 %doc LICENSE
 
 %changelog
