--- SPECS/stm-target-netkit-ftp.spec	2007-02-08 18:20:03.000000000 +0100
+++ SPECS/stm-target-netkit-ftp.spec-new	2007-07-09 20:29:54.000000000 +0200
@@ -1,5 +1,5 @@
 Summary: The FTP client.
-Name: %{_stm_pkg_prefix}-target-netkit-ftp
+Name: %{_hhl_pkg_prefix}-target-netkit-ftp
 Version: 0.17
 Release: 7
 Source: netkit-ftp_%{version}.orig.tar.gz
@@ -8,10 +8,10 @@
 Patch2: netkit-ftp_%{version}-makefile.patch
 License: GPL
 Group: Applications/Internet
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir	%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-netkit-ftp
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir	%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-netkit-ftp
 
 %description
 ftp is the user interface to the ARPANET standard File Transfer Protocol.
@@ -43,30 +43,31 @@
 %patch2 -p1
 %build
 %target_setup
+rm -rf %{buildroot}
 # configure script is non-autoconf
-./configure --with-c-compiler=$CC --installroot=%{?buildroot:%{buildroot}}%{_stm_cross_target_dir} --prefix=%{_stm_target_prefix} --exec-prefix=%{_stm_target_exec_prefix}
+./configure --with-c-compiler=$CC --installroot=%{?buildroot:%{buildroot}}%{_hhl_cross_target_dir} --prefix=%{_hhl_target_prefix} --exec-prefix=%{_hhl_target_exec_prefix}
 
 %make all
 
 %install
 %target_setup
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_prefix}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_exec_prefix}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_include_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}/usr/man/man{1,2,3,4,5,6,7,8}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_prefix}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_exec_prefix}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_include_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}/usr/man/man{1,2,3,4,5,6,7,8}
 make install
-pushd %{buildroot}%{_stm_cross_target_dir}/usr/man
-tar cf - . | (cd %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}; tar xf - )
+pushd %{buildroot}%{_hhl_cross_target_dir}/usr/man
+tar cf - . | (cd %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}; tar xf - )
 popd
-rm -rf %{buildroot}%{_stm_cross_target_dir}/usr/man
-(gzip -r %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir} || true)
+rm -rf %{buildroot}%{_hhl_cross_target_dir}/usr/man
+(gzip -r %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir} || true)
 cp debian/copyright LICENSE
 #  Fix link
-cd %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man1
+cd %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man1
 rm -f pftp.1
 ln -s ftp.1.gz pftp.1.gz
 
@@ -75,11 +76,11 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/*
 %doc LICENSE
 %files -n %{_pkgname}-doc
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man*/*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man*/*
 %doc BUGS ChangeLog README
 %doc LICENSE
 
@@ -99,7 +100,7 @@
 * Tue Feb 22 2005 Carl Shaw <carl.shaw@st.com>
 - Changed man copying section
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Thu Aug 15 2002 Stuart Menefy <stuart.menefy@st.com>
 - No change, need to force rebuild to pick up readline library
 
