--- SPECS/stm-target-netkit-telnet.spec	2007-02-08 18:20:07.000000000 +0100
+++ SPECS/stm-target-netkit-telnet.spec-new	2007-09-20 08:43:20.000000000 +0200
@@ -1,5 +1,5 @@
 Summary: remote shell tools
-Name: %{_stm_pkg_prefix}-target-netkit-telnet
+Name: %{_hhl_pkg_prefix}-target-netkit-telnet
 Version: 0.17
 Release: 7
 Source: netkit-telnet_%{version}.orig.tar.gz
@@ -7,10 +7,10 @@
 Patch1: netkit-telnet_%{version}-cross.diff.gz
 License: GPL
 Group: Applications/Internet
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir	%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-netkit-telnet
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir	%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-netkit-telnet
 
 %description
 Clients and servers for the telnet protocol
@@ -41,28 +41,28 @@
 %build
 %target_setup
 # configure script is non-autoconf
-./configure --with-c-compiler=$CC --installroot=%{?buildroot:%{buildroot}}%{_stm_cross_target_dir} --prefix=%{_stm_target_prefix} --exec-prefix=%{_stm_target_exec_prefix}
+./configure --with-c-compiler=$CC --installroot=%{?buildroot:%{buildroot}}%{_hhl_cross_target_dir} --prefix=%{_hhl_target_prefix} --exec-prefix=%{_hhl_target_exec_prefix}
 
 %make all LDFLAGS=""
 
 %install
 %target_setup
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_prefix}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_exec_prefix}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_include_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}/usr/man/man{1,2,3,4,5,6,7,8}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_prefix}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_exec_prefix}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_include_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}/usr/man/man{1,2,3,4,5,6,7,8}
 make install
-pushd %{buildroot}%{_stm_cross_target_dir}/usr/man
-tar cf - . | (cd %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}; tar xf - )
+pushd %{buildroot}%{_hhl_cross_target_dir}/usr/man
+tar cf - . | (cd %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}; tar xf - )
 popd
-rm -rf %{buildroot}%{_stm_cross_target_dir}/usr/man
+rm -rf %{buildroot}%{_hhl_cross_target_dir}/usr/man
 
-mv %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/telnetlogin \
-	%{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/.
+mv %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/telnetlogin \
+	%{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/.
 cp debian/copyright LICENSE
 
 %clean
@@ -70,21 +70,21 @@
 
 %files -n %{_pkgname}-client
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/*
 
 %doc LICENSE
 
 # rexecd, rlogind and telnetd.
 %files -n %{_pkgname}-server
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/*
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*
 
 %doc LICENSE
 
 %files -n %{_pkgname}-doc
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man*/*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man*/*
 %doc BUGS ChangeLog README
 
 %doc LICENSE
@@ -102,6 +102,6 @@
 * Tue Feb 22 2005 Carl Shaw <carl.shaw@st.com>
 - Changed man copying section
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Wed Jan 24 2002 Stuart Menefy <stuart.menefy@st.com>
 - Copied from MontaVista version 2.0.3
