--- SPECS/stm-target-glibc.spec.orig	2010-05-23 20:59:18.000000000 +0200
+++ SPECS/stm-target-glibc.spec	2010-05-23 21:00:12.000000000 +0200
@@ -23,6 +23,7 @@
 Patch1: glibc-2.6-tzconfig.patch
 Patch2: glibc-2.3.6-nptl-config.patch
 Patch3: glibc-2.6.1-alloca_fix.patch
+Patch4: stm-target-glibc-sysincludes.patch
 
 # SH4
 Patch100: glibc-2.5-ustat.patch
@@ -160,6 +161,7 @@
 %patch1 -p1
 %patch2 -p1
 %patch3 -p1
+%patch4 -p1
 
 %if_target_arch sh4 
 %patch100 -p1
