--- SPECS/stm-target-util-linux.spec	2007-06-16 13:38:36.000000000 +0200
+++ SPECS/stm-target-util-linux.spec-new	2007-06-16 13:43:18.000000000 +0200
@@ -1,5 +1,5 @@
 Summary: Miscellaneous system utilities.
-Name: %{_stm_pkg_prefix}-target-util-linux
+Name: %{_hhl_pkg_prefix}-target-util-linux
 Version: 2.12m
 Release: 11
 License: GPL
@@ -11,21 +11,21 @@
 Patch3: util-linux-2.12m-swapon.patch
 Patch4: util-linux-2.12m-st200.patch
 Patch5: util-linux-2.12m-arm-sh4-pagesize.patch
-BuildRoot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-%define targetroot %{_stm_cross_target_dir}
+BuildRoot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+%define targetroot %{_hhl_cross_target_dir}
 Prefix: %{targetroot}
-%define _pkgname	%{_stm_pkg_prefix}-%{_stm_target_arch}-util-linux
-%define pkgmount	%{_stm_pkg_prefix}-%{_stm_target_arch}-mount
-%define pkgbsdutils	%{_stm_pkg_prefix}-%{_stm_target_arch}-bsdutils
-
-%define _bindir		%{targetroot}%{_stm_target_bin_dir}
-%define _sbindir	%{targetroot}%{_stm_target_sbin_dir}
-%define _etcdir		%{targetroot}%{_stm_target_sysconf_dir}
-%define _libdir		%{targetroot}%{_stm_target_lib_dir}
-%define _datadir	%{targetroot}%{_stm_target_sharedstate_dir}
-%define _mandir		%{targetroot}%{_stm_target_man_dir}
-%define _infodir	%{targetroot}%{_stm_target_info_dir}
-%define _docdir		%{targetroot}%{_stm_target_doc_dir}
+%define _pkgname	%{_hhl_pkg_prefix}-%{_hhl_target_arch}-util-linux
+%define pkgmount	%{_hhl_pkg_prefix}-%{_hhl_target_arch}-mount
+%define pkgbsdutils	%{_hhl_pkg_prefix}-%{_hhl_target_arch}-bsdutils
+
+%define _bindir		%{targetroot}%{_hhl_target_bin_dir}
+%define _sbindir	%{targetroot}%{_hhl_target_sbin_dir}
+%define _etcdir		%{targetroot}%{_hhl_target_sysconf_dir}
+%define _libdir		%{targetroot}%{_hhl_target_lib_dir}
+%define _datadir	%{targetroot}%{_hhl_target_sharedstate_dir}
+%define _mandir		%{targetroot}%{_hhl_target_man_dir}
+%define _infodir	%{targetroot}%{_hhl_target_info_dir}
+%define _docdir		%{targetroot}%{_hhl_target_doc_dir}
 
 %description
 A mixed bag of system utilities: arch cytune dmesg fdisk
@@ -83,13 +83,13 @@
 sed 's/CURSESFLAGS=.*/CURSESFLAGS=-DNCH=1/' make_include > make_include.new
 mv make_include make_include.bak
 mv make_include.new make_include
-%make ARCH=%{_stm_target_arch} %{makeflags}
+%make ARCH=%{_hhl_target_arch} %{makeflags}
 
 %install
 %target_setup
 rm -rf %{buildroot}
 
-make ARCH=%{_stm_target_arch} %{makeflags} \
+make ARCH=%{_hhl_target_arch} %{makeflags} \
 	USE_TTY_GROUP=no INSTALLSUID='$(INSTALL) -m $(SUIDMODE)' \
 	DESTDIR=%{buildroot}%{targetroot} \
 	install
@@ -292,12 +292,12 @@
 %doc debian/README.script LICENSE
 
 %post -n %{_pkgname}
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir \
-	%{_stm_cross_target_dir}%{_stm_target_info_dir}/ipc.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir \
+	%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/ipc.info
 
 %preun -n %{_pkgname}
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir --remove \
-	%{_stm_cross_target_dir}%{_stm_target_info_dir}/info.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir --remove \
+	%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/info.info
 
 
 %changelog
@@ -322,7 +322,7 @@
 * Wed Dec 22 2004 Stuart Menefy <stuart.menefy@st.com>
 - Updated to 2.12m
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Tue Nov 04 2003 Stuart Menefy <stuart.menefy@st.com>
 - Removed last and mesg from the install list, already part of SysVinit
   Also sln as part of glibc
