--- stm-host-rpmconfig.spec.orig	2013-04-25 12:49:56.395332915 +0200
+++ stm-host-rpmconfig.spec	2013-04-25 12:49:56.375332920 +0200
@@ -1,3 +1,13 @@
+# These six macros would normally be defined in config/rpm/common,
+# but as we are currently building that file define our own version here.
+
+%define _stm_build_id		2.4
+%define _stm_short_build_id	24
+%define _stm_install_prefix	/opt/STM
+%define _stm_pkg_prefix	stlinux24
+%define __find_requires	/usr/lib/rpm/find-requires
+%define __find_provides	/usr/lib/rpm/find-provides
+
 %define _build_cpu		sh4_uclibc
 %define _stm_base_prefix	%{_stm_install_prefix}/STLinux-%{_stm_build_id}
 %define _stm_config_dir		%{_stm_base_prefix}/config
@@ -11,6 +21,8 @@
 Version: %{_stm_build_id}
 Release: 33
 Source0: stm-host-rpmconfig-%{version}-%{_build_cpu}.tar.gz
+Patch0: stm-host-rpmconfig-%{version}-%{release}-ignore-skip-cvs-errors.patch
+Patch1: stm-host-rpmconfig-%{version}-%{release}-autoreconf-add-libtool-macros.patch
 License: GPL
 Group: Development/Tools
 BuildRoot: %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
@@ -26,24 +38,13 @@
 
 %prep
 %setup -q -n rpmconfig-%{version}
+%patch0 -p1
+%patch1 -p1
 
 %install
 make	RPMCONFIGDIR=%{buildroot}%{_stm_config_dir}/rpm \
 	install
 
-%clean
-rm -rf %{buildroot}
-
-
-%post
-if [ ${RPM_INSTALL_PREFIX} != %{_stm_install_prefix} ]; then
- RELOCATION=${RPM_INSTALL_PREFIX}`expr %{_stm_config_dir}/rpm : '%{_stm_install_prefix}\(.*$\)'`
- cd $RELOCATION
- for i in common ; do
-   sed "s,%{_stm_install_prefix},${RPM_INSTALL_PREFIX}," -i $i
- done
-fi
-
 # create config.cache files
 for arch in armv5 armv5_uclibc armv6 armv6_uclibc armv7 armv7_uclibc sh4 sh4_uclibc st231 st240 ; do
   sed -e "s,@PREFIX@,%{_stm_target_prefix},g" \
@@ -66,10 +67,23 @@
     -e "s,@PKGCONFIGDIR@,%{_stm_target_pkgconfig_dir},g" \
     -e "s,@X11FONTDIR@,%{_stm_target_x11font_dir},g" \
     -e "s,@TESTSDIR@,%{_stm_target_tests_dir},g" \
-    < %{_stm_config_dir}/rpm/config.site.template > %{_stm_config_dir}/config.site.${arch}
-  cat %{_stm_config_dir}/rpm/site_configs/config.site.${arch} >> %{_stm_config_dir}/config.site.${arch}
+    < %{buildroot}%{_stm_config_dir}/rpm/config.site.template > %{buildroot}%{_stm_config_dir}/config.site.${arch}
+  cat %{buildroot}%{_stm_config_dir}/rpm/site_configs/config.site.${arch} >> %{buildroot}%{_stm_config_dir}/config.site.${arch}
 done
 
+%clean
+rm -rf %{buildroot}
+
+
+%post
+if [ ${RPM_INSTALL_PREFIX} != %{_stm_install_prefix} ]; then
+ RELOCATION=${RPM_INSTALL_PREFIX}`expr %{_stm_config_dir}/rpm : '%{_stm_install_prefix}\(.*$\)'`
+ cd $RELOCATION
+ for i in common ; do
+   sed "s,%{_stm_install_prefix},${RPM_INSTALL_PREFIX}," -i $i
+ done
+fi
+
 %files
 %defattr(-,root,root)
 %dir %{_stm_config_dir}/rpm
@@ -86,6 +100,7 @@
 %{_stm_config_dir}/rpm/hosts/*
 %{_stm_config_dir}/rpm/targets/*
 %{_stm_config_dir}/rpm/site_configs/*
+%{_stm_config_dir}/config.site.*
 
 %changelog
 * Thu Mar 07 2013 Francesco Virlinzi <francesco.virlinzi@st.com> 33
