--- SPECS/stm-cross-binutils-sh4processed.spec	2006-10-12 22:27:32.000000000 +0200
+++ SPECS/stm-cross-binutils-sh4processed.spec-new	2008-01-18 07:26:21.000000000 +0100
@@ -13,6 +13,7 @@
 Summary: The GNU assembler, linker and binary utilities.
 Name: %{_stm_pkg_prefix}-cross-binutils
 Source0: binutils-%{binutils_version}.tar.bz2
+Patch: stm-cross-binutils.diff
 
 Group: Development/Languages 
 %description
@@ -59,6 +60,7 @@
 %prep
 %target_check
 %setup -n binutils-%{binutils_version}
+%patch -p1
 #@arm@%patch0 -p1
 #@sh4@%patch0 -p1
 
--- SPECS/stm-cross-binutils-sh4processed.spec	2008-01-18 22:17:50.000000000 +0100
+++ SPECS/stm-cross-binutils-sh4processed.spec-new	2008-01-18 22:26:55.000000000 +0100
@@ -120,18 +120,18 @@
 rm -rf %{buildroot}/%{_stm_cross_dir}/%{_stm_host_config}
 
 cd $srcdir
-%find_lang binutils
-%find_lang opcodes
-%find_lang bfd
-%find_lang gas
-%find_lang ld
-(%find_lang gprof || touch gprof.lang)
-
-cat opcodes.lang >> binutils.lang
-cat bfd.lang >> binutils.lang
-cat gas.lang >> binutils.lang
-cat ld.lang >> binutils.lang
-cat gprof.lang >> binutils.lang
+%find_lang binutils || true
+%find_lang opcodes || true
+%find_lang bfd || true
+%find_lang gas || true
+%find_lang ld || true
+(%find_lang gprof || touch gprof.lang) || true
+
+cat opcodes.lang >> binutils.lang || true
+cat bfd.lang >> binutils.lang || true
+cat gas.lang >> binutils.lang || true
+cat ld.lang >> binutils.lang || true
+cat gprof.lang >> binutils.lang || true
 
 #  Remove some gprof info files not required
 rm -f %{buildroot}/%{_stm_cross_man_dir}/man1/st231-linux-gprof.1
--- SPECS/stm-cross-binutils-sh4processed.spec	2008-06-30 07:29:34.000000000 +0200
+++ SPECS/stm-cross-binutils-sh4processed.spec-new	2008-06-30 07:34:56.000000000 +0200
@@ -78,6 +78,7 @@
        --enable-nls \
        --disable-multilib \
        --enable-install-libbfd \
+       --disable-werror \
        --with-sysroot=%{_stm_cross_target_dir}
 make
 
