--- SPECS/stm-target-freetype.spec	2007-07-14 11:34:33.000000000 +0200
+++ SPECS/stm-target-freetype.spec-new	2007-07-14 11:35:27.000000000 +0200
@@ -1,7 +1,7 @@
 Name:		%{_hhl_pkg_prefix}-target-freetype
 Version:	2.1.9
 Release: 7
-Copyright:	BSD-like
+License:	BSD-like
 Source:		http://prdownloads.sourceforge.net/freetype/freetype-%{version}.tar.bz2
 Source1:	http://prdownloads.sourceforge.net/freetype/ft2demos-%{version}.tar.bz2
 URL:		http://www.freetype.org/
@@ -186,6 +186,8 @@
 ../freetype2/builds/unix/libtool --mode=install cp -av \
 	bin/[a-z]* %{buildroot}%{_hhl_cross_target_dir}/usr/X11R6/bin
 
+rm %{buildroot}%{_hhl_cross_target_dir}/usr/X11R6/bin/README || true
+
 #-- CLEAN ------------------------------------------------------------
 
 %clean
