--- SPECS/stm-target-tcp-wrappers.spec	2007-02-08 18:21:29.000000000 +0100
+++ SPECS/stm-target-tcp-wrappers.spec-new	2007-06-16 13:21:56.000000000 +0200
@@ -1,5 +1,5 @@
 Summary: Wietse Venema's TCP wrappers library
-Name: %{_stm_pkg_prefix}-target-tcp-wrappers
+Name: %{_hhl_pkg_prefix}-target-tcp-wrappers
 Version: 7.6
 Release: 6
 Source: ftp://coast.cs.purdue.edu/pub/tools/unix/netutils/tcp_wrappers/tcp_wrappers_7.6.tar.gz
@@ -10,10 +10,10 @@
 Patch10: tcp-wrappers-owner.diff
 License: Other
 Group: System Environment/Daemons
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _docdir 	%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-tcp-wrappers
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _docdir 	%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-tcp-wrappers
 
 %description
 Wietse Venema's network logger, also known as TCPD or LOG_TCP.
@@ -41,7 +41,7 @@
 %package -n %{_pkgname}-dev
 Summary: Wietse Venema's TCP wrappers library, development files
 Group: Development/Libraries
-Requires: %{_stm_pkg_prefix}-%{_stm_target_arch}-tcp-wrappers = %{version}
+Requires: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-tcp-wrappers = %{version}
 %description -n %{_pkgname}-dev
 Wietse Venema's network logger, also known as TCPD or LOG_TCP.
 
@@ -62,53 +62,63 @@
 %build
 %target_setup
 
-%make 	REAL_DAEMON_DIR=%{_stm_target_sbin_dir} \
+%make 	REAL_DAEMON_DIR=%{_hhl_target_sbin_dir} \
 	LIBS=-lnsl \
 	ARFLAGS=rv AUX_OBJ=weak_symbols.o \
 	NETGROUP=-DNETGROUP TLI= VSYSLOG= BUGS= \
 	EXTRA_CFLAGS="-DSYS_ERRLIST_DEFINED -DHAVE_STRERROR -DHAVE_WEAKSYMS -D_REENTRANT -DINET6=1" \
-	CC=%{_stm_target_toolprefix}gcc \
+	CC=%{_hhl_target_toolprefix}gcc \
 	all
 
 %install
 %target_setup
 
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man3
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man5
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_include_dir}
-
-install -m 0644 shared/libwrap.so.0.7.6 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
-ln -s libwrap.so.0.7.6 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libwrap.so.0
-ln -s libwrap.so.0 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libwrap.so
-
-install -m 0644 libwrap.a %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
-install -m 0644 tcpd.h %{buildroot}%{_stm_cross_target_dir}%{_stm_target_include_dir}
-
-install -m 0755 tcpd %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-install -m 0755 tcpdchk %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-install -m 0755 tcpdmatch %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-install -m 0755 try-from %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-install -m 0755 safe_finger %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir}
-
-install -m 0644 hosts_access.3 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man3/
-
-install -m 0644 hosts_access.5 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man5/
-install -m 0644 hosts_options.5 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man5/
-
-install -m 0644 tcpd.8 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-install -m 0644 tcpdchk.8 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-install -m 0644 tcpdmatch.8 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-install -m 0644 extra/try-from.8 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-install -m 0644 extra/safe_finger.8 %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man8
-
-%target_install_fixup
-
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}
-cp %{SOURCE1} %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/hosts.allow
-cp %{SOURCE2} %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/hosts.deny
+rm -rf %{buildroot}
+
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_include_dir}
+
+install -m 0644 shared/libwrap.so.0.7.6 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
+ln -s libwrap.so.0.7.6 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libwrap.so.0
+ln -s libwrap.so.0 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libwrap.so
+
+install -m 0644 libwrap.a %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
+install -m 0644 tcpd.h %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_include_dir}
+
+install -m 0755 tcpd %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+install -m 0755 tcpdchk %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+install -m 0755 tcpdmatch %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+install -m 0755 try-from %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+install -m 0755 safe_finger %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}
+
+install -m 0644 hosts_access.3 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3/
+
+install -m 0644 hosts_access.5 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/
+install -m 0644 hosts_options.5 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/
+
+install -m 0644 tcpd.8 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
+install -m 0644 tcpdchk.8 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
+install -m 0644 tcpdmatch.8 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
+install -m 0644 extra/try-from.8 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
+install -m 0644 extra/safe_finger.8 %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8
+
+
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}
+cp %{SOURCE1} %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/hosts.allow
+cp %{SOURCE2} %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/hosts.deny
+
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/hosts_access.5
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/hosts_options.5
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/tcpdmatch.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/try-from.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/safe_finger.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/tcpd.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/tcpdchk.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3/hosts_access.3
 
 cp debian/copyright LICENSE
 
@@ -117,32 +127,32 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/tcpd
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/tcpdchk
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/tcpdmatch
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/try-from
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/safe_finger
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libwrap.so.0
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libwrap.so.0.7.6
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man5/hosts_access.5.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man5/hosts_options.5.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/tcpdmatch.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/try-from.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/safe_finger.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/tcpd.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/tcpdchk.8.gz
-%config(noreplace) %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/hosts.allow
-%config(noreplace) %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/hosts.deny
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/tcpd
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/tcpdchk
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/tcpdmatch
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/try-from
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/safe_finger
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libwrap.so.0
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libwrap.so.0.7.6
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/hosts_access.5.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/hosts_options.5.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/tcpdmatch.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/try-from.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/safe_finger.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/tcpd.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/tcpdchk.8.gz
+%config(noreplace) %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/hosts.allow
+%config(noreplace) %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/hosts.deny
 %doc BLURB CHANGES DISCLAIMER README README.NIS
 %doc debian/README.Debian debian/changelog
 %doc LICENSE
 
 %files -n %{_pkgname}-dev
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/tcpd.h
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libwrap.a
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/libwrap.so
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man3/hosts_access.3.gz
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/tcpd.h
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libwrap.a
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/libwrap.so
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man3/hosts_access.3.gz
 %doc LICENSE
 
 %changelog
@@ -154,7 +164,7 @@
 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
 - Renamed target packages tcp-wrappers and merged tcpd package.
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 
 * Tue Jul 13 2004  Stuart Menefy <stuart.menefy@st.com>
 - Remove generation of build.log.
