--- SPECS/stm-target-nfs-utils.spec	2007-03-22 16:43:10.000000000 +0100
+++ SPECS/stm-target-nfs-utils.spec-new	2007-06-14 14:48:52.000000000 +0200
@@ -1,5 +1,5 @@
 Summary   	: NFS utlilities and supporting daemons for the kernel NFS server.
-Name	  	: %{_stm_pkg_prefix}-target-nfs-utils
+Name	  	: %{_hhl_pkg_prefix}-target-nfs-utils
 Version	  	: 1.0.6
 Release		: 6
 Source0		: ftp://prdownloads.sourceforge.net/nfs/nfs-utils-%{version}.tar.gz
@@ -9,10 +9,10 @@
 Patch3		: nfs-utils-1.0.6-prefix.patch
 License 	: GPL 
 Group	  	: System Environment/Daemons
-Buildroot 	: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix	  	: %{_stm_cross_target_dir}
-%define _defaultdocdir	%{_stm_cross_target_dir}%{_stm_target_doc_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-nfs-utils
+Buildroot 	: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix	  	: %{_hhl_cross_target_dir}
+%define _defaultdocdir	%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-nfs-utils
 
 #
 #	SRPM
@@ -34,9 +34,9 @@
 %package -n %{_pkgname}
 Summary		: NFS utlilities and supporting daemons for the kernel NFS server.
 Group		: System Environment/Daemons
-Conflicts	: %{_stm_pkg_prefix}-%{_stm_target_arch}-nfs-user-server
-PreReq		: %{_stm_pkg_prefix}-%{_stm_target_arch}-initdconfig
-PreReq		: %{_stm_pkg_prefix}-%{_stm_target_arch}-initscripts
+Conflicts	: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-nfs-user-server
+PreReq		: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-initdconfig
+PreReq		: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-initscripts
 
 %description -n %{_pkgname}
 
@@ -57,7 +57,7 @@
 %patch1 -p1 
 %patch2 -p1
 %patch3 -p1
-export PATH="%{_stm_host_bin_dir}:$PATH"
+export PATH="%{_hhl_host_bin_dir}:$PATH"
 autoconf
 
 %build
@@ -76,20 +76,42 @@
 %install
 %target_setup
 
+rm -rf %{buildroot}
+
 make \
-	SBINDIR=%{buildroot}%{_stm_cross_target_dir}%{_stm_target_sbin_dir} \
-	exec_prefix=%{buildroot}%{_stm_cross_target_dir}%{_stm_target_prefix} \
-	MANDIR=%{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir} \
-	STATEDIR=%{buildroot}%{_stm_cross_target_dir}%{_stm_target_localstate_dir}/lib/nfs/ \
+	SBINDIR=%{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir} \
+	exec_prefix=%{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_prefix} \
+	MANDIR=%{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir} \
+	STATEDIR=%{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_localstate_dir}/lib/nfs/ \
 	install
 
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d/
-
-install -m 755 debian/nfs-common.init %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d/nfs-common
-install -m 755 debian/nfs-kernel-server.init %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d/nfs-kernel-server
-install -m 644 debian/etc.exports %{buildroot}%{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/exports
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d/
 
-%target_install_fixup
+install -m 755 debian/nfs-common.init %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d/nfs-common
+install -m 755 debian/nfs-kernel-server.init %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d/nfs-kernel-server
+install -m 644 debian/etc.exports %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/exports
+
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/exports.5
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/mountd.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man7/nfsd.7
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/exportfs.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nfsd.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nhfsgraph.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nhfsnums.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nhfsrun.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nhfsstone.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/showmount.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/lockd.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nfsstat.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/rquotad.8
+gzip -9f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/statd.8
+( cd %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8 && \
+	rm rpc.nfsd.8; ln -sf nfsd.8.gz rpc.nfsd.8.gz && \
+	rm rpc.lockd.8; ln -sf lockd.8.gz rpc.lockd.8.gz && \
+	rm rpc.mountd.8; ln -sf mountd.8.gz rpc.mountd.8.gz && \
+	rm rpc.rquotad.8; ln -sf rquotad.8.gz rpc.rquotad.8.gz && \
+	rm rpc.statd.8; ln -sf statd.8.gz rpc.statd.8.gz \
+)
 
 cp debian/copyright LICENSE
 
@@ -98,53 +120,53 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d/nfs-kernel-server
-%config %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/init.d/nfs-common
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/exportfs
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/rpc.lockd
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/nfsstat
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/rpc.rquotad
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/rpc.statd
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/nhfsgraph
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/nhfsnums
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/nhfsrun
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/nhfsstone
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/rpc.mountd
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/rpc.nfsd
-%{_stm_cross_target_dir}%{_stm_target_sbin_dir}/showmount
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man5/exports.5.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/mountd.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man7/nfsd.7.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/exportfs.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/nfsd.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/nhfsgraph.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/nhfsnums.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/nhfsrun.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/nhfsstone.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/rpc.mountd.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/rpc.nfsd.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/showmount.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/lockd.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/nfsstat.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/rquotad.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/statd.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/rpc.lockd.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/rpc.rquotad.8.gz
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man8/rpc.statd.8.gz
-%dir               %{_stm_cross_target_dir}%{_stm_target_localstate_dir}/lib/nfs
-%config(noreplace) %{_stm_cross_target_dir}%{_stm_target_localstate_dir}/lib/nfs/etab
-%config(noreplace) %{_stm_cross_target_dir}%{_stm_target_localstate_dir}/lib/nfs/rmtab
-%config(noreplace) %{_stm_cross_target_dir}%{_stm_target_localstate_dir}/lib/nfs/xtab
-%config(noreplace) %{_stm_cross_target_dir}%{_stm_target_localstate_dir}/lib/nfs/state
-%config(noreplace) %{_stm_cross_target_dir}%{_stm_target_sysconf_dir}/exports
+%config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d/nfs-kernel-server
+%config %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/init.d/nfs-common
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/exportfs
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/rpc.lockd
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/nfsstat
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/rpc.rquotad
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/rpc.statd
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/nhfsgraph
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/nhfsnums
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/nhfsrun
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/nhfsstone
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/rpc.mountd
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/rpc.nfsd
+%{_hhl_cross_target_dir}%{_hhl_target_sbin_dir}/showmount
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man5/exports.5.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/mountd.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man7/nfsd.7.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/exportfs.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nfsd.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nhfsgraph.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nhfsnums.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nhfsrun.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nhfsstone.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/rpc.mountd.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/rpc.nfsd.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/showmount.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/lockd.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/nfsstat.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/rquotad.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/statd.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/rpc.lockd.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/rpc.rquotad.8.gz
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man8/rpc.statd.8.gz
+%dir               %{_hhl_cross_target_dir}%{_hhl_target_localstate_dir}/lib/nfs
+%config(noreplace) %{_hhl_cross_target_dir}%{_hhl_target_localstate_dir}/lib/nfs/etab
+%config(noreplace) %{_hhl_cross_target_dir}%{_hhl_target_localstate_dir}/lib/nfs/rmtab
+%config(noreplace) %{_hhl_cross_target_dir}%{_hhl_target_localstate_dir}/lib/nfs/xtab
+%config(noreplace) %{_hhl_cross_target_dir}%{_hhl_target_localstate_dir}/lib/nfs/state
+%config(noreplace) %{_hhl_cross_target_dir}%{_hhl_target_sysconf_dir}/exports
 %doc LICENSE
 
 %post -n %{_pkgname}
 %target_script_prefix_setup
 for service in nfs-common nfs-kernel-server ; do
-  (%{_stm_target_toolprefix}initdconfig --add $service) &> /dev/null \
+  (%{_hhl_target_toolprefix}initdconfig --add $service) &> /dev/null \
 	|| echo "Unable to enable initd service: $service"
-  (%{_stm_target_toolprefix}initdconfig $service off) &> /dev/null \
+  (%{_hhl_target_toolprefix}initdconfig $service off) &> /dev/null \
 	|| echo "Unable to switch off initd service: $service"
 done
 
@@ -167,6 +189,6 @@
 * Mon Sep 13 2004 Stuart Menefy <stuart.menefy@st.com>
 - Updated to 1.0.6 and merged kernel and common packages (no support for user).
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Sat Jan 26 2002 Stuart Menefy <stuart.menefy@st.com>
 - Copied from MontaVista version 0.3-hhl2.0.1
