--- SPECS/stm-target-coreutils.spec	2006-10-12 22:14:34.000000000 +0200
+++ SPECS/stm-target-coreutils.spec-new	2007-06-23 10:27:00.000000000 +0200
@@ -1,5 +1,5 @@
 Summary: The GNU core utilities: a set of tools commonly used in shell scripts
-Name: %{_stm_pkg_prefix}-target-coreutils
+Name: %{_hhl_pkg_prefix}-target-coreutils
 Version: 5.2.1
 Release: 9
 Source: ftp://ftp.gnu.org/pub/gnu/coreutils/coreutils-%{version}.tar.bz2
@@ -8,18 +8,18 @@
 Patch2: coreutils-5.2.1-teefix.patch
 License: GPL
 Group: System Environment/Base
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-coreutils
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-coreutils
 
-%define targetroot %{_stm_cross_target_dir}
+%define targetroot %{_hhl_cross_target_dir}
 
 %define _rbindir	%{targetroot}/bin
-%define _bindir		%{targetroot}%{_stm_target_bin_dir}
-%define _docdir		%{targetroot}%{_stm_target_doc_dir}
-%define _mandir		%{targetroot}%{_stm_target_man_dir}
-%define _datadir	%{targetroot}%{_stm_target_sharedstate_dir}
-%define _infodir	%{targetroot}%{_stm_target_info_dir}
+%define _bindir		%{targetroot}%{_hhl_target_bin_dir}
+%define _docdir		%{targetroot}%{_hhl_target_doc_dir}
+%define _mandir		%{targetroot}%{_hhl_target_man_dir}
+%define _datadir	%{targetroot}%{_hhl_target_sharedstate_dir}
+%define _infodir	%{targetroot}%{_hhl_target_info_dir}
 
 %description
 These are the GNU core utilities.  This package is the combination of
@@ -158,84 +158,84 @@
 %doc %{_mandir}/man*/*.gz
 %doc ABOUT-NLS LICENSE NEWS README THANKS TODO old/*
 %defattr (644, root, root, 755)
-%lang(af) %{targetroot}%{_stm_target_sharedstate_dir}/locale/af/LC_MESSAGES/coreutils.mo
-%lang(be) %{targetroot}%{_stm_target_sharedstate_dir}/locale/be/LC_MESSAGES/coreutils.mo
-%lang(ca) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ca/LC_MESSAGES/coreutils.mo
-%lang(da) %{targetroot}%{_stm_target_sharedstate_dir}/locale/da/LC_MESSAGES/coreutils.mo
-%lang(de) %{targetroot}%{_stm_target_sharedstate_dir}/locale/de/LC_MESSAGES/coreutils.mo
-%lang(cs) %{targetroot}%{_stm_target_sharedstate_dir}/locale/cs/LC_MESSAGES/coreutils.mo
-%lang(el) %{targetroot}%{_stm_target_sharedstate_dir}/locale/el/LC_MESSAGES/coreutils.mo
-%lang(es) %{targetroot}%{_stm_target_sharedstate_dir}/locale/es/LC_MESSAGES/coreutils.mo
-%lang(et) %{targetroot}%{_stm_target_sharedstate_dir}/locale/et/LC_MESSAGES/coreutils.mo
-%lang(fi) %{targetroot}%{_stm_target_sharedstate_dir}/locale/fi/LC_MESSAGES/coreutils.mo
-%lang(ga) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ga/LC_MESSAGES/coreutils.mo
-%lang(fr) %{targetroot}%{_stm_target_sharedstate_dir}/locale/fr/LC_MESSAGES/coreutils.mo
-%lang(gl) %{targetroot}%{_stm_target_sharedstate_dir}/locale/gl/LC_MESSAGES/coreutils.mo
-%lang(hu) %{targetroot}%{_stm_target_sharedstate_dir}/locale/hu/LC_MESSAGES/coreutils.mo
-%lang(ja) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ja/LC_MESSAGES/coreutils.mo
-%lang(it) %{targetroot}%{_stm_target_sharedstate_dir}/locale/it/LC_MESSAGES/coreutils.mo
-%lang(ko) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ko/LC_MESSAGES/coreutils.mo
-%lang(lg) %{targetroot}%{_stm_target_sharedstate_dir}/locale/lg/LC_MESSAGES/coreutils.mo
-%lang(nb) %{targetroot}%{_stm_target_sharedstate_dir}/locale/nb/LC_MESSAGES/coreutils.mo
-%lang(ms) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ms/LC_MESSAGES/coreutils.mo
-%lang(nl) %{targetroot}%{_stm_target_sharedstate_dir}/locale/nl/LC_MESSAGES/coreutils.mo
-%lang(no) %{targetroot}%{_stm_target_sharedstate_dir}/locale/no/LC_MESSAGES/coreutils.mo
-%lang(pl) %{targetroot}%{_stm_target_sharedstate_dir}/locale/pl/LC_MESSAGES/coreutils.mo
-%lang(pt) %{targetroot}%{_stm_target_sharedstate_dir}/locale/pt/LC_MESSAGES/coreutils.mo
-%lang(ru) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ru/LC_MESSAGES/coreutils.mo
-%lang(sk) %{targetroot}%{_stm_target_sharedstate_dir}/locale/sk/LC_MESSAGES/coreutils.mo
-%lang(sl) %{targetroot}%{_stm_target_sharedstate_dir}/locale/sl/LC_MESSAGES/coreutils.mo
-%lang(sv) %{targetroot}%{_stm_target_sharedstate_dir}/locale/sv/LC_MESSAGES/coreutils.mo
-%lang(tr) %{targetroot}%{_stm_target_sharedstate_dir}/locale/tr/LC_MESSAGES/coreutils.mo
-%lang(pt_BR) %{targetroot}%{_stm_target_sharedstate_dir}/locale/pt_BR/LC_MESSAGES/coreutils.mo
-%lang(zh_CN) %{targetroot}%{_stm_target_sharedstate_dir}/locale/zh_CN/LC_MESSAGES/coreutils.mo
-%lang(zh_TW) %{targetroot}%{_stm_target_sharedstate_dir}/locale/zh_TW/LC_MESSAGES/coreutils.mo
+%lang(af) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/af/LC_MESSAGES/coreutils.mo
+%lang(be) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/be/LC_MESSAGES/coreutils.mo
+%lang(ca) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ca/LC_MESSAGES/coreutils.mo
+%lang(da) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/da/LC_MESSAGES/coreutils.mo
+%lang(de) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/de/LC_MESSAGES/coreutils.mo
+%lang(cs) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/cs/LC_MESSAGES/coreutils.mo
+%lang(el) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/el/LC_MESSAGES/coreutils.mo
+%lang(es) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/es/LC_MESSAGES/coreutils.mo
+%lang(et) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/et/LC_MESSAGES/coreutils.mo
+%lang(fi) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/fi/LC_MESSAGES/coreutils.mo
+%lang(ga) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ga/LC_MESSAGES/coreutils.mo
+%lang(fr) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/fr/LC_MESSAGES/coreutils.mo
+%lang(gl) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/gl/LC_MESSAGES/coreutils.mo
+%lang(hu) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/hu/LC_MESSAGES/coreutils.mo
+%lang(ja) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ja/LC_MESSAGES/coreutils.mo
+%lang(it) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/it/LC_MESSAGES/coreutils.mo
+%lang(ko) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ko/LC_MESSAGES/coreutils.mo
+%lang(lg) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/lg/LC_MESSAGES/coreutils.mo
+%lang(nb) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/nb/LC_MESSAGES/coreutils.mo
+%lang(ms) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ms/LC_MESSAGES/coreutils.mo
+%lang(nl) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/nl/LC_MESSAGES/coreutils.mo
+%lang(no) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/no/LC_MESSAGES/coreutils.mo
+%lang(pl) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/pl/LC_MESSAGES/coreutils.mo
+%lang(pt) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/pt/LC_MESSAGES/coreutils.mo
+%lang(ru) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ru/LC_MESSAGES/coreutils.mo
+%lang(sk) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/sk/LC_MESSAGES/coreutils.mo
+%lang(sl) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/sl/LC_MESSAGES/coreutils.mo
+%lang(sv) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/sv/LC_MESSAGES/coreutils.mo
+%lang(tr) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/tr/LC_MESSAGES/coreutils.mo
+%lang(pt_BR) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/pt_BR/LC_MESSAGES/coreutils.mo
+%lang(zh_CN) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/zh_CN/LC_MESSAGES/coreutils.mo
+%lang(zh_TW) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/zh_TW/LC_MESSAGES/coreutils.mo
 # Associated links must be made language conditional as well, or we will end up with a
 # lot of links pointing to non-existant files
-%lang(af) %{targetroot}%{_stm_target_sharedstate_dir}/locale/af/LC_TIME/coreutils.mo
-%lang(be) %{targetroot}%{_stm_target_sharedstate_dir}/locale/be/LC_TIME/coreutils.mo
-%lang(ca) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ca/LC_TIME/coreutils.mo
-%lang(da) %{targetroot}%{_stm_target_sharedstate_dir}/locale/da/LC_TIME/coreutils.mo
-%lang(de) %{targetroot}%{_stm_target_sharedstate_dir}/locale/de/LC_TIME/coreutils.mo
-%lang(cs) %{targetroot}%{_stm_target_sharedstate_dir}/locale/cs/LC_TIME/coreutils.mo
-%lang(el) %{targetroot}%{_stm_target_sharedstate_dir}/locale/el/LC_TIME/coreutils.mo
-%lang(es) %{targetroot}%{_stm_target_sharedstate_dir}/locale/es/LC_TIME/coreutils.mo
-%lang(et) %{targetroot}%{_stm_target_sharedstate_dir}/locale/et/LC_TIME/coreutils.mo
-%lang(fi) %{targetroot}%{_stm_target_sharedstate_dir}/locale/fi/LC_TIME/coreutils.mo
-%lang(ga) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ga/LC_TIME/coreutils.mo
-%lang(fr) %{targetroot}%{_stm_target_sharedstate_dir}/locale/fr/LC_TIME/coreutils.mo
-%lang(gl) %{targetroot}%{_stm_target_sharedstate_dir}/locale/gl/LC_TIME/coreutils.mo
-%lang(hu) %{targetroot}%{_stm_target_sharedstate_dir}/locale/hu/LC_TIME/coreutils.mo
-%lang(ja) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ja/LC_TIME/coreutils.mo
-%lang(it) %{targetroot}%{_stm_target_sharedstate_dir}/locale/it/LC_TIME/coreutils.mo
-%lang(ko) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ko/LC_TIME/coreutils.mo
-%lang(lg) %{targetroot}%{_stm_target_sharedstate_dir}/locale/lg/LC_TIME/coreutils.mo
-%lang(nb) %{targetroot}%{_stm_target_sharedstate_dir}/locale/nb/LC_TIME/coreutils.mo
-%lang(ms) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ms/LC_TIME/coreutils.mo
-%lang(nl) %{targetroot}%{_stm_target_sharedstate_dir}/locale/nl/LC_TIME/coreutils.mo
-%lang(no) %{targetroot}%{_stm_target_sharedstate_dir}/locale/no/LC_TIME/coreutils.mo
-%lang(pl) %{targetroot}%{_stm_target_sharedstate_dir}/locale/pl/LC_TIME/coreutils.mo
-%lang(pt) %{targetroot}%{_stm_target_sharedstate_dir}/locale/pt/LC_TIME/coreutils.mo
-%lang(ru) %{targetroot}%{_stm_target_sharedstate_dir}/locale/ru/LC_TIME/coreutils.mo
-%lang(sk) %{targetroot}%{_stm_target_sharedstate_dir}/locale/sk/LC_TIME/coreutils.mo
-%lang(sl) %{targetroot}%{_stm_target_sharedstate_dir}/locale/sl/LC_TIME/coreutils.mo
-%lang(sv) %{targetroot}%{_stm_target_sharedstate_dir}/locale/sv/LC_TIME/coreutils.mo
-%lang(tr) %{targetroot}%{_stm_target_sharedstate_dir}/locale/tr/LC_TIME/coreutils.mo
-%lang(pt_BR) %{targetroot}%{_stm_target_sharedstate_dir}/locale/pt_BR/LC_TIME/coreutils.mo
-%lang(zh_CN) %{targetroot}%{_stm_target_sharedstate_dir}/locale/zh_CN/LC_TIME/coreutils.mo
-%lang(zh_TW) %{targetroot}%{_stm_target_sharedstate_dir}/locale/zh_TW/LC_TIME/coreutils.mo
+%lang(af) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/af/LC_TIME/coreutils.mo
+%lang(be) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/be/LC_TIME/coreutils.mo
+%lang(ca) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ca/LC_TIME/coreutils.mo
+%lang(da) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/da/LC_TIME/coreutils.mo
+%lang(de) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/de/LC_TIME/coreutils.mo
+%lang(cs) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/cs/LC_TIME/coreutils.mo
+%lang(el) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/el/LC_TIME/coreutils.mo
+%lang(es) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/es/LC_TIME/coreutils.mo
+%lang(et) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/et/LC_TIME/coreutils.mo
+%lang(fi) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/fi/LC_TIME/coreutils.mo
+%lang(ga) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ga/LC_TIME/coreutils.mo
+%lang(fr) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/fr/LC_TIME/coreutils.mo
+%lang(gl) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/gl/LC_TIME/coreutils.mo
+%lang(hu) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/hu/LC_TIME/coreutils.mo
+%lang(ja) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ja/LC_TIME/coreutils.mo
+%lang(it) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/it/LC_TIME/coreutils.mo
+%lang(ko) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ko/LC_TIME/coreutils.mo
+%lang(lg) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/lg/LC_TIME/coreutils.mo
+%lang(nb) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/nb/LC_TIME/coreutils.mo
+%lang(ms) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ms/LC_TIME/coreutils.mo
+%lang(nl) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/nl/LC_TIME/coreutils.mo
+%lang(no) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/no/LC_TIME/coreutils.mo
+%lang(pl) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/pl/LC_TIME/coreutils.mo
+%lang(pt) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/pt/LC_TIME/coreutils.mo
+%lang(ru) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/ru/LC_TIME/coreutils.mo
+%lang(sk) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/sk/LC_TIME/coreutils.mo
+%lang(sl) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/sl/LC_TIME/coreutils.mo
+%lang(sv) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/sv/LC_TIME/coreutils.mo
+%lang(tr) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/tr/LC_TIME/coreutils.mo
+%lang(pt_BR) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/pt_BR/LC_TIME/coreutils.mo
+%lang(zh_CN) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/zh_CN/LC_TIME/coreutils.mo
+%lang(zh_TW) %{targetroot}%{_hhl_target_sharedstate_dir}/locale/zh_TW/LC_TIME/coreutils.mo
 
 %post -n %{_pkgname}
 %target_script_prefix_setup
 
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir \
-        %{_stm_cross_target_dir}%{_stm_target_info_dir}/coreutils.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir \
+        %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/coreutils.info
 
 %preun -n %{_pkgname}
 %target_script_prefix_setup
 
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir --remove \
-        %{_stm_cross_target_dir}%{_stm_target_info_dir}/coreutils.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir --remove \
+        %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/coreutils.info
 
 
 %changelog
@@ -263,6 +263,6 @@
 * Wed Dec 29 2004 Stuart Menefy <stuart.menefy@st.com>
 - Remove info dir file.
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Fri Sep 3 2004 Stuart Menefy <stuart.menefy@st.com>
 - Created
