--- SPECS/stm-target-readline.spec	2007-03-22 16:43:10.000000000 +0100
+++ SPECS/stm-target-readline.spec-new	2007-07-09 21:02:43.000000000 +0200
@@ -1,16 +1,16 @@
 Summary: GNU readline and history libraries, run-time libraries.  
-Name: %{_stm_pkg_prefix}-target-readline 
+Name: %{_hhl_pkg_prefix}-target-readline 
 Version: 5.1
 Release: 7
 Source: readline-%{version}.tar.gz
 URL: ftp://ftp.gnu.org/gnu/readline/readline
 License: GPL 
 Group: System Environment/Libraries 
-Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix: %{_stm_cross_target_dir}
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-readline
+Buildroot: %{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix: %{_hhl_cross_target_dir}
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-readline
 
-%define _docdir	%{_stm_cross_target_dir}%{_stm_target_doc_dir}
+%define _docdir	%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
 
 %description
 The GNU readline library aids in the consistency of user interface
@@ -45,7 +45,7 @@
 %package -n %{_pkgname}-doc
 Summary: GNU readline and history libraries, run-time libraries.
 Group: Documentation/Development
-PreReq: %{_stm_pkg_prefix}-%{_stm_target_arch}-info
+PreReq: %{_hhl_pkg_prefix}-%{_hhl_target_arch}-info
 
 %description -n %{_pkgname}-doc
 The GNU readline library aids in the consistency of user interface
@@ -62,6 +62,7 @@
 
 %build
 %target_setup
+rm -rf %{buildroot}
 mkdir objdir
 cat > objdir/config.cache << EOF
 ac_cv_func_strcoll_works=yes
@@ -80,24 +81,22 @@
 
 %install
 %target_setup
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_prefix}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_exec_prefix}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_include_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_bin_dir}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_prefix}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_exec_prefix}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_include_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir}
 %target_makeinstall
 pushd shlib
 %define _objdir .
 %target_makeinstall
 %define _objdir objdir
 popd
-%target_install_fixup
-rm -f %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir*
-%target_install_fixup
+rm -f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir*
 
-pushd %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
+pushd %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
 ln -sf libhistory.so.4.3 libhistory.so.4
 ln -sf libhistory.so.4 libhistory.so
 ln -sf libreadline.so.4.3 libreadline.so.4
@@ -111,32 +110,32 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.so*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.so*
 %doc LICENSE
 
 %files -n %{_pkgname}-dev
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/*
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.a
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.a
 %doc LICENSE
 
 %files -n %{_pkgname}-doc
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man?/*
-%doc %{_stm_cross_target_dir}%{_stm_target_info_dir}/*.info*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man?/*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/*.info*
 %doc CHANGELOG CHANGES COPYING README USAGE
 %doc LICENSE
 
 %post -n %{_pkgname}-doc
 for info in history readline rluserman ; do
-  install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir \
-    %{_stm_cross_target_dir}%{_stm_target_info_dir}/$info.info
+  install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir \
+    %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/$info.info
 done
 
 %preun -n %{_pkgname}-doc
 for info in history readline rluserman ; do
-  install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir --remove \
-    %{_stm_cross_target_dir}%{_stm_target_info_dir}/$info.info
+  install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir --remove \
+    %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/$info.info
 done
 
 %changelog
@@ -156,7 +155,7 @@
 - Fixed linking bug
 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Fri Aug 20 2004 Stuart Menefy <stuart.menefy@st.com>
 - Updated to 4.3
 
