--- SPECS/stm-target-mpc.spec.orig	2010-04-28 13:53:55.000000000 +0200
+++ SPECS/stm-target-mpc.spec	2010-04-28 13:55:22.000000000 +0200
@@ -5,6 +5,7 @@
 Source		: mpc-%{version}.tar.gz
 License 	: GPL
 Group	  	: Development/Tools 
+Patch0	  	: stm-target-mpc.%{version}-%{release}.diff
 Buildroot       : %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-%{_stm_target_arch}-XXXXXX)
 BuildRequires: %{_stm_pkg_prefix}-%{_stm_target_arch}-gmp
 BuildRequires: %{_stm_pkg_prefix}-%{_stm_target_arch}-mpfr
@@ -27,6 +28,7 @@
 
 %prep
 %setup -n mpc-%{version}
+%patch0 -p1
 
 %build
 %target_setup
