diff -Nhur rpmconfig-2.2.orig/common rpmconfig-2.2/common
--- rpmconfig-2.2/common	2010-05-13 10:02:10.000000000 +0200
+++ rpmconfig-2.2.orig//common	2010-05-13 10:00:42.000000000 +0200
@@ -438,7 +438,7 @@
 %autoremake \
 install="--install" ; \
 [ -n "`autoreconf --version | grep 2.13`" ] && install="" ; \
-autoreconf --verbose --force $install
+autoreconf --verbose --force $install -I%{_stm_host_sharedstate_dir}/aclocal
 
 
 %autoremakemine \
@@ -491,7 +491,7 @@
     cn=`basename $conf`; \
     echo "cd $dn"; \
     pushd $dn ; \
-    autoreconf --force ; \
+    autoreconf --force -I%{_stm_host_sharedstate_dir}/aclocal ; \
     %replacelibtool \
     popd; \
   done ;
