--- SPECS/stm-cross-distributionutils.spec	2006-10-12 22:14:34.000000000 +0200
+++ SPECS/stm-cross-distributionutils.spec-new	2007-12-11 23:04:56.000000000 +0100
@@ -4,6 +4,7 @@
 Release: 3
 Source: cross-hardhatutils.tar.gz
 Patch: cross-hardhatutils-stm.patch
+Patch1: hardhatutils-srcdir.diff
 License: GPL
 Group: System Environment/Base
 Buildroot: %{_tmppath}/%{name}-%{_stm_target_arch}-root
@@ -32,6 +33,7 @@
 %target_check
 %setup -n hardhatutils -q
 %patch -p1
+%patch1 -p1
 
 %build
 make
