--- old/stm-target-alsa-lib.spec	2010-02-21 21:53:46.439801199 +0100
+++ new/stm-target-alsa-lib.spec	2010-02-21 21:54:48.471065841 +0100
@@ -47,13 +47,13 @@
 
 %build
 %target_setup
-%target_autoreconf
+libtoolize -f -c
+autoreconf -f -i
 ./configure \
 	--host=%{_stm_target_config} \
 	--target=%{_stm_target_config} \
 	--prefix=%{_stm_target_prefix} \
 	--exec-prefix=%{_stm_target_exec_prefix} \
-	--program-transform-name=s,x,x \
 	--with-debug=no \
 	--disable-python
 #	--with-pythonlibs="-lpthread -lm -ldl -lpython2.4" \
