--- stm-host-rpmconfig-2.2.spec	2007-10-19 17:30:25.000000000 +0200
+++ stm-host-rpmconfig-2.2.spec.new	2010-02-05 23:59:23.000000000 +0100
@@ -18,6 +18,7 @@
 Version: %{_stm_build_id}
 Release: 10 
 Source0: stm-host-rpmconfig-%{version}-%{_build_cpu}.tar.gz
+Patch1: stm-host-rpmconfig-compress_man-allways-true.patch
 License: GPL
 Group: Development/Tools
 BuildRoot: %{_tmppath}/%{name}-root
@@ -33,6 +34,7 @@
 
 %prep
 %setup -q -n rpmconfig-%{version}
+%patch1 -p1
 
 %install
 make	RPMCONFIGDIR=%{buildroot}%{_stm_config_dir}/rpm \
