--- SPECS/stm-target-grep.spec	2005-09-23 22:30:09.000000000 +0200
+++ SPECS/stm-target-grep.spec-new	2007-11-18 17:44:59.000000000 +0100
@@ -2,7 +2,7 @@
 Name: %{_hhl_pkg_prefix}-target-grep
 Version: 2.5.1
 Release: 5
-Copyright: GPL
+License: GPL
 Group: Applications/Text
 Source: ftp://ftp.gnu.org/gnu/grep/grep-%{version}.tar.bz2
 Patch0: ftp://ftp.debian.org/debian/pool/main/g/grep/grep_2.5.1.ds1-3.diff.gz
@@ -59,7 +59,7 @@
 %target_setup
 export mkdir_p="mkdir -p"
 %target_makeinstall
-rm -f %{buildroot}%{_infodir}/dir
+rm -f %{buildroot}%{_infodir}/dir* || true
 %compress_man %{buildroot}%{_infodir}
 %compress_man %{buildroot}%{_mandir}
 
