--- SPECS/stm-target-splashutils.spec	2007-12-19 08:38:21.000000000 +0100
+++ SPECS/stm-target-splashutils.spec-new	2008-07-24 16:16:33.000000000 +0200
@@ -38,8 +38,8 @@
 export LIBPNG_CONFIG=%{_stm_cross_target_dir}%{_stm_target_bin_dir}/libpng12-config
 export PNG_CFLAGS="-I%{_stm_cross_target_dir}%{_stm_target_include_dir}/libpng12 -DPNG_NO_MMX_CODE"
 export KLCC=$CC
-%replaceautofiles
-%replacelibtool
+#
+#
 %target_configure --enable-static --with-png --with-ttf --with-mng \
 								--with-ttf-kernel --without-gpm
 %make QUIET=false
