--- SPECS/stm-target-glib.spec	2006-10-12 22:14:34.000000000 +0200
+++ SPECS/stm-target-glib.spec-new	2007-07-30 14:26:21.000000000 +0200
@@ -1,20 +1,20 @@
 # $Id: stm-target-glib.spec,v 1.19 2006/10/12 19:57:31 shawc Exp $
 
-Name:		%{_stm_pkg_prefix}-target-glib
-%define _pkgname %{_stm_pkg_prefix}-%{_stm_target_arch}-glib
+Name:		%{_hhl_pkg_prefix}-target-glib
+%define _pkgname %{_hhl_pkg_prefix}-%{_hhl_target_arch}-glib
 
 Version:	1.2.10
 Release: 8
 License:	LGPL
 Source:		ftp://ftp.gimp.org/pub/gtk/v1.2/glib-%{version}.tar.gz
 URL:		http://www.gtk.org/
-Buildroot:	%{_tmppath}/%{name}-%{_stm_target_arch}-root
-Prefix:		%{_stm_cross_target_dir}
-BuildRequires:	%{_stm_pkg_prefix}-host-rpmconfig autoconf >= 2.13
+Buildroot:	%{_tmppath}/%{name}-%{_hhl_target_arch}-root
+Prefix:		%{_hhl_cross_target_dir}
+BuildRequires:	%{_hhl_pkg_prefix}-host-rpmconfig autoconf >= 2.13
 Patch0:		glib-1.2.10-configure.patch
 Patch1:		glib-1.2.10-pretty_fn.patch
 
-%define _docdir		%{_stm_cross_target_dir}%{_stm_target_doc_dir}
+%define _docdir		%{_hhl_cross_target_dir}%{_hhl_target_doc_dir}
 
 #
 #  Source Package
@@ -49,7 +49,7 @@
 
 %files -n %{_pkgname}
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.so.*
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.so.*
 %doc LICENSE
 
 #
@@ -67,18 +67,18 @@
 
 %files -n %{_pkgname}-dev
 %defattr(-,root,root)
-%{_stm_cross_target_dir}%{_stm_target_bin_dir}/glib-config
-%{_stm_cross_target_dir}%{_stm_target_include_dir}/glib-1.2
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.so
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.a
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/*.la
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/glib
-%{_stm_cross_target_dir}%{_stm_target_lib_dir}/pkgconfig/*.pc
-%{_stm_cross_target_dir}%{_stm_target_data_dir}/aclocal/*.m4
-%{_stm_host_dir}/share/aclocal/*.m4
-%{_stm_cross_bin_dir}/*
-%{_stm_cross_libtool_dir}/*
-%{_stm_cross_pkgconfig_dir}/*
+%{_hhl_cross_target_dir}%{_hhl_target_bin_dir}/glib-config
+%{_hhl_cross_target_dir}%{_hhl_target_include_dir}/glib-1.2
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.so
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.a
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/*.la
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/glib
+%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}/pkgconfig/*.pc
+%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/aclocal/*.m4
+%{_hhl_host_dir}/share/aclocal/*.m4
+%{_hhl_cross_bin_dir}/*
+%{_hhl_cross_libtool_dir}/*
+%{_hhl_cross_pkgconfig_dir}/*
 %doc LICENSE
 
 #
@@ -98,8 +98,8 @@
 
 %files -n %{_pkgname}-doc
 %defattr(-,root,root)
-%doc %{_stm_cross_target_dir}%{_stm_target_man_dir}/man?/*
-%doc %{_stm_cross_target_dir}%{_stm_target_info_dir}/glib.*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_man_dir}/man?/*
+%doc %{_hhl_cross_target_dir}%{_hhl_target_info_dir}/glib.*
 %doc README NEWS COPYING
 %doc LICENSE
 
@@ -150,12 +150,12 @@
 export CROSS_COMPILE=1
 
 # rebuild libtool for cross compilation support
-_libtool_prefix=`echo %{_stm_target_toolprefix} | sed 's/-$//'`
+_libtool_prefix=`echo %{_hhl_target_toolprefix} | sed 's/-$//'`
 ./ltconfig --cache-file="./config.cache" --with-gcc --with-gnu-ld --no-verify \
            ./ltmain.sh ${_libtool_prefix}
 
-./configure --build="%{_stm_host_config}" --host="%{_stm_target_config}" \
-            --prefix="%{_stm_target_prefix}" --disable-threads --disable-nls
+./configure --build="%{_hhl_host_config}" --host="%{_hhl_target_config}" \
+            --prefix="%{_hhl_target_prefix}" --disable-threads --disable-nls
 
 cat > glibconfig-sysdefs.h << EOF
 #define GLIB_SYSDEF_POLLIN =1
@@ -174,29 +174,29 @@
 %host_setup
 
 rm -rf %{buildroot}
-mkdir -p %{buildroot}%{_stm_cross_target_dir}%{_stm_target_prefix}
-mkdir -p %{buildroot}%{_stm_cross_bin_dir}
-mkdir -p %{buildroot}%{_stm_host_dir}/share/aclocal
+mkdir -p %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_prefix}
+mkdir -p %{buildroot}%{_hhl_cross_bin_dir}
+mkdir -p %{buildroot}%{_hhl_host_dir}/share/aclocal
 
 %define _objdir .
 %target_makeinstall
 
 # Copy aclocal m4 script for use by host tools
-cp %{buildroot}%{_stm_cross_target_dir}%{_stm_target_data_dir}/aclocal/*.m4  %{buildroot}%{_stm_host_dir}/share/aclocal
+cp %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/aclocal/*.m4  %{buildroot}%{_hhl_host_dir}/share/aclocal
 
 # Create host glib-config
-sed -e "s#/usr#%{_stm_cross_target_dir}%{_stm_target_prefix}#g" < glib-config > %{_stm_target_toolprefix}glib-config
-%{__install} -m 755 %{_stm_target_toolprefix}glib-config %{buildroot}%{_stm_cross_bin_dir}
+sed -e "s#/usr#%{_hhl_cross_target_dir}%{_hhl_target_prefix}#g" < glib-config > %{_hhl_target_toolprefix}glib-config
+%{__install} -m 755 %{_hhl_target_toolprefix}glib-config %{buildroot}%{_hhl_cross_bin_dir}
 
 # Remove info dir file
-rm -f %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir
+rm -f %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir
 
 # compress documentation
-(gzip -9r %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir} || true)
-gzip -9r %{buildroot}%{_stm_cross_target_dir}%{_stm_target_info_dir}
+(gzip -9r %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_man_dir} || true)
+gzip -9r %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_info_dir}
 
 # fix some flags
-pushd %{buildroot}%{_stm_cross_target_dir}%{_stm_target_lib_dir}
+pushd %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_lib_dir}
 chmod -x *.la
 chmod +x *.so.*
 popd
@@ -210,12 +210,12 @@
 rm -rf %{buildroot}
 
 %post -n %{_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir \
-	%{_stm_cross_target_dir}%{_stm_target_info_dir}/glib.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir \
+	%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/glib.info
 
 %preun -n %{_pkgname}-doc
-install-info --quiet --dir-file=%{_stm_cross_target_dir}%{_stm_target_info_dir}/dir --remove \
-	%{_stm_cross_target_dir}%{_stm_target_info_dir}/glib.info
+install-info --quiet --dir-file=%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/dir --remove \
+	%{_hhl_cross_target_dir}%{_hhl_target_info_dir}/glib.info
 
 %changelog
 * Mon Jun  6 2005 Carl Shaw <carl.shaw@st.com>
@@ -226,7 +226,7 @@
 - manpages are not generated (tools missing on host)
 
 * Wed Sep  8 2004 Stuart Menefy <stuart.menefy@st.com>
-- Automated introduction of %{_stm_pkg_prefix} and %{_pkgname}
+- Automated introduction of %{_hhl_pkg_prefix} and %{_pkgname}
 * Mon Aug 16 2004 Stuart Menefy <stuart.menefy@st.com>
 - Added  glib-1.2.10-pretty_fn.patch.
 - Use %autoremake and removed explicit calls to autoconf.
