--- SPECS/stm-target-ncurses.spec	2005-09-23 22:30:09.000000000 +0200
+++ SPECS/stm-target-ncurses.spec-new	2007-06-25 20:42:16.000000000 +0200
@@ -154,7 +154,7 @@
 echo "%dir %{_hhl_cross_target_dir}%{_hhl_target_data_dir}/terminfo" >> ncurses-base.filelist
 echo "%dir %{_hhl_cross_target_dir}%{_hhl_target_data_dir}/terminfo/*" >> ncurses-base.filelist
 
-for i in `find %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/terminfo -type f | sed -e s,%{buildroot},,`; do
+for i in `find -L %{buildroot}%{_hhl_cross_target_dir}%{_hhl_target_data_dir}/terminfo -type f | sed -e s,%{buildroot},,`; do
  case "$i" in \
  */ansi | \
  */dumb | \
