--- rpmconfig-2.4/Makefile.orig	2009-10-21 05:05:07.000000000 +0200
+++ rpmconfig-2.4/Makefile	2010-04-04 20:21:38.626803626 +0200
@@ -10,15 +10,21 @@
 install::
 	install -d -m 755 $(HOSTDIR)
 	for host in hosts/*; do				\
-	    install -m 644 $${host} $(HOSTDIR);		\
-	done
+	    [ $$host != "hosts/CVS" ]                   \
+	    && install -m 644 $${host} $(HOSTDIR)       \
+	    || true;                                    \
+	done
 	install -d -m 755 $(TARGETDIR)
-	for target in targets/*; do			\
-	    install -m 644 $${target} $(TARGETDIR);	\
+	for target in targets/*; do                     \
+	    [ $$target != "targets/CVS" ]               \
+	    && install -m 644 $${target} $(TARGETDIR)   \
+	    || true;                                    \
 	done
 	install -d -m 755 $(CONFIGDIR)
-	for config in site_configs/*; do		\
-	    install -m 644 $${config} $(CONFIGDIR);	\
+	for config in site_configs/*; do                \
+	    [ $$config != "site_configs/CVS" ]          \
+	    && install -m 644 $${config} $(CONFIGDIR)   \
+	    || true;                                    \
 	done
 	install -m 644 common $(RPMCONFIGDIR)
 	install -m 644 config.site.template $(RPMCONFIGDIR)
