--- stm-cross-binutils_2.22.64_org.spec	2013-04-05 22:55:27.031824892 +0200
+++ stm-cross-binutils_2.22.64.spec	2013-04-05 23:12:16.011866722 +0200
@@ -140,9 +140,7 @@
        --enable-packer="no" \
        --disable-secinfo \
 %endif
-%if_target_cpu arm
        --disable-werror \
-%endif
        --with-pkgversion="GNU Binutils - STMicroelectronics/Linux Base" \
        --with-bugurl="https://bugzilla.stlinux.com" \
        --enable-install-libbfd \
