--- a/tuxtxt.c.org	2012-08-31 19:48:39.687357537 +0200
+++ b/tuxtxt.c	2012-08-31 19:48:39.107357561 +0200
@@ -204,11 +204,19 @@
 	tuxtxt_cache.vtxtpid = pid;
 
 	/* open Framebuffer */
+#ifdef __sh__
+	if ((renderinfo.fb=open("/dev/fb0", O_RDWR)) == -1)
+	{
+		perror("TuxTxt <open /dev/fb0 || /dev/fb0>");
+		return 0;
+	}
+#else
 	if ((renderinfo.fb=open("/dev/fb/0", O_RDWR)) == -1)
 	{
 		perror("TuxTxt <open /dev/fb/0>");
 		return 0;
 	}
+#endif
 	rc[0] = rc[1] =-1;
 	while(rc_num < 2)
 	{
@@ -225,13 +233,21 @@
 		}
 		if (ioctl(rc[rc_num], EVIOCGNAME(128), tmp) < 0)
 			perror("EVIOCGNAME failed");
+#ifdef __sh__
+		if (!strstr(tmp, "TDT RC event driver"))
+#else
 		if (!strstr(tmp, "remote control"))
+#endif
 		{
 			close(rc[rc_num]);
 			rc[rc_num] = -1;
 		}
 		else
+#ifdef __sh__
+			break;
+#else
 			++rc_num;
+#endif
 		++cnt;
 	}
 
@@ -2926,6 +2942,9 @@
 				case KEY_HELP:		RCCode = RC_HELP;	break;
 				case KEY_MENU:		RCCode = RC_DBOX;	break;
 				case KEY_EXIT:		RCCode = RC_HOME;	break;
+#ifdef __sh__
+				case KEY_HOME:		RCCode = RC_HOME;	break;
+#endif
 				case KEY_POWER:		RCCode = RC_STANDBY;	break;
 				default:			RCCode = -1;		break;
 				}
