--- SPECS/stm-host-u-boot.spec	2010-04-16 11:55:36.000000000 +0200
+++ SPECS/stm-host-u-boot-patched.spec	2011-07-02 21:49:10.391078427 +0200
@@ -20,6 +20,7 @@
 Patch8: u-boot-1.3.1_stm23_0045.patch
 Patch9: u-boot-1.3.1_stm23_0046.patch
 Patch10: u-boot-1.3.1_stm24_0047.patch
+Patch11: uboot-1.3.1_lzma_stm24.patch
 %else
 Patch300: u-boot-1.3.1_nmdk.patch
 Patch301: u-boot-1.3.1_nmdk_001-ethernet.patch
@@ -67,6 +69,8 @@
 %patch7 -p1
 %patch8 -p1
 %patch9 -p1
+%patch10 -p1
+%patch11 -p1
 %else
 %patch300 -p1
 %patch301 -p1

