*** orig-linux-2.6.17.14_stm22_0040/drivers/serial/stasc.c	2009-01-31 16:49:08.000000000 +0100
--- linux-2.6.17.14_stm22_0040/drivers/serial/stasc.c	2009-01-31 17:09:02.000000000 +0100
***************
*** 379,384 ****
--- 379,385 ----
  		.pio_port	= 4,
  		.pio_pin	= {3, 2, 4, 5},
  	},
+ #ifdef OHNE_MICOM 
  	/* UART3 */
  	{
  		.port	= {
***************
*** 394,399 ****
--- 395,401 ----
  		.pio_port	= 5,
  		.pio_pin	= {0, 1, 2, 3},
  	},
+ #endif
  #elif defined(CONFIG_CPU_SUBTYPE_STX7200)
  	/* UART2 */
  	{
