*** linux/drivers/mtd/maps/stboards.c	2009-04-21 21:50:49.000000000 +0200
--- UFS910/stboards.c	2009-04-23 01:00:05.000000000 +0200
***************
*** 54,60 ****
  #define ONBOARD_SIZE (8*1024*1024)
  #define ONBOARD_BANKWIDTH 2
  #elif defined(CONFIG_SH_STB7100_REF)
! #define ONBOARD_SIZE (8*1024*1024)
  #define ONBOARD_BANKWIDTH 2
  #elif defined(CONFIG_SH_STB5202_REF)
  #define ONBOARD_SIZE (32*1024*1024)
--- 54,60 ----
  #define ONBOARD_SIZE (8*1024*1024)
  #define ONBOARD_BANKWIDTH 2
  #elif defined(CONFIG_SH_STB7100_REF)
! #define ONBOARD_SIZE (16*1024*1024)
  #define ONBOARD_BANKWIDTH 2
  #elif defined(CONFIG_SH_STB5202_REF)
  #define ONBOARD_SIZE (32*1024*1024)
***************
*** 127,150 ****
  };
  #endif
  
! static struct mtd_partition onboard_partitions[3] = {
! 	{
! 	 .name = "Boot firmware",
! 	 .size = 0x00040000,
! 	 .offset = 0x00000000,
! 	 /* mask_flags: MTD_WRITEABLE *//* force read-only */
! 	 },
! 	{
! 	 .name = "Kernel",
! 	 .size = 0x00100000,
! 	 .offset = 0x00040000,
! 
! 	 },
! 	{
! 	 .name = "Root FS",
! 	 .size = MTDPART_SIZ_FULL,	/* will expand to the end of the flash */
! 	 .offset = 0x00140000,
! 	 }
  };
  static struct mtd_partition *parsed_parts;
  static const char *probes[] = { "cmdlinepart", NULL };
--- 127,173 ----
  };
  #endif
  
! static struct mtd_partition onboard_partitions[8] = {
! {
!  .name = "Boot firmware :        0xA000.0000-0xA001.FFFF", 
!  .size = 0x00020000, 
!  .offset = 0x00000000,
!  },
! {
!  .name = "Kernel -               0xA004.0000-0xA01F.FFFF", 
!  .size = 0x001C0000, 
!  .offset = 0x00040000,
!  },
! {
!  .name = "Config FS -            0xA020.0000-0xA029.FFFF",  
!  .size = 0x000A0000,  
!  .offset = 0x00200000,  
!  },
! {
!  .name = "Root FS-               0xA02A.0000-0xA04D.FFFF",  
!  .size = 0x00240000,  
!  .offset = 0x002A0000,  
!  },
! {
!  .name = "APP_Modules            0xA04E.0000-0xA0AD.FFFF",  
!  .size = 0x00600000,  
!  .offset = 0x004E0000,  
!  },
! {
!  .name = "EmergencyRoot          0xA0AE.0000-0xA0BF.FFFF",  
!  .size = 0x00120000,  
!  .offset = 0x00AE0000,  
!  },
! {
!  .name = "OtherData              0xA0C0.0000-0xA0FF.FFFF",  
!  .size = 0x00400000,  
!  .offset = 0x00C00000,  
!  },
! {
!  .name = "BootConfiguration      0xA002.0000-0xA003.FFFF",  
!  .size = 0x00020000,  
!  .offset = 0x00020000,  
!  }
  };
  static struct mtd_partition *parsed_parts;
  static const char *probes[] = { "cmdlinepart", NULL };
