*** linux/drivers/serial/stasc.c	2009-02-11 08:53:25.000000000 +0100
--- stasc.c.new	2009-02-11 08:53:36.000000000 +0100
***************
*** 379,384 ****
--- 379,385 ----
  		.pio_port	= 4,
  		.pio_pin	= {3, 2, 4, 5},
  	},
+ #ifdef OHNE_MICOM
  	/* UART3 */
  	{
  		.port	= {
***************
*** 394,399 ****
--- 395,401 ----
  		.pio_port	= 5,
  		.pio_pin	= {0, 1, 2, 3},
  	},
+ #endif
  #else
  #error "ASC error: CPU subtype not defined"
  #endif
