diff -Nur webkitdfb_2010-11-18.orig/configure.ac webkitdfb_2010-11-18/configure.ac
--- webkitdfb_2010-11-18.orig/configure.ac	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/configure.ac	2011-04-10 05:06:07.513283601 -0700
@@ -930,8 +930,6 @@
     html_flags=yes
 fi
 
-GTK_DOC_CHECK([1.10])
-
 # OS conditionals
 AM_CONDITIONAL([OS_WIN32],[test "$os_win32" = "yes"])
 AM_CONDITIONAL([OS_UNIX],[test "$os_win32" = "no"])
diff -Nur webkitdfb_2010-11-18.orig/Makefile webkitdfb_2010-11-18/Makefile
--- webkitdfb_2010-11-18.orig/Makefile	2011-01-06 15:08:19.000000000 -0800
+++ webkitdfb_2010-11-18/Makefile	2011-04-09 08:39:49.000000000 -0700
@@ -1,4 +1,4 @@
-MODULES = JavaScriptCore JavaScriptGlue ANGLE WebCore WebKit WebKit2 WebKitTools 
+MODULES = JavaScriptCore JavaScriptGlue WebCore WebKit WebKitTools
 
 all:
 	@for dir in $(MODULES); do ${MAKE} $@ -C $$dir; exit_status=$$?; \
diff -Nur webkitdfb_2010-11-18.orig/WebKit/dfb/docs/GNUmakefile.am webkitdfb_2010-11-18/WebKit/dfb/docs/GNUmakefile.am
--- webkitdfb_2010-11-18.orig/WebKit/dfb/docs/GNUmakefile.am	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/WebKit/dfb/docs/GNUmakefile.am	2011-04-09 08:41:57.000000000 -0700
@@ -88,11 +88,11 @@
 	$(LIBSOUP_LIBS)
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/gtk-doc.make
+#include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST +=
+EXTRA_DIST =
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
diff -Nur webkitdfb_2010-11-18.orig/WebKit/gtk/docs/GNUmakefile.am webkitdfb_2010-11-18/WebKit/gtk/docs/GNUmakefile.am
--- webkitdfb_2010-11-18.orig/WebKit/gtk/docs/GNUmakefile.am	2011-01-06 15:08:38.000000000 -0800
+++ webkitdfb_2010-11-18/WebKit/gtk/docs/GNUmakefile.am	2011-04-09 08:42:09.000000000 -0700
@@ -90,11 +90,11 @@
 	$(LIBSOUP_LIBS)
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/gtk-doc.make
+#include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST +=
+EXTRA_DIST =
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
