diff -Nur webkitdfb_2010-11-18.orig/autotools/webkit.m4 webkitdfb_2010-11-18/autotools/webkit.m4
--- webkitdfb_2010-11-18.orig/autotools/webkit.m4	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/autotools/webkit.m4	2011-04-15 13:24:27.125967525 -0700
@@ -156,17 +156,8 @@
 		UNICODE_LIBS="-licuin -licuuc"
                 ;;
             *)
-		AC_PATH_PROG(icu_config, icu-config, no)
-		if test "$icu_config" = "no"; then
-			AC_MSG_ERROR([Cannot find icu-config. The ICU library is needed.])
-		fi
-
-		# We don't use --cflags as this gives us a lot of things that we don't
-		# necessarily want, like debugging and optimization flags
-		# See man (1) icu-config for more info.
-		UNICODE_CFLAGS=`$icu_config --cppflags`
-		UNICODE_LIBS=`$icu_config --ldflags-libsonly`
-                ;;
+		UNICODE_CFLAGS=""
+		UNICODE_LIBS="-licui18n -licuuc -licudata -lpthread -ldl -lm"
         esac
 fi
 
diff -Nur webkitdfb_2010-11-18.orig/configure.ac webkitdfb_2010-11-18/configure.ac
--- webkitdfb_2010-11-18.orig/configure.ac	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/configure.ac	2011-04-15 12:13:14.126031518 -0700
@@ -17,9 +17,20 @@
 AC_CONFIG_AUX_DIR([autotools])
 AC_SUBST(ACLOCAL_AMFLAGS, "-I autotools")
 
+dnl Save user specified CFLAGS/CXXFLAGS. 
+dnl We want to separate it from configure generated defaults
+USER_CFLAGS="$CFLAGS"
+USER_CXXFLAGS="$CXXFLAGS"
+AC_SUBST([USER_CFLAGS])
+AC_SUBST([USER_CXXFLAGS])
+
 AC_CONFIG_HEADERS([autotoolsconfig.h])
 AC_CANONICAL_HOST
 
+dnl Restore user specified flags. Discard configure generated defaults.
+CFLAGS="$USER_CFLAGS"
+CXXFLAGS="$USER_CXXFLAGS"
+
 WEBKIT_MAJOR_VERSION=webkit_major_version
 WEBKIT_MINOR_VERSION=webkit_minor_version
 WEBKIT_MICRO_VERSION=webkit_micro_version
@@ -38,10 +49,10 @@
 LIBWEBKITGTK_VERSION=3:1:3
 AC_SUBST([LIBWEBKITGTK_VERSION])
 
-AM_INIT_AUTOMAKE([foreign subdir-objects tar-ustar])
+AM_INIT_AUTOMAKE([foreign subdir-objects])
 
 # Use AM_SILENT_RULES if present
-m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
+#m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
 # host checking - inspired by the GTK+ configure.in
 # TODO: move these to webkit.m4?
@@ -82,6 +93,7 @@
 AC_DISABLE_STATIC
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
+AM_PROG_CC_C_O
 # use dolt to speedup the build
 DOLT
 
@@ -167,6 +179,35 @@
 fi
 AC_SUBST([PNG_LIBS])
 
+# determine the Unicode backend
+AC_MSG_CHECKING([the Unicode backend to use])
+AC_ARG_WITH(unicode_backend,
+            AC_HELP_STRING([--with-unicode-backend=@<:@icu@:>@],
+                           [Select Unicode backend [default=icu]]),
+            [],[unicode_backend="icu"])
+
+case "$unicode_backend" in
+     icu) ;;
+     *) AC_MSG_ERROR([Invalid Unicode backend: must be icu.]) ;;
+esac
+
+AC_MSG_RESULT([$unicode_backend])
+
+if test "$unicode_backend" = "icu"; then
+       if test "$os_darwin" = "yes"; then
+               UNICODE_CFLAGS="-I\$(srcdir)/JavaScriptCore/icu -I\$(srcdir)/WebCore/icu"
+               UNICODE_LIBS="-licucore"
+       elif test "$os_win32" = "yes"; then
+               UNICODE_CFLAGS=""
+               UNICODE_LIBS="-licuin -licuuc"
+       else
+               UNICODE_CFLAGS=""
+               UNICODE_LIBS="-licui18n -licuuc -licudata -lpthread -ldl -lm"
+       fi
+fi
+
+AC_SUBST([UNICODE_CFLAGS])
+AC_SUBST([UNICODE_LIBS])
 
 if test "$os_win32" = "yes"; then
   WINMM_LIBS=-lwinmm
@@ -930,8 +971,6 @@
     html_flags=yes
 fi
 
-GTK_DOC_CHECK([1.10])
-
 # OS conditionals
 AM_CONDITIONAL([OS_WIN32],[test "$os_win32" = "yes"])
 AM_CONDITIONAL([OS_UNIX],[test "$os_win32" = "no"])
diff -Nur webkitdfb_2010-11-18.orig/JavaScriptCore/GNUmakefile.am webkitdfb_2010-11-18/JavaScriptCore/GNUmakefile.am
--- webkitdfb_2010-11-18.orig/JavaScriptCore/GNUmakefile.am	2011-01-06 15:12:12.000000000 -0800
+++ webkitdfb_2010-11-18/JavaScriptCore/GNUmakefile.am	2011-04-15 13:02:38.430954321 -0700
@@ -600,12 +600,14 @@
 JavaScriptCore/pcre/chartables.c: $(srcdir)/JavaScriptCore/pcre/dftables
 	$(AM_V_GEN)$(PERL) $^ $@
 
-bin_PROGRAMS += \
-	Programs/jsc-@WEBKITGTK_API_MAJOR_VERSION@$(EXEEXT)
-
-noinst_PROGRAMS += \
-	Programs/jsc \
-	Programs/minidom
+bin_PROGRAMS =
+#bin_PROGRAMS += \
+#	Programs/jsc-@WEBKITGTK_API_MAJOR_VERSION@$(EXEEXT)
+
+noinst_PROGRAMS =
+#noinst_PROGRAMS += \
+#	Programs/jsc \
+#	Programs/minidom
 
 # minidom
 Programs_minidom_SOURCES = \
@@ -631,6 +633,7 @@
 
 Programs_minidom_LDADD = \
 	 libJavaScriptCore.la \
+	 $(UNICODE_LIBS) \
 	 $(WINMM_LIBS) \
 	 -lm \
 	 -lstdc++
@@ -661,8 +664,12 @@
 
 Programs_jsc_LDADD = \
 	libJavaScriptCore.la \
+	$(UNICODE_LIBS) \
 	$(WINMM_LIBS)
 
+Programs_jsc_LDFLAGS = \
+	$(UNICODE_LIBS)
+
 EXTRA_DIST += \
 	JavaScriptCore/AUTHORS \
 	JavaScriptCore/ChangeLog \
diff -Nur webkitdfb_2010-11-18.orig/Makefile webkitdfb_2010-11-18/Makefile
--- webkitdfb_2010-11-18.orig/Makefile	2011-01-06 15:08:19.000000000 -0800
+++ webkitdfb_2010-11-18/Makefile	2011-04-13 10:46:21.000000000 -0700
@@ -1,4 +1,4 @@
-MODULES = JavaScriptCore JavaScriptGlue ANGLE WebCore WebKit WebKit2 WebKitTools 
+MODULES = JavaScriptCore JavaScriptGlue WebCore WebKit WebKitTools
 
 all:
 	@for dir in $(MODULES); do ${MAKE} $@ -C $$dir; exit_status=$$?; \
diff -Nur webkitdfb_2010-11-18.orig/WebCore/bridge/npapi.h webkitdfb_2010-11-18/WebCore/bridge/npapi.h
--- webkitdfb_2010-11-18.orig/WebCore/bridge/npapi.h	2011-01-06 15:08:26.000000000 -0800
+++ webkitdfb_2010-11-18/WebCore/bridge/npapi.h	2011-04-13 10:46:21.000000000 -0700
@@ -89,9 +89,11 @@
 #endif
 #endif
 
-#if defined(XP_UNIX)
-#include <X11/Xlib.h>
-#include <X11/Xutil.h>
+#ifdef XP_UNIX
+#if defined(MOZ_X11)
+     #include <X11/Xlib.h>
+     #include <X11/Xutil.h>
+#endif
 #include <stdio.h>
 #endif
 
@@ -266,10 +268,12 @@
 typedef struct
 {
   int32_t      type;
+#if defined(MOZ_X11)
   Display*     display;
   Visual*      visual;
   Colormap     colormap;
   unsigned int depth;
+#endif
 } NPSetWindowCallbackStruct;
 
 typedef struct
@@ -533,8 +537,12 @@
   uint32_t lParam;
 } NPEvent;
 #elif defined(XP_UNIX)
+#if defined(MOZ_X11)
 typedef XEvent NPEvent;
 #else
+typedef GdkEvent NPEvent;
+#endif
+#else
 typedef void*  NPEvent;
 #endif
 
@@ -546,7 +554,7 @@
 typedef CGPathRef NPCGRegion;
 #elif defined(XP_WIN)
 typedef HRGN NPRegion;
-#elif defined(XP_UNIX)
+#elif defined(XP_UNIX)&& defined(MOZ_X11)
 typedef Region NPRegion;
 #elif defined(XP_SYMBIAN)
 typedef QRegion* NPRegion;
diff -Nur webkitdfb_2010-11-18.orig/WebCore/bridge/npruntime_internal.h webkitdfb_2010-11-18/WebCore/bridge/npruntime_internal.h
--- webkitdfb_2010-11-18.orig/WebCore/bridge/npruntime_internal.h	2011-01-06 15:08:26.000000000 -0800
+++ webkitdfb_2010-11-18/WebCore/bridge/npruntime_internal.h	2011-04-13 10:46:21.000000000 -0700
@@ -30,7 +30,9 @@
 #include "npruntime.h"
 
 #ifdef XP_UNIX
+#if PLATFORM(X11)
     #include <X11/Xresource.h>
+#endif
 
     #undef None
     #undef Above
diff -Nur webkitdfb_2010-11-18.orig/WebCore/css/StyleSheet.cpp webkitdfb_2010-11-18/WebCore/css/StyleSheet.cpp
--- webkitdfb_2010-11-18.orig/WebCore/css/StyleSheet.cpp	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/WebCore/css/StyleSheet.cpp	2011-04-16 12:52:01.321164400 -0700
@@ -25,6 +25,8 @@
 #include "Node.h"
 #include "SVGNames.h"
 
+#include "Document.h"
+
 namespace WebCore {
 
 #if !ASSERT_DISABLED
diff -Nur webkitdfb_2010-11-18.orig/WebCore/platform/network/curl/ResourceHandleManager.cpp webkitdfb_2010-11-18/WebCore/platform/network/curl/ResourceHandleManager.cpp
--- webkitdfb_2010-11-18.orig/WebCore/platform/network/curl/ResourceHandleManager.cpp	2011-01-06 15:08:36.000000000 -0800
+++ webkitdfb_2010-11-18/WebCore/platform/network/curl/ResourceHandleManager.cpp	2011-04-13 10:46:21.000000000 -0700
@@ -338,9 +338,9 @@
     // Retry 'select' if it was interrupted by a process signal.
     int rc = 0;
     do {
-        FD_ZERO(&fdread);
-        FD_ZERO(&fdwrite);
-        FD_ZERO(&fdexcep);
+        //FD_ZERO(&fdread);
+        //FD_ZERO(&fdwrite);
+        //FD_ZERO(&fdexcep);
         curl_multi_fdset(m_curlMultiHandle, &fdread, &fdwrite, &fdexcep, &maxfd);
         // When the 3 file descriptors are empty, winsock will return -1
         // and bail out, stopping the file download. So make sure we
diff -Nur webkitdfb_2010-11-18.orig/WebCore/plugins/PluginView.cpp webkitdfb_2010-11-18/WebCore/plugins/PluginView.cpp
--- webkitdfb_2010-11-18.orig/WebCore/plugins/PluginView.cpp	2011-01-06 15:08:36.000000000 -0800
+++ webkitdfb_2010-11-18/WebCore/plugins/PluginView.cpp	2011-04-13 10:46:21.000000000 -0700
@@ -870,11 +870,13 @@
 #endif
 #if defined(XP_UNIX) && ENABLE(NETSCAPE_PLUGIN_API)
     , m_hasPendingGeometryChange(true)
+#if PLATFORM(X11)
     , m_drawable(0)
     , m_visual(0)
     , m_colormap(0)
     , m_pluginDisplay(0)
 #endif
+#endif
 #if PLATFORM(QT) && defined(MOZ_PLATFORM_MAEMO) && (MOZ_PLATFORM_MAEMO >= 5)
     , m_renderToImage(false)
 #endif
diff -Nur webkitdfb_2010-11-18.orig/WebCore/plugins/PluginView.h webkitdfb_2010-11-18/WebCore/plugins/PluginView.h
--- webkitdfb_2010-11-18.orig/WebCore/plugins/PluginView.h	2011-01-06 15:08:36.000000000 -0800
+++ webkitdfb_2010-11-18/WebCore/plugins/PluginView.h	2011-04-13 10:46:21.000000000 -0700
@@ -406,12 +406,13 @@
 
 #if defined(XP_UNIX) && ENABLE(NETSCAPE_PLUGIN_API)
         bool m_hasPendingGeometryChange;
+#if PLATFORM(X11)
         Pixmap m_drawable;
         Visual* m_visual;
         Colormap m_colormap;
         Display* m_pluginDisplay;
 
-        void initXEvent(XEvent* event);
+#endif
 #endif
 
 #if PLATFORM(QT) 
diff -Nur webkitdfb_2010-11-18.orig/WebKit/dfb/docs/GNUmakefile.am webkitdfb_2010-11-18/WebKit/dfb/docs/GNUmakefile.am
--- webkitdfb_2010-11-18.orig/WebKit/dfb/docs/GNUmakefile.am	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/WebKit/dfb/docs/GNUmakefile.am	2011-04-13 10:46:21.000000000 -0700
@@ -88,11 +88,11 @@
 	$(LIBSOUP_LIBS)
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/gtk-doc.make
+#include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST +=
+EXTRA_DIST =
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
diff -Nur webkitdfb_2010-11-18.orig/WebKit/dfb/tests/testdfb.cpp webkitdfb_2010-11-18/WebKit/dfb/tests/testdfb.cpp
--- webkitdfb_2010-11-18.orig/WebKit/dfb/tests/testdfb.cpp	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/WebKit/dfb/tests/testdfb.cpp	2011-04-14 11:05:35.887087311 -0700
@@ -48,7 +48,7 @@
 #include <glib-object.h>
 
 #include <webkitdfb.h>
-#include <webkit/webkitrender.h>
+//#include <webkit/webkitrender.h>
 #include <GraphicsContext.h>
 
 #if PLATFORM(CAIRO)
diff -Nur webkitdfb_2010-11-18.orig/WebKit/dfb/webview.cpp webkitdfb_2010-11-18/WebKit/dfb/webview.cpp
--- webkitdfb_2010-11-18.orig/WebKit/dfb/webview.cpp	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/WebKit/dfb/webview.cpp	2011-04-16 07:22:22.468683200 -0700
@@ -70,8 +70,15 @@
 #include <Document.h>
 #include <FocusController.h>
 
+#include <JavaScriptCore/APICast.h>
+#include <JavaScriptCore/JavaScript.h>
+
+
+
 #include "webview.h"
 
+#include <WebCore/bindings/js/JSDOMBinding.h>
+
 D_DEBUG_DOMAIN(LiteWebViewDomain, "LiTE/WebView", "LiteWebView");
 
 LiteWebViewTheme *liteDefaultWebViewTheme = NULL;
@@ -322,6 +329,39 @@
 
      return DFB_OK;
 }
+
+DFBResult
+lite_webview_handleKeyboardEvent( LiteWebView *webview, DFBWindowEvent *evt )
+{
+     LITE_NULL_PARAMETER_CHECK(webview);
+     
+     //WebCore::PlatformKeyboardEvent keyEvent(&event);
+     WebCore::PlatformKeyboardEvent event( (DFBEvent*) evt );
+     webview->page->mainFrame()->eventHandler()->keyEvent( event );
+     
+
+     return DFB_OK;
+}
+
+JSGlobalContextRef
+lite_webview_get_global_context( LiteWebView *webview )
+{
+     return toGlobalRef(webview->page->mainFrame()->script()->globalObject(WebCore::mainThreadNormalWorld())->globalExec());
+}
+
+DFBResult lite_webview_execute_script( LiteWebView *webview, char* script )
+{
+     webview->page->mainFrame()->script()->executeScript(script, true);
+
+     return DFB_OK;
+}
+
+DFBResult lite_webview_set_transparent(LiteWebView* webview, gboolean flag)
+{
+     webview->page->mainFrame()->view()->setTransparent(flag);
+
+     return DFB_OK;
+}
 
 DFBResult 
 lite_on_webview_doc_loaded(LiteWebView           *webview,
diff -Nur webkitdfb_2010-11-18.orig/WebKit/dfb/webview.h webkitdfb_2010-11-18/WebKit/dfb/webview.h
--- webkitdfb_2010-11-18.orig/WebKit/dfb/webview.h	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/WebKit/dfb/webview.h	2011-04-16 07:20:36.646659600 -0700
@@ -38,6 +38,8 @@
 {
 #endif
 
+#include <JavaScriptCore/JSBase.h>
+
 #include <lite/box.h>
 #include <lite/theme.h>
 
@@ -105,6 +107,11 @@
 DFBResult lite_webview_scroll_down( LiteWebView *webview );
 DFBResult lite_webview_scroll_up  ( LiteWebView *webview );
 
+DFBResult lite_webview_handleKeyboardEvent( LiteWebView *webview, DFBWindowEvent *evt );
+JSGlobalContextRef lite_webview_get_global_context( LiteWebView *webview );
+DFBResult lite_webview_execute_script( LiteWebView *webview, char* script );
+DFBResult lite_webview_set_transparent(LiteWebView* webview, gboolean flag);
+
 /* @brief Install the WebView navigation callback
  * This function will install a WebView navigation callback. This
  * callback is triggered every time the WebView navigates to a page.
diff -Nur webkitdfb_2010-11-18.orig/WebKit/gtk/docs/GNUmakefile.am webkitdfb_2010-11-18/WebKit/gtk/docs/GNUmakefile.am
--- webkitdfb_2010-11-18.orig/WebKit/gtk/docs/GNUmakefile.am	2011-01-06 15:08:38.000000000 -0800
+++ webkitdfb_2010-11-18/WebKit/gtk/docs/GNUmakefile.am	2011-04-13 10:46:21.000000000 -0700
@@ -90,11 +90,11 @@
 	$(LIBSOUP_LIBS)
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/gtk-doc.make
+#include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST +=
+EXTRA_DIST =
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
diff -Nur webkitdfb_2010-11-18.orig/WebKitTools/GNUmakefile.am webkitdfb_2010-11-18/WebKitTools/GNUmakefile.am
--- webkitdfb_2010-11-18.orig/WebKitTools/GNUmakefile.am	2011-01-06 15:12:13.000000000 -0800
+++ webkitdfb_2010-11-18/WebKitTools/GNUmakefile.am	2011-04-13 11:19:57.613145739 -0700
@@ -81,6 +81,7 @@
 Programs_WebKitDFB_LDADD = \
 	libwebkitgtk-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la \
 	$(DIRECTFB_LIBS) \
+	$(UNICODE_LIBS) \
 	$(GLIB_LIBS)
 
 if USE_CURL
--- a/WebCore/platform/image-decoders/png/PNGImageDecoder.cpp
+++ b/WebCore/platform/image-decoders/png/PNGImageDecoder.cpp
@@ -226,7 +226,11 @@ static ColorProfile readColorProfile(png_structp png, png_infop info)
 #ifdef PNG_iCCP_SUPPORTED
     char* profileName;
     int compressionType;
-    char* profile;
+#if (PNG_LIBPNG_VER < 10500)
+    png_charp profile;
+#else
+    png_bytep profile;
+#endif
     png_uint_32 profileLength;
     if (png_get_iCCP(png, info, &profileName, &compressionType, &profile, &profileLength)) {
         ColorProfile colorProfile;
@@ -241,11 +245,11 @@ void PNGImageDecoder::headerAvailable()
 {
     png_structp png = m_reader->pngPtr();
     png_infop info = m_reader->infoPtr();
-    png_uint_32 width = png->width;
-    png_uint_32 height = png->height;
+    png_uint_32 width = png_get_image_width(png, info);
+    png_uint_32 height = png_get_image_height(png, info);
     
     // Protect against large images.
-    if (png->width > cMaxPNGSize || png->height > cMaxPNGSize) {
+    if (width > cMaxPNGSize || height > cMaxPNGSize) {
         longjmp(JMPBUF(png), 1);
         return;
     }
@@ -318,9 +322,14 @@ void PNGImageDecoder::headerAvailable()
     m_reader->setHasAlpha(channels == 4);
 
     if (m_reader->decodingSizeOnly()) {
-        // If we only needed the size, halt the reader.     
+        // If we only needed the size, halt the reader.
+#if defined(PNG_LIBPNG_VER_MAJOR) && defined(PNG_LIBPNG_VER_MINOR) && (PNG_LIBPNG_VER_MAJOR > 1 || (PNG_LIBPNG_VER_MAJOR == 1 && PNG_LIBPNG_VER_MINOR >= 5))
+        // '0' argument to png_process_data_pause means: Do not cache unprocessed data.
+        m_reader->setReadOffset(m_reader->currentBufferSize() - png_process_data_pause(png, 0));
+#else
         m_reader->setReadOffset(m_reader->currentBufferSize() - png->buffer_size);
         png->buffer_size = 0;
+#endif
     }
 }
 
@@ -343,7 +352,7 @@ void PNGImageDecoder::rowAvailable(unsigned char* rowBuffer, unsigned rowIndex,
         // For PNGs, the frame always fills the entire image.
         buffer.setOriginalFrameRect(IntRect(IntPoint(), size()));
 
-        if (m_reader->pngPtr()->interlaced)
+        if (png_get_interlace_type(m_reader->pngPtr(), m_reader->infoPtr()) != PNG_INTERLACE_NONE)
             m_reader->createInterlaceBuffer((m_reader->hasAlpha() ? 4 : 3) * size().width() * size().height());
     }
 
